<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Stephen Pederson" />


<title>Comparison of all ChIP Targets</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/core-js-2.5.3/shim.min.js"></script>
<script src="site_libs/react-17.0.0/react.min.js"></script>
<script src="site_libs/react-17.0.0/react-dom.min.js"></script>
<script src="site_libs/reactwidget-1.0.0/react-tools.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="site_libs/reactable-binding-0.3.0/reactable.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MDA-MB-453: Apocrine Signature</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="comparison.html">Comparison</a>
</li>
<li>
  <a href="motif_analysis.html">Motif Analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Comparison of all ChIP Targets</h1>
<h4 class="author">Stephen Pederson</h4>
<h4 class="date">08 June, 2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-06-08
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>apocrine_signature_mdamb453/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220427code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220427)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220427code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220427)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDRMCRLapocrinesignaturemdamb453tree1f959c32bb2c748b3bb8b8748bbc6b305c418348targetblank1f959c3a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/tree/1f959c32bb2c748b3bb8b8748bbc6b305c418348" target="_blank">1f959c3</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomDRMCRLapocrinesignaturemdamb453tree1f959c32bb2c748b3bb8b8748bbc6b305c418348targetblank1f959c3a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/tree/1f959c32bb2c748b3bb8b8748bbc6b305c418348" target="_blank">1f959c3</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/bigwig/

Unstaged changes:
    Modified:   analysis/comparison.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/comparison.Rmd</code>) and HTML
(<code>docs/comparison.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/analysis/comparison.Rmd" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
<td>
Finished checking motif distances
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/apocrine_signature_mdamb453/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/comparison.html" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
<td>
Finished checking motif distances
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/b882cb8ccca0c67325724df3bf74f151f8b8c2f1/analysis/comparison.Rmd" target="_blank">b882cb8</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-07
</td>
<td>
Ranges are no longer centred
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1c43dd8c804647b9ddd68322a9071dee324afbd/analysis/comparison.Rmd" target="_blank">e1c43dd</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-06
</td>
<td>
Tidied genomic plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/apocrine_signature_mdamb453/e1c43dd8c804647b9ddd68322a9071dee324afbd/docs/comparison.html" target="_blank">e1c43dd</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-06
</td>
<td>
Tidied genomic plots
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/480c17a7cdf7e9fb11fd6a338e135c51c1f99728/analysis/comparison.Rmd" target="_blank">480c17a</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-06
</td>
<td>
Corrected typo
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/apocrine_signature_mdamb453/480c17a7cdf7e9fb11fd6a338e135c51c1f99728/docs/comparison.html" target="_blank">480c17a</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-06
</td>
<td>
Corrected typo
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1088ac1595379dd4f700c6dfad34e638f6d977b/analysis/comparison.Rmd" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
<td>
revised comparison
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/apocrine_signature_mdamb453/e1088ac1595379dd4f700c6dfad34e638f6d977b/docs/comparison.html" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
<td>
revised comparison
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/cf12cbae86fa27c7dbb5fd5af6a81cd8e7139891/analysis/comparison.Rmd" target="_blank">cf12cba</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-01
</td>
<td>
Moved AR to the correct section
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/apocrine_signature_mdamb453/cf12cbae86fa27c7dbb5fd5af6a81cd8e7139891/docs/comparison.html" target="_blank">cf12cba</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-01
</td>
<td>
Moved AR to the correct section
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/a6276af47cce84f4875cecc503bf745b668b606c/analysis/comparison.Rmd" target="_blank">a6276af</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-01
</td>
<td>
Changed HiC plot names to HiChIP
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/apocrine_signature_mdamb453/a6276af47cce84f4875cecc503bf745b668b606c/docs/comparison.html" target="_blank">a6276af</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-01
</td>
<td>
Changed HiC plot names to HiChIP
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/88360904ce5089346cccb1f5c4c86a011fe28cd4/analysis/comparison.Rmd" target="_blank">8836090</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-26
</td>
<td>
Added motif centrality
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/apocrine_signature_mdamb453/88360904ce5089346cccb1f5c4c86a011fe28cd4/docs/comparison.html" target="_blank">8836090</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-26
</td>
<td>
Added motif centrality
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/apocrine_signature_mdamb453/65e26b822f70462eecc2b399e344026d22517a0b/docs/comparison.html" target="_blank">65e26b8</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/3a37987c1975f40afba230b3cb62e15c7e562efe/analysis/comparison.Rmd" target="_blank">3a37987</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-20
</td>
<td>
Finished first draft without motifs
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/994ec9a4dd00a58d4f3791fef5929bb29993c3f1/analysis/comparison.Rmd" target="_blank">994ec9a</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-20
</td>
<td>
Added enrichment testing
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/apocrine_signature_mdamb453/994ec9a4dd00a58d4f3791fef5929bb29993c3f1/docs/comparison.html" target="_blank">994ec9a</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-20
</td>
<td>
Added enrichment testing
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/5597186a516a1edb81751d8e0acda27d0c9475f7/analysis/comparison.Rmd" target="_blank">5597186</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-19
</td>
<td>
Added MYC plot
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1ba686b361b4d341eed2fdafce98bddff4a700d0/analysis/comparison.Rmd" target="_blank">1ba686b</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-19
</td>
<td>
EOD Wed
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e742af366dd30827c4b155751815179f90567f74/analysis/comparison.Rmd" target="_blank">e742af3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-16
</td>
<td>
Started exploring mapped genes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/apocrine_signature_mdamb453/e742af366dd30827c4b155751815179f90567f74/docs/comparison.html" target="_blank">e742af3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-16
</td>
<td>
Started exploring mapped genes
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/792bb98bbc24161495206e7b0bf8f311a9e5d287/analysis/comparison.Rmd" target="_blank">792bb98</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-13
</td>
<td>
Added dist to TSS
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/apocrine_signature_mdamb453/792bb98bbc24161495206e7b0bf8f311a9e5d287/docs/comparison.html" target="_blank">792bb98</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-13
</td>
<td>
Added dist to TSS
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e9956d4839ca1ee6c0f1660d36729ce1ce734e3b/analysis/comparison.Rmd" target="_blank">e9956d4</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-13
</td>
<td>
Added 95% CIs for targets
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/apocrine_signature_mdamb453/e9956d4839ca1ee6c0f1660d36729ce1ce734e3b/docs/comparison.html" target="_blank">e9956d4</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-13
</td>
<td>
Added 95% CIs for targets
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1582607b19db6df592097b193ebcd67fd14a8328/analysis/comparison.Rmd" target="_blank">1582607</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-12
</td>
<td>
Started looking at H3K27ac
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/8217e8a21e6dfffb0e5597e720e3baea1a6f2398/analysis/comparison.Rmd" target="_blank">8217e8a</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-10
</td>
<td>
Started explorations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/apocrine_signature_mdamb453/8217e8a21e6dfffb0e5597e720e3baea1a6f2398/docs/comparison.html" target="_blank">8217e8a</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-10
</td>
<td>
Started explorations
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<pre class="r"><code>library(tidyverse)
library(magrittr)
library(extraChIPs)
library(plyranges)
library(pander)
library(scales)
library(reactable)
library(htmltools)
library(UpSetR)
library(rtracklayer)
library(GenomicInteractions)
library(multcomp)
library(glue)
library(nnet)
library(effects)
library(Gviz)
library(readxl)
library(corrplot)
library(goseq)
library(cowplot)
library(scico)
library(BSgenome.Hsapiens.UCSC.hg19)
theme_set(theme_bw())</code></pre>
<pre class="r"><code>bar_chart &lt;- function(label, width = &quot;100%&quot;, height = &quot;16px&quot;, fill = &quot;#00bfc4&quot;, background = NULL) {
  bar &lt;- div(style = list(background = fill, width = width, height = height))
  chart &lt;- div(style = list(flexGrow = 1, marginLeft = &quot;8px&quot;, background = background), bar)
  div(style = list(display = &quot;flex&quot;, alignItems = &quot;center&quot;), label, chart)
}
bar_style &lt;- function(width = 1, fill = &quot;#e6e6e6&quot;, height = &quot;75%&quot;, align = c(&quot;left&quot;, &quot;right&quot;), color = NULL) {
  align &lt;- match.arg(align)
  if (align == &quot;left&quot;) {
    position &lt;- paste0(width * 100, &quot;%&quot;)
    image &lt;- sprintf(&quot;linear-gradient(90deg, %1$s %2$s, transparent %2$s)&quot;, fill, position)
  } else {
    position &lt;- paste0(100 - width * 100, &quot;%&quot;)
    image &lt;- sprintf(&quot;linear-gradient(90deg, transparent %1$s, %2$s %1$s)&quot;, position, fill)
  }
  list(
    backgroundImage = image,
    backgroundSize = paste(&quot;100%&quot;, height),
    backgroundRepeat = &quot;no-repeat&quot;,
    backgroundPosition = &quot;center&quot;,
    color = color
  )
}</code></pre>
<pre class="r"><code>dht_peaks &lt;- here::here(&quot;data&quot;, &quot;peaks&quot;) %&gt;% 
  list.files(recursive = TRUE, pattern = &quot;oracle&quot;, full.names = TRUE) %&gt;% 
  sapply(read_rds, simplify = FALSE) %&gt;% 
  lapply(function(x) x[[&quot;DHT&quot;]]) %&gt;% 
  lapply(setNames, nm = c()) %&gt;% 
  setNames(str_extract_all(names(.), &quot;AR|FOXA1|GATA3|TFAP2B&quot;))
targets &lt;- names(dht_peaks)
sq &lt;- seqinfo(dht_peaks[[1]])
dht_peaks %&gt;% 
  lapply(mutate, w = width) %&gt;%
  lapply(as_tibble) %&gt;%
  bind_rows(.id = &quot;target&quot;) %&gt;%
  ggplot(aes(as.factor(n_reps), w, fill = target)) +
  geom_violin(draw_quantiles = 0.5) +
  scale_y_log10() +
  labs(x = &quot;Number of Replicates&quot;, y = &quot;Peak Width&quot;, fill = &quot;Target&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/dht-peaks-1.png" alt="Peak width distribution based on the number of replicates a peak was detected in." width="960" />
<p class="caption">
Peak width distribution based on the number of replicates a peak was
detected in.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-dht-peaks-1">
Past versions of dht-peaks-1.png
</button>
</p>
<div id="fig-dht-peaks-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/comparison.Rmd/dht-peaks-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Given that narrow peaks were primarily associated with those detected
in a single replicate, the decision was made to only retain peaks
detected in <span class="math inline">\(\geq2\)</span> replicates for
each target. This would remove the following peak numbers from each
target</p>
<ul>
<li><strong>AR</strong>: 1,326 (6.8%)</li>
<li><strong>FOXA1</strong>: 1,872 (2.0%)</li>
<li><strong>GATA3</strong>: 1,245 (4.2%)</li>
<li><strong>TFAP2B</strong>: 691 (3.3%)</li>
</ul>
<!-- end of list -->
<pre class="r"><code>cp &lt;- htmltools::tags$em(
  &quot;
  Summary of all oracle peaks from DHT-treated samples. 
  FOXA1 clearly showed the most binding activity. 
  Discarded peaks are those only detected in one replicate.
  &quot;
)
tbl &lt;- dht_peaks %&gt;% 
  lapply(
    function(x) {
      tibble(
        n = sum(x$n_reps &gt; 1),
        w = median(width(x)),
        kb = sum(width(x)) / 1e3,
        discarded = sum(x$n_reps == 1)
      )
    }
  ) %&gt;% 
  lapply(list) %&gt;% 
  as_tibble() %&gt;% 
  pivot_longer(cols = everything(), names_to = &quot;target&quot;) %&gt;% 
  unnest(everything()) %&gt;% 
  reactable(
    filterable = FALSE, searchable = FALSE,
    columns = list(
      target = colDef(name = &quot;ChIP Target&quot;),
      n = colDef(name = &quot;Total Peaks&quot;),
      w = colDef(name = &quot;Median Width&quot;),
      kb = colDef(name = &quot;Total Width (kb)&quot;, format = colFormat(digits = 1)),
      discarded = colDef(name = &quot;Discarded Peaks&quot;)
    ),
    defaultColDef = colDef(
      format = colFormat(separators = TRUE, digits = 0)
    )
  )
div(class = &quot;table&quot;,
  div(class = &quot;table-header&quot;,
      div(class = &quot;caption&quot;, cp),
      tbl
  )
)</code></pre>
<div class="table">
<div class="table-header">
<div class="caption">
<em>
  Summary of all oracle peaks from DHT-treated samples. 
  FOXA1 clearly showed the most binding activity. 
  Discarded peaks are those only detected in one replicate.
  </em>
</div>
<div id="htmlwidget-9929d7c7769fe1c64f8c" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-9929d7c7769fe1c64f8c">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"target":["AR","FOXA1","GATA3","TFAP2B"],"n":[18080,90879,28421,20317],"w":[329,374,297,310],"kb":[7156.449,41675.084,10059.128,7661.389],"discarded":[1326,1872,1245,691]},"columns":[{"accessor":"target","name":"ChIP Target","type":"character","format":{"cell":{"digits":0,"separators":true},"aggregated":{"digits":0,"separators":true}}},{"accessor":"n","name":"Total Peaks","type":"numeric","format":{"cell":{"digits":0,"separators":true},"aggregated":{"digits":0,"separators":true}}},{"accessor":"w","name":"Median Width","type":"numeric","format":{"cell":{"digits":0,"separators":true},"aggregated":{"digits":0,"separators":true}}},{"accessor":"kb","name":"Total Width (kb)","type":"numeric","format":{"cell":{"digits":1},"aggregated":{"digits":1}}},{"accessor":"discarded","name":"Discarded Peaks","type":"numeric","format":{"cell":{"digits":0,"separators":true},"aggregated":{"digits":0,"separators":true}}}],"defaultPageSize":10,"paginationType":"numbers","showPageInfo":true,"minRows":1,"dataKey":"12a06796abecc2338ba2855caeac659b"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<pre class="r"><code>all_gr &lt;- here::here(&quot;data&quot;, &quot;annotations&quot;, &quot;all_gr.rds&quot;) %&gt;% 
  read_rds() 
tss &lt;- resize(all_gr$transcript, width = 1)
features &lt;- here::here(&quot;data&quot;, &quot;h3k27ac&quot;) %&gt;% 
  list.files(full.names = TRUE, pattern = &quot;bed$&quot;) %&gt;% 
  sapply(import.bed, seqinfo = sq) %&gt;% 
  lapply(granges) %&gt;% 
  setNames(basename(names(.))) %&gt;% 
  setNames(str_remove_all(names(.), &quot;s.bed&quot;)) %&gt;% 
  GRangesList() %&gt;% 
  unlist() %&gt;% 
  names_to_column(&quot;feature&quot;) %&gt;% 
  sort()
dht_consensus &lt;- dht_peaks %&gt;% 
  lapply(filter, n_reps &gt; 1) %&gt;% 
  lapply(mutate, w = width) %&gt;% 
  lapply(
    function(x) {
      x &lt;- shift(x, x$peak)
      x &lt;- resize(x, width = 1, fix = &#39;start&#39;)
      resize(x, width = round(0.9*x$w, 0), fix = &#39;center&#39;)
    }
  ) %&gt;% 
  lapply(granges) %&gt;% 
  GRangesList() %&gt;% 
  unlist() %&gt;% 
  reduce()%&gt;% 
  mutate(
    AR = overlapsAny(., dht_peaks$AR),
    FOXA1 = overlapsAny(., dht_peaks$FOXA1),
    GATA3 = overlapsAny(., dht_peaks$GATA3),
    TFAP2B = overlapsAny(., dht_peaks$TFAP2B),
    H3K27ac = propOverlap(., features) &gt; 0.5,
    promoter = bestOverlap(., features, var = &quot;feature&quot;) == &quot;promoter&quot; &amp; H3K27ac,
    enhancer = bestOverlap(., features, var = &quot;feature&quot;) == &quot;enhancer&quot; &amp; H3K27ac,
    dist_to_tss = mcols(distanceToNearest(., tss, ignore.strand = TRUE))$distance
  )</code></pre>
<p>Consensus peaks were then formed by:</p>
<ol style="list-style-type: decimal">
<li>Centering each target’s peak-set at the peak summit</li>
<li>Resizing peaks to 90% of the initial width</li>
<li>Merging any overlapping peaks from multiple targets</li>
</ol>
<p>Target overlap was defined as the merged peak having any overlap with
the set of original peaks from each target. H3K27ac overlap was
considered to be true if &gt;50% of the final peak overlapped an
H3K27ac-marked region.</p>
</div>
<div id="relationship-between-binding-regions" class="section level2">
<h2>Relationship Between Binding Regions</h2>
<p>Oracle peaks from the DHT-treated samples in each ChIP target were
obtained previously using the GRAVI workflow. AR and GATA3 peaks were
derived from the same samples/passages, whilst FOXA1 and TFAP2B ChIP-Seq
experiments were performed separately.</p>
<p>A set of target-agnostic set of binding regions was then defined as
the union of all DHT-treat peaks across all targets.</p>
<pre class="r"><code>dht_consensus %&gt;% 
  as_tibble() %&gt;% 
  pivot_longer(cols = all_of(targets), names_to = &quot;target&quot;, values_to = &quot;bound&quot;) %&gt;% 
  dplyr::filter(bound) %&gt;% 
  split(.$target) %&gt;% 
  lapply(pull, &quot;range&quot;) %&gt;% 
  fromList() %&gt;% 
  upset(
    sets = rev(targets), keep.order = TRUE, 
    order.by = &quot;freq&quot;, 
    set_size.show = TRUE, set_size.scale_max = nrow(.)
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/upset-dht-consensus-1.png" alt="Using the union of all binding regions, those which overlapped an oracle peak from each ChIP target are shown" width="960" />
<p class="caption">
Using the union of all binding regions, those which overlapped an oracle
peak from each ChIP target are shown
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-upset-dht-consensus-1">
Past versions of upset-dht-consensus-1.png
</button>
</p>
<div id="fig-upset-dht-consensus-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/comparison.Rmd/upset-dht-consensus-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1088ac1595379dd4f700c6dfad34e638f6d977b/docs/figure/comparison.Rmd/upset-dht-consensus-1.png" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e742af366dd30827c4b155751815179f90567f74/docs/figure/comparison.Rmd/upset-dht-consensus-1.png" target="_blank">e742af3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/8217e8a21e6dfffb0e5597e720e3baea1a6f2398/docs/figure/comparison.Rmd/upset-dht-consensus-1.png" target="_blank">8217e8a</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="peak-width" class="section level3">
<h3>Peak Width</h3>
<pre class="r"><code>df &lt;- dht_consensus %&gt;% 
  mutate(w = width) %&gt;% 
  as_tibble() %&gt;% 
  pivot_longer(
    cols = all_of(targets),
    names_to = &quot;targets&quot;, values_to = &quot;detected&quot;
  ) %&gt;% 
  dplyr::filter(detected) %&gt;% 
  group_by(range, H3K27ac, w) %&gt;% 
  summarise(
    targets = paste(sort(targets), collapse = &quot; &quot;),
    .groups = &quot;drop&quot;
  ) 
df %&gt;% 
  mutate(n_targets = str_count(targets, &quot; &quot;) + 1) %&gt;% 
  arrange(n_targets, targets) %&gt;% 
  mutate(targets = fct_inorder(targets)) %&gt;% 
  ggplot(aes(w, fct_rev(targets), fill = H3K27ac)) +
  geom_boxplot() +
  scale_x_log10() +
  scale_fill_scico_d(palette = &quot;berlin&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/plot-group-peak-widths-1.png" alt="In most groups, wider peaks were more strongly associated with H3K27ac signal. Similarly, wider peaks were more commonly associated with multiple targets." width="960" />
<p class="caption">
In most groups, wider peaks were more strongly associated with H3K27ac
signal. Similarly, wider peaks were more commonly associated with
multiple targets.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-group-peak-widths-1">
Past versions of plot-group-peak-widths-1.png
</button>
</p>
<div id="fig-plot-group-peak-widths-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/comparison.Rmd/plot-group-peak-widths-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="h3k27ac-signal" class="section level2 tabset">
<h2 class="tabset">H3K27ac Signal</h2>
<p>Given the hypothesis that the co-ocurrence of all 4 ChIP targets
should be associated with increased transcriptional or regulatory
activity, the set of all DHT peaks were then compared to promoters and
enhancers derived H2K27ac binding, as detected in the existing GATA3/AR
dataset produced by Leila Hosseinzadeh. Any H3K27ac peak detected in
this dataset was classified either as a promoter or enhancer, and thus
these features can be considered as the complete set of regions with
detectable H3K27ac signal from this experiment.</p>
<pre class="r"><code>fs &lt;- 12
cp &lt;- htmltools::tags$em(
  glue(
    &quot;Summary of all sites, taking the union of all sites with any target bound. &quot;,
    &quot;{comma(length(dht_consensus))} sites were found across all targets. &quot;,
    &quot;Individual ChIP targets can be searched within this table using &#39;t&#39; or &#39;f&#39; &quot;,
    &quot;to denote True or False. Summaries regarding association with H3K27ac &quot;, 
    &quot;peaks and distance to the nearest TSS are also provided. &quot;,
    &quot;The total numbers of sites overlapping H3K27ac-defined Promoters and &quot;, 
    &quot;Enhancers are shown with the % of sites being relative to to those &quot;, 
    &quot;overlapping H3K27ac-defined features&quot;
  )
)
tbl &lt;- dht_consensus %&gt;%
  as_tibble() %&gt;% 
  group_by(!!!syms(targets)) %&gt;% 
  summarise(
    n = dplyr::n(),
    nH3K27ac = sum(H3K27ac),
    H3K27ac = mean(H3K27ac),
    promoter = sum(promoter),
    enhancer = sum(enhancer),
    d = median(dist_to_tss),
    .groups = &quot;drop&quot;
  ) %&gt;% 
  arrange(desc(n)) %T&gt;%
  write_tsv(
    here::here(&quot;output/sites_by_h3k27ac.tsv&quot;)
  ) %&gt;% 
  reactable(
    filterable = TRUE,
    pagination = FALSE,
    columns = list(
      AR = colDef(
        cell = function(value) {
          ifelse(value, &quot;\u2714&quot;, &quot;\u2716&quot;)
        }, 
        style = function(value) {
          cl &lt;- ifelse(value, &quot;forestgreen&quot;, &quot;red&quot;)
          list(color = cl)
        },
        maxWidth = 60
      ),
      FOXA1 = colDef(
        cell = function(value) {
          ifelse(value, &quot;\u2714&quot;, &quot;\u2716&quot;)
        }, 
        style = function(value) {
          cl &lt;- ifelse(value, &quot;forestgreen&quot;, &quot;red&quot;)
          list(color = cl)
        },
        maxWidth = 60
      ),
      GATA3 = colDef(
        cell = function(value) {
          ifelse(value, &quot;\u2714&quot;, &quot;\u2716&quot;)
        }, 
        style = function(value) {
          cl &lt;- ifelse(value, &quot;forestgreen&quot;, &quot;red&quot;)
          list(color = cl)
        },
        maxWidth = 60
      ),
      TFAP2B = colDef(
        cell = function(value) {
          ifelse(value, &quot;\u2714&quot;, &quot;\u2716&quot;)
        }, 
        style = function(value) {
          cl &lt;- ifelse(value, &quot;forestgreen&quot;, &quot;red&quot;)
          list(color = cl)
        },
        maxWidth = 70
      ),
      n = colDef(
        name = &quot;Nbr. Sites&quot;,
        format = colFormat(separators = TRUE),
        maxWidth = 75
      ),
      nH3K27ac = colDef(
        name = &quot;Overlapping H3K27ac&quot;,
        format = colFormat(separators = TRUE),
        maxWidth = 85
      ),
      H3K27ac = colDef(
        name = &quot;% H3K27ac Overlap&quot;,
        format = colFormat(digits = 1, percent = TRUE),
        style = function(value) {
            bar_style(width = value, align = &quot;right&quot;)
        },
        maxWidth = 80
      ),
      promoter = colDef(
        name = &quot;Promoters&quot;,
        style = function(value, index) {
            bar_style(
              width = 0.5*(value / .$nH3K27ac[[index]]), align = &quot;right&quot;, 
              fill = rgb(1, 0.416, 0.416)
            )
        },
        cell = function(value, index) {
          lb &lt;- glue(&quot;{value} ({percent(value / .$nH3K27ac[[index]], 1)})&quot;)
          as.character(lb)
        },
        align = &quot;left&quot;,
        minWidth = 120
      ),
      enhancer = colDef(
        name = &quot;Enhancers&quot;,
        style = function(value, index) {
            bar_style(
              width = 0.5*(value / .$nH3K27ac[[index]]), align = &quot;left&quot;, 
              fill = rgb(1, 0.65, 0.2)
            )
        },
        cell = function(value, index) {
          lb &lt;- glue(&quot;{value} ({percent(value / .$nH3K27ac[[index]], 1)})&quot;)
          as.character(lb)
        },
        align = &quot;right&quot;,
        minWidth = 120
      ),
      d = colDef(
        name = &quot;Median Distance to TSS (bp)&quot;,
        format = colFormat(digits = 0, separators = TRUE)
      )
    ),
    style = list(fontSize = fs)
  )
div(class = &quot;table&quot;,
  div(class = &quot;table-header&quot;,
      div(class = &quot;caption&quot;, cp),
      tbl
  )
)</code></pre>
<div class="table">
<div class="table-header">
<div class="caption">
<em>Summary of all sites, taking the union of all sites with any target bound. 102,856 sites were found across all targets. Individual ChIP targets can be searched within this table using 't' or 'f' to denote True or False. Summaries regarding association with H3K27ac peaks and distance to the nearest TSS are also provided. The total numbers of sites overlapping H3K27ac-defined Promoters and Enhancers are shown with the % of sites being relative to to those overlapping H3K27ac-defined features</em>
</div>
<div id="htmlwidget-24fc5aecc494988bd31d" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-24fc5aecc494988bd31d">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"AR":[false,false,true,false,true,false,false,true,true,false,false,true,true,true,true],"FOXA1":[true,true,true,false,true,false,true,true,true,true,false,false,false,false,false],"GATA3":[false,true,true,false,true,true,false,false,false,true,true,false,true,false,true],"TFAP2B":[false,false,false,true,true,false,true,false,true,true,true,false,false,true,true],"n":[57755,9906,6553,6412,6321,4019,3967,3580,1800,1434,411,344,157,134,63],"nH3K27ac":[5668,1179,1005,4792,2674,644,2144,416,860,551,219,36,20,66,30],"H3K27ac":[0.0981386892909705,0.119018776499091,0.153364871051427,0.747348721147848,0.423034330011074,0.1602388653894,0.540458784976052,0.116201117318436,0.477777777777778,0.384239888423989,0.532846715328467,0.104651162790698,0.127388535031847,0.492537313432836,0.476190476190476],"promoter":[3541,443,192,4273,652,433,1404,122,322,233,158,17,5,41,14],"enhancer":[2127,736,813,519,2022,211,740,294,538,318,61,19,15,25,16],"d":[12020,13653,15319,56,11818,9268,2883,12283.5,7778.5,8346,1807,10653.5,13305,2636.5,4923]},"columns":[{"accessor":"AR","name":"AR","type":"logical","cell":["✖","✖","✔","✖","✔","✖","✖","✔","✔","✖","✖","✔","✔","✔","✔"],"maxWidth":60,"style":[{"color":"red"},{"color":"red"},{"color":"forestgreen"},{"color":"red"},{"color":"forestgreen"},{"color":"red"},{"color":"red"},{"color":"forestgreen"},{"color":"forestgreen"},{"color":"red"},{"color":"red"},{"color":"forestgreen"},{"color":"forestgreen"},{"color":"forestgreen"},{"color":"forestgreen"}]},{"accessor":"FOXA1","name":"FOXA1","type":"logical","cell":["✔","✔","✔","✖","✔","✖","✔","✔","✔","✔","✖","✖","✖","✖","✖"],"maxWidth":60,"style":[{"color":"forestgreen"},{"color":"forestgreen"},{"color":"forestgreen"},{"color":"red"},{"color":"forestgreen"},{"color":"red"},{"color":"forestgreen"},{"color":"forestgreen"},{"color":"forestgreen"},{"color":"forestgreen"},{"color":"red"},{"color":"red"},{"color":"red"},{"color":"red"},{"color":"red"}]},{"accessor":"GATA3","name":"GATA3","type":"logical","cell":["✖","✔","✔","✖","✔","✔","✖","✖","✖","✔","✔","✖","✔","✖","✔"],"maxWidth":60,"style":[{"color":"red"},{"color":"forestgreen"},{"color":"forestgreen"},{"color":"red"},{"color":"forestgreen"},{"color":"forestgreen"},{"color":"red"},{"color":"red"},{"color":"red"},{"color":"forestgreen"},{"color":"forestgreen"},{"color":"red"},{"color":"forestgreen"},{"color":"red"},{"color":"forestgreen"}]},{"accessor":"TFAP2B","name":"TFAP2B","type":"logical","cell":["✖","✖","✖","✔","✔","✖","✔","✖","✔","✔","✔","✖","✖","✔","✔"],"maxWidth":70,"style":[{"color":"red"},{"color":"red"},{"color":"red"},{"color":"forestgreen"},{"color":"forestgreen"},{"color":"red"},{"color":"forestgreen"},{"color":"red"},{"color":"forestgreen"},{"color":"forestgreen"},{"color":"forestgreen"},{"color":"red"},{"color":"red"},{"color":"forestgreen"},{"color":"forestgreen"}]},{"accessor":"n","name":"Nbr. Sites","type":"numeric","format":{"cell":{"separators":true},"aggregated":{"separators":true}},"maxWidth":75},{"accessor":"nH3K27ac","name":"Overlapping H3K27ac","type":"numeric","format":{"cell":{"separators":true},"aggregated":{"separators":true}},"maxWidth":85},{"accessor":"H3K27ac","name":"% H3K27ac Overlap","type":"numeric","format":{"cell":{"digits":1,"percent":true},"aggregated":{"digits":1,"percent":true}},"maxWidth":80,"style":[{"backgroundImage":"linear-gradient(90deg, transparent 90.1861310709029%, #e6e6e6 90.1861310709029%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 88.0981223500908%, #e6e6e6 88.0981223500908%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 84.6635128948573%, #e6e6e6 84.6635128948573%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 25.2651278852152%, #e6e6e6 25.2651278852152%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 57.6965669988926%, #e6e6e6 57.6965669988926%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 83.97611346106%, #e6e6e6 83.97611346106%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 45.9541215023947%, #e6e6e6 45.9541215023947%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 88.3798882681564%, #e6e6e6 88.3798882681564%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 52.2222222222222%, #e6e6e6 52.2222222222222%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 61.5760111576011%, #e6e6e6 61.5760111576011%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 46.7153284671533%, #e6e6e6 46.7153284671533%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 89.5348837209302%, #e6e6e6 89.5348837209302%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 87.2611464968153%, #e6e6e6 87.2611464968153%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 50.7462686567164%, #e6e6e6 50.7462686567164%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 52.3809523809524%, #e6e6e6 52.3809523809524%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null}]},{"accessor":"promoter","name":"Promoters","type":"numeric","cell":["3541 (62%)","443 (38%)","192 (19%)","4273 (89%)","652 (24%)","433 (67%)","1404 (65%)","122 (29%)","322 (37%)","233 (42%)","158 (72%)","17 (47%)","5 (25%)","41 (62%)","14 (47%)"],"minWidth":120,"align":"left","style":[{"backgroundImage":"linear-gradient(90deg, transparent 68.7632321806634%, #FF6A6A 68.7632321806634%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 81.2128922815946%, #FF6A6A 81.2128922815946%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 90.4477611940299%, #FF6A6A 90.4477611940299%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 55.4152754590985%, #FF6A6A 55.4152754590985%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 87.8085265519821%, #FF6A6A 87.8085265519821%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 66.3819875776397%, #FF6A6A 66.3819875776397%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 67.2574626865672%, #FF6A6A 67.2574626865672%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 85.3365384615385%, #FF6A6A 85.3365384615385%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 81.2790697674419%, #FF6A6A 81.2790697674419%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 78.8566243194192%, #FF6A6A 78.8566243194192%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 63.9269406392694%, #FF6A6A 63.9269406392694%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 76.3888888888889%, #FF6A6A 76.3888888888889%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 87.5%, #FF6A6A 87.5%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 68.9393939393939%, #FF6A6A 68.9393939393939%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, transparent 76.6666666666667%, #FF6A6A 76.6666666666667%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null}]},{"accessor":"enhancer","name":"Enhancers","type":"numeric","cell":["2127 (38%)","736 (62%)","813 (81%)","519 (11%)","2022 (76%)","211 (33%)","740 (35%)","294 (71%)","538 (63%)","318 (58%)","61 (28%)","19 (53%)","15 (75%)","25 (38%)","16 (53%)"],"minWidth":120,"align":"right","style":[{"backgroundImage":"linear-gradient(90deg, #FFA633 18.7632321806634%, transparent 18.7632321806634%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, #FFA633 31.2128922815946%, transparent 31.2128922815946%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, #FFA633 40.4477611940299%, transparent 40.4477611940299%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, #FFA633 5.4152754590985%, transparent 5.4152754590985%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, #FFA633 37.808526551982%, transparent 37.808526551982%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, #FFA633 16.3819875776398%, transparent 16.3819875776398%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, #FFA633 17.2574626865672%, transparent 17.2574626865672%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, #FFA633 35.3365384615385%, transparent 35.3365384615385%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, #FFA633 31.2790697674419%, transparent 31.2790697674419%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, #FFA633 28.8566243194192%, transparent 28.8566243194192%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, #FFA633 13.9269406392694%, transparent 13.9269406392694%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, #FFA633 26.3888888888889%, transparent 26.3888888888889%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, #FFA633 37.5%, transparent 37.5%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, #FFA633 18.9393939393939%, transparent 18.9393939393939%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null},{"backgroundImage":"linear-gradient(90deg, #FFA633 26.6666666666667%, transparent 26.6666666666667%)","backgroundSize":"100% 75%","backgroundRepeat":"no-repeat","backgroundPosition":"center","color":null}]},{"accessor":"d","name":"Median Distance to TSS (bp)","type":"numeric","format":{"cell":{"digits":0,"separators":true},"aggregated":{"digits":0,"separators":true}}}],"filterable":true,"pagination":false,"defaultPageSize":10,"paginationType":"numbers","showPageInfo":true,"minRows":1,"style":{"fontSize":12},"dataKey":"e0d2a551c551aae8f8bb91bc9de83682"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="all-h3k27ac-peaks" class="section level3">
<h3>All H3K27ac Peaks</h3>
<pre class="r"><code>grp_h3k27ac &lt;- dht_consensus %&gt;% 
  as_tibble() %&gt;% 
  dplyr::select(range, all_of(targets), H3K27ac) %&gt;% 
  pivot_longer(cols = all_of(targets), names_to = &quot;targets&quot;) %&gt;% 
  dplyr::filter(value) %&gt;% 
  group_by(range, H3K27ac) %&gt;% 
  summarise(
    targets = paste(targets, collapse = &quot;+&quot;), 
    .groups = &quot;drop&quot;
  ) %&gt;% 
  mutate(width = width(GRanges(range)))
glm_h3k27ac &lt;- glm(H3K27ac ~ 0 + targets + width, family = &quot;binomial&quot;, data = grp_h3k27ac)</code></pre>
<p>A simple logistic regression model was then run to estimate the
probability of overlapping a peak associated with the histone mark
H3K27ac for each combination of ChIP targets. Binding of TFAP2<span
class="math inline">\(\beta\)</span> in isolation clearly had the
strongest association with H3K27ac peaks, whilst the absence of
TFAP2<span class="math inline">\(\beta\)</span> clearly reduced the
likelihood of all other ChIP targets being associated with H3K27ac
signal.</p>
<pre class="r"><code>inv.logit &lt;- binomial()$linkinv
new_df &lt;- grp_h3k27ac %&gt;% 
  group_by(targets) %&gt;% 
  summarise(width = median(width), .groups = &quot;drop&quot;)
z &lt;- qnorm(1 - 0.025 / nrow(new_df))
df &lt;- predict(glm_h3k27ac, newdata = new_df, se.fit = TRUE) %&gt;% 
  as_tibble() %&gt;% 
  dplyr::select(ends_with(&quot;fit&quot;)) %&gt;% 
  cbind(new_df) %&gt;% 
  mutate(
    lwr = inv.logit(fit - z*se.fit), 
    upr = inv.logit(fit + z*se.fit), 
    p = inv.logit(fit),
    n_targets = str_count(targets, &quot;\\+&quot;) + 1,
    TFAP2B = str_detect(targets, &quot;TFAP2B&quot;),
    targets = str_remove_all(targets, &quot;\\+TFAP2B&quot;)
  ) %&gt;%
  arrange(n_targets, targets) %&gt;% 
  mutate(
    targets = fct_inorder(targets) %&gt;% fct_rev(),
    y = as.integer(targets) - 0.2 + 0.4*TFAP2B
  )
df %&gt;% 
  ggplot(aes(p, y, colour = TFAP2B)) +
  geom_point() +
  geom_errorbarh(aes(xmin = lwr, xmax = upr)) +
  scale_y_continuous(
    breaks = seq_along(levels(df$targets)),
    labels = levels(df$targets),
    expand = expansion(0.02)
  ) +
  scale_x_continuous(
    breaks = seq(0, 1, by = 0.2),
    limits = c(0, 1), expand = expansion(c(0, 0))
  ) +
  scale_colour_manual(values = c(&quot;grey30&quot;, &quot;red&quot;)) +
  labs(
    x = &quot;Probability of H3K27ac Peak Overlap&quot;,
    y = &quot;Detected Targets&quot;,
    colour = &quot;TFAP2B\nDetected&quot;
  ) +
  theme(
    legend.position = c(0.99, 0.01),
    legend.justification = c(1, 0)
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/confint-glm-h3k27ac-1.png" alt="Family-wise 95% Confidence Intervals for the probability of overlapping an H3K27ac-derived feature, based on the combinations of detected ChIP targets. Given the impact peak width has on this probablity, intervals were generated holding width to be fixed at the median width for each group of targets." width="960" />
<p class="caption">
Family-wise 95% Confidence Intervals for the probability of overlapping
an H3K27ac-derived feature, based on the combinations of detected ChIP
targets. Given the impact peak width has on this probablity, intervals
were generated holding width to be fixed at the median width for each
group of targets.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-confint-glm-h3k27ac-1">
Past versions of confint-glm-h3k27ac-1.png
</button>
</p>
<div id="fig-confint-glm-h3k27ac-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/comparison.Rmd/confint-glm-h3k27ac-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1088ac1595379dd4f700c6dfad34e638f6d977b/docs/figure/comparison.Rmd/confint-glm-h3k27ac-1.png" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/994ec9a4dd00a58d4f3791fef5929bb29993c3f1/docs/figure/comparison.Rmd/confint-glm-h3k27ac-1.png" target="_blank">994ec9a</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e742af366dd30827c4b155751815179f90567f74/docs/figure/comparison.Rmd/confint-glm-h3k27ac-1.png" target="_blank">e742af3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/792bb98bbc24161495206e7b0bf8f311a9e5d287/docs/figure/comparison.Rmd/confint-glm-h3k27ac-1.png" target="_blank">792bb98</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e9956d4839ca1ee6c0f1660d36729ce1ce734e3b/docs/figure/comparison.Rmd/confint-glm-h3k27ac-1.png" target="_blank">e9956d4</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="promoters-only" class="section level3">
<h3>Promoters Only</h3>
<pre class="r"><code>grp_prom &lt;- dht_consensus %&gt;% 
  as_tibble() %&gt;% 
  dplyr::select(range, all_of(targets), promoter) %&gt;% 
  pivot_longer(cols = all_of(targets), names_to = &quot;targets&quot;) %&gt;% 
  dplyr::filter(value) %&gt;% 
  group_by(range, promoter) %&gt;% 
  summarise(targets = paste(targets, collapse = &quot;+&quot;), .groups = &quot;drop&quot;)%&gt;% 
  mutate(width = width(GRanges(range)))
glm_prom &lt;- glm(promoter ~ 0 + targets + width, family = &quot;binomial&quot;, data = grp_prom)
df &lt;- predict(glm_prom, newdata = new_df, se.fit = TRUE) %&gt;% 
  as_tibble() %&gt;% 
  dplyr::select(ends_with(&quot;fit&quot;)) %&gt;% 
  cbind(new_df) %&gt;% 
  mutate(
    lwr = inv.logit(fit - z*se.fit), 
    upr = inv.logit(fit + z*se.fit), 
    p = inv.logit(fit),
    n_targets = str_count(targets, &quot;\\+&quot;) + 1,
    TFAP2B = str_detect(targets, &quot;TFAP2B&quot;),
    targets = str_remove_all(targets, &quot;\\+TFAP2B&quot;)
  ) %&gt;%
  arrange(n_targets, targets) %&gt;% 
  mutate(
    targets = fct_inorder(targets) %&gt;% fct_rev(),
    y = as.integer(targets) - 0.2 + 0.4*TFAP2B
  )
df %&gt;% 
  ggplot(aes(p, y, colour = TFAP2B)) +
  geom_point() +
  geom_errorbarh(aes(xmin = lwr, xmax = upr)) +
  scale_y_continuous(
    breaks = seq_along(levels(df$targets)),
    labels = levels(df$targets),
    expand = expansion(0.02)
  ) +
  scale_x_continuous(
    breaks = seq(0, 1, by = 0.2),
    limits = c(0, 1), expand = expansion(c(0, 0))
  ) +
  scale_colour_manual(values = c(&quot;grey30&quot;, &quot;red&quot;)) +
  labs(
    x = &quot;Probability of Promoter Overlap&quot;,
    y = &quot;Detected Targets&quot;,
    colour = &quot;TFAP2B\nDetected&quot;
  ) +
  theme(
    legend.position = c(0.99, 0.01),
    legend.justification = c(1, 0)
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/glm-prom-1.png" alt="Family-wise 95% Confidence Intervals for the probability of overlapping an H3K27ac-derived promoter, based on the combinations of detected ChIP targets. Given the impact peak width has on this probablity, intervals were generated holding width to be fixed at the median width for each group of targets." width="960" />
<p class="caption">
Family-wise 95% Confidence Intervals for the probability of overlapping
an H3K27ac-derived promoter, based on the combinations of detected ChIP
targets. Given the impact peak width has on this probablity, intervals
were generated holding width to be fixed at the median width for each
group of targets.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-glm-prom-1">
Past versions of glm-prom-1.png
</button>
</p>
<div id="fig-glm-prom-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/comparison.Rmd/glm-prom-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1088ac1595379dd4f700c6dfad34e638f6d977b/docs/figure/comparison.Rmd/glm-prom-1.png" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/994ec9a4dd00a58d4f3791fef5929bb29993c3f1/docs/figure/comparison.Rmd/glm-prom-1.png" target="_blank">994ec9a</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1ba686b361b4d341eed2fdafce98bddff4a700d0/docs/figure/comparison.Rmd/glm-prom-1.png" target="_blank">1ba686b</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="enhancers-only" class="section level3">
<h3>Enhancers Only</h3>
<pre class="r"><code>grp_enh &lt;- dht_consensus %&gt;% 
  as_tibble() %&gt;% 
  dplyr::select(range, all_of(targets), enhancer) %&gt;% 
  pivot_longer(cols = all_of(targets), names_to = &quot;targets&quot;) %&gt;% 
  dplyr::filter(value) %&gt;% 
  group_by(range, enhancer) %&gt;% 
  summarise(targets = paste(targets, collapse = &quot;+&quot;), .groups = &quot;drop&quot;) %&gt;% 
  mutate(width = width(GRanges(range)))
glm_enh &lt;- glm(enhancer ~ 0 + targets + width, family = &quot;binomial&quot;, data = grp_enh)
df &lt;- predict(glm_enh, newdata = new_df, se.fit = TRUE) %&gt;% 
  as_tibble() %&gt;% 
  dplyr::select(ends_with(&quot;fit&quot;)) %&gt;% 
  cbind(new_df) %&gt;% 
  mutate(
    lwr = inv.logit(fit - z*se.fit), 
    upr = inv.logit(fit + z*se.fit), 
    p = inv.logit(fit),
    n_targets = str_count(targets, &quot;\\+&quot;) + 1,
    TFAP2B = str_detect(targets, &quot;TFAP2B&quot;),
    targets = str_remove_all(targets, &quot;\\+TFAP2B&quot;)
  ) %&gt;%
  arrange(n_targets, targets) %&gt;% 
  mutate(
    targets = fct_inorder(targets) %&gt;% fct_rev(),
    y = as.integer(targets) - 0.2 + 0.4*TFAP2B
  )
df %&gt;% 
  ggplot(aes(p, y, colour = TFAP2B)) +
  geom_point() +
  geom_errorbarh(aes(xmin = lwr, xmax = upr)) +
  scale_y_continuous(
    breaks = seq_along(levels(df$targets)),
    labels = levels(df$targets),
    expand = expansion(0.02)
  ) +
  scale_x_continuous(
    breaks = seq(0, 1, by = 0.2),
    limits = c(0, 1), expand = expansion(c(0, 0))
  ) +
  scale_colour_manual(values = c(&quot;grey30&quot;, &quot;red&quot;)) +
  labs(
    x = &quot;Probability of Enhancer Overlap&quot;,
    y = &quot;Detected Targets&quot;,
    colour = &quot;TFAP2B\nDetected&quot;
  ) +
  theme(
    legend.position = c(0.99, 0.01),
    legend.justification = c(1, 0)
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/glm-enh-1.png" alt="Family-wise 95% Confidence Intervals for the probability of overlapping an H3K27ac-derived enhancer, based on the combinations of detected ChIP targets. Given the impact peak width has on this probablity, intervals were generated holding width to be fixed at the median width for each group of targets." width="960" />
<p class="caption">
Family-wise 95% Confidence Intervals for the probability of overlapping
an H3K27ac-derived enhancer, based on the combinations of detected ChIP
targets. Given the impact peak width has on this probablity, intervals
were generated holding width to be fixed at the median width for each
group of targets.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-glm-enh-1">
Past versions of glm-enh-1.png
</button>
</p>
<div id="fig-glm-enh-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/comparison.Rmd/glm-enh-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1088ac1595379dd4f700c6dfad34e638f6d977b/docs/figure/comparison.Rmd/glm-enh-1.png" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/994ec9a4dd00a58d4f3791fef5929bb29993c3f1/docs/figure/comparison.Rmd/glm-enh-1.png" target="_blank">994ec9a</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1ba686b361b4d341eed2fdafce98bddff4a700d0/docs/figure/comparison.Rmd/glm-enh-1.png" target="_blank">1ba686b</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="distance-to-nearest-tss" class="section level2">
<h2>Distance to Nearest TSS</h2>
<p>The distance to the nearest TSS was also checked for each of the
union peaks, with the vast majority of TFAP2<span
class="math inline">\(\beta\)</span>-only peaks being with 1kb of a TSS,
indicating a fundamental role for TFAP2<span
class="math inline">\(\beta\)</span> in transcriptional regulation.
These were much further away when bound in combination with other ChIP
targets, to the point that most sites where all four were found bound
were &gt;3kb from any TSS, suggesting that an alternative role for
TFAP2<span class="math inline">\(\beta\)</span> is enhancer-based gene
regulation, when acting in concert with the remaining targets.</p>
<pre class="r"><code>df &lt;- dht_consensus %&gt;% 
  as_tibble() %&gt;% 
  dplyr::select(range, all_of(targets), dist_to_tss, H3K27ac) %&gt;% 
  pivot_longer(cols = all_of(targets), names_to = &quot;targets&quot;) %&gt;% 
  dplyr::filter(value) %&gt;% 
  group_by(range, dist_to_tss, H3K27ac) %&gt;% 
  arrange(range, targets) %&gt;% 
  summarise(targets = paste(targets, collapse = &quot;+&quot;), .groups = &quot;drop&quot;) %&gt;% 
  mutate(
    n_targets = str_count(targets, &quot;\\+&quot;) + 1,
    TFAP2B = ifelse(str_detect(targets, &quot;TFAP2B&quot;), &quot;TFAP2B&quot;, &quot;No TFAP2B&quot;)
  ) %&gt;% 
  arrange(desc(n_targets), desc(targets)) %&gt;% 
  mutate(
    targets = fct_inorder(targets) %&gt;% 
      fct_relabel(str_remove_all, pattern = &quot;\\+TFAP2B&quot;)
  ) %&gt;% 
  group_by(targets, TFAP2B, H3K27ac) %&gt;% 
  mutate(
    q = rank(dist_to_tss, ties.method = &quot;first&quot;) / dplyr::n()
  ) %&gt;% 
  ungroup() %&gt;% 
  dplyr::filter(dist_to_tss &lt; 5e3)
brk &lt;- c(0, 100, 2e3, Inf)
lb &lt;- paste(c(&quot;Core (&lt;100)&quot;, &quot;Distal (&lt;2kb)&quot;), &quot;Promoter&quot;) %&gt;% 
  c(&quot;Enhancer&quot;)
a &lt;- df %&gt;% 
  mutate(
    region = dist_to_tss %&gt;% 
      cut(
        breaks = brk, include.lowest = TRUE, labels = lb
      )
  ) %&gt;% 
  group_by(targets, TFAP2B, H3K27ac, region) %&gt;%
  summarise(n = dplyr::n(), .groups = &quot;drop_last&quot;) %&gt;% 
  mutate(prop = n / sum(n)) %&gt;% 
  ungroup() %&gt;% 
  mutate(
    status = ifelse(
      H3K27ac,
      paste(&quot;Active -&quot;, TFAP2B),
      paste(&quot;Inactive -&quot;, TFAP2B)
    ) %&gt;% 
      factor(
        levels = c(
          &quot;Active - TFAP2B&quot;, &quot;Active - No TFAP2B&quot;, 
          &quot;Inactive - TFAP2B&quot;, &quot;Inactive - No TFAP2B&quot;
        )
      )
  ) %&gt;% 
  ggplot(aes(prop, fct_rev(status), fill = fct_rev(region))) +
  geom_col() +
  facet_wrap(~fct_rev(targets), ncol = 4) +
  scale_x_continuous(labels = percent) +
  scale_fill_viridis_d(direction = -1) +
  labs(
    x = &quot;Percent (%)&quot;,
    y = &quot;H3K27ac Marks + TFAP2B Status&quot;,
    fill = &quot;Regulatory Region&quot;
  ) +
  theme(legend.position = &quot;bottom&quot;)
b &lt;- df %&gt;% 
  mutate(
    region = dist_to_tss %&gt;% 
      cut(
        breaks = brk, include.lowest = TRUE, labels = lb
      )
  ) %&gt;% 
  group_by(targets, TFAP2B, H3K27ac, region) %&gt;%
  summarise(n = dplyr::n(), .groups = &quot;drop_last&quot;) %&gt;% 
  mutate(prop = n / sum(n)) %&gt;% 
  ungroup() %&gt;% 
  mutate(
    status = ifelse(
      H3K27ac,
      paste(&quot;Active -&quot;, TFAP2B),
      paste(&quot;Inactive -&quot;, TFAP2B)
    ) %&gt;% 
      factor(
        levels = c(
          &quot;Active - TFAP2B&quot;, &quot;Active - No TFAP2B&quot;, 
          &quot;Inactive - TFAP2B&quot;, &quot;Inactive - No TFAP2B&quot;
        )
      )
  ) %&gt;% 
  ggplot(aes(status, n, fill = fct_rev(region))) +
  geom_col() +
  facet_wrap(~fct_rev(targets), ncol = 4) +
  scale_y_continuous(labels = comma) +
  scale_fill_viridis_d(direction = -1) +
  labs(
    y = &quot;Number of Peaks&quot;,
    x = &quot;H3K27ac Marks + TFAP2B Status&quot;,
    fill = &quot;Regulatory Region&quot;
  ) +
  theme(
    legend.position = &quot;none&quot;,
    axis.text.x = element_text(angle = 90, hjust = 1, vjust = 1/2)
  )
plot_grid(
  a,
  b,
  labels = c(&quot;A&quot;, &quot;B&quot;),
  nrow = 2
  # rel_widths = c(0.5, 0.5)
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/reg-region-by-targets-1.png" alt="Breaking peak overlaps into overlaps with the Core Promoter (&lt;100bp from TSS), Distal Promoter (&lt;2kb from TSS) and Enhancers (&gt;2kb from TSS), the presence of TFAP2B showed in increased affinity for promoters, particularly when only binding with a single additional factor. Similarly, peaks marked as active by an overlap with H3K27ac signal were more strongly associated with promoters. Of the H3K27ac-marked peaks where TFAP2B was detected, the largets proportion which bound enhancers, was found where all three remaining targets were also detected, indicating an increased preference for enhancers with this bindin combination. Union peaks are shown A) as percentages of peaks which overlap a given region, or B) the total number of peaks which overlap a given region. The most common binding event in the dataset was clearly FOXA1 in isolation, binding regions which are not marked by H3K27ac and do not also bind TFAP2B. The next most common set of peaks was those binding FOXA1 alone in an H3K27ac marked region, followed by TFAP2B in an H3K27ac marked region." width="960" />
<p class="caption">
Breaking peak overlaps into overlaps with the Core Promoter (&lt;100bp
from TSS), Distal Promoter (&lt;2kb from TSS) and Enhancers (&gt;2kb
from TSS), the presence of TFAP2B showed in increased affinity for
promoters, particularly when only binding with a single additional
factor. Similarly, peaks marked as active by an overlap with H3K27ac
signal were more strongly associated with promoters. Of the
H3K27ac-marked peaks where TFAP2B was detected, the largets proportion
which bound enhancers, was found where all three remaining targets were
also detected, indicating an increased preference for enhancers with
this bindin combination. Union peaks are shown A) as percentages of
peaks which overlap a given region, or B) the total number of peaks
which overlap a given region. The most common binding event in the
dataset was clearly FOXA1 in isolation, binding regions which are not
marked by H3K27ac and do not also bind TFAP2B. The next most common set
of peaks was those binding FOXA1 alone in an H3K27ac marked region,
followed by TFAP2B in an H3K27ac marked region.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-reg-region-by-targets-1">
Past versions of reg-region-by-targets-1.png
</button>
</p>
<div id="fig-reg-region-by-targets-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/comparison.Rmd/reg-region-by-targets-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="comparison-with-h3k27ac-hichip-data" class="section level3">
<h3>Comparison With H3K27ac HiChIP Data</h3>
<pre class="r"><code>fl &lt;- here::here(&quot;data&quot;, &quot;hichip&quot;) %&gt;% 
  list.files(full.names = TRUE, pattern = &quot;gi.+rds&quot;) 
hic &lt;- GInteractions()
for (f in fl) {
  hic &lt;- c(hic, read_rds(f))
}
hic &lt;- sort(hic)</code></pre>
<p>H3K27ac-HiChIP obtained from <a
href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE157381">SRA</a>
and <a
href="https://github.com/DRMCRL/MDA-MB-453-H3K27Ac-HiChIP">analysed
previously</a>, was additionally included to improve mapping of peaks to
genes. Whilst the previous H3K27ac-derived features were obtained from
the same passages/experiments as GATA3 and AR, HiChIP data was obtained
from a public dataset, not produced with the DRMCRL, and were defined
using only the Vehicle controls from an Abemaciclib Vs. Vehicle
experiment. Long range interactions were included for bin sizes of
<em>5</em>, <em>10</em>, <em>20</em> and <em>40</em> kb. Before
proceeding, the comparability of the H3K27ac-HiChIP and H3K27ac-derived
features was checked. 93% of HiChIP long-range interactions overlapped a
promoter or enhancer derived from H3K27ac ChIP-seq. Conversely, 86% or
ChIP-Seq features mapped to a long-range interaction, confirming that
the H3K27ac signatures within MDA-MB-453 cells are highly comparable
across laboratories.</p>
</div>
</div>
<div id="comparison-to-key-genes" class="section level2">
<h2>Comparison To Key Genes</h2>
<pre class="r"><code>data(&quot;grch37.cytobands&quot;)
tm &lt;- read_rds(
  here::here(&quot;data&quot;, &quot;annotations&quot;, &quot;trans_models.rds&quot;)
)
bwfl &lt;- list(
  AR = here::here(&quot;data&quot;, &quot;bigwig&quot;, &quot;AR&quot;) %&gt;% 
    list.files(pattern = &quot;DHT_merged.+bw&quot;, full.names = TRUE) %&gt;% 
    BigWigFileList() %&gt;% 
    setNames(
      path(.) %&gt;% 
        basename() %&gt;% 
        str_remove_all(&quot;_merged_treat.+&quot;)
    ),
  FOXA1 = here::here(&quot;data&quot;, &quot;bigwig&quot;, &quot;FOXA1&quot;) %&gt;% 
    list.files(pattern = &quot;DHT_merged.+bw&quot;, full.names = TRUE) %&gt;% 
    BigWigFileList() %&gt;% 
    setNames(
      path(.) %&gt;% 
        basename() %&gt;% 
        str_remove_all(&quot;_merged_treat.+&quot;)
    ),
  GATA3 = here::here(&quot;data&quot;, &quot;bigwig&quot;, &quot;GATA3&quot;) %&gt;% 
    list.files(pattern = &quot;DHT_merged.+bw&quot;, full.names = TRUE) %&gt;% 
    BigWigFileList() %&gt;% 
    setNames(
      path(.) %&gt;% 
        basename() %&gt;% 
        str_remove_all(&quot;_merged_treat.+&quot;)
    ),
  TFAP2B = here::here(&quot;data&quot;, &quot;bigwig&quot;, &quot;TFAP2B&quot;) %&gt;% 
    list.files(pattern = &quot;DHT_merged.+bw&quot;, full.names = TRUE) %&gt;% 
    BigWigFileList() %&gt;% 
    setNames(
      path(.) %&gt;% 
        basename() %&gt;% 
        str_remove_all(&quot;_merged_treat.+&quot;)
    ),
  H3K27ac = here::here(&quot;data&quot;, &quot;bigwig&quot;, &quot;H3K27ac&quot;) %&gt;% 
    list.files(pattern = &quot;merged_DHT.bw&quot;, full.names = TRUE) %&gt;% 
    BigWigFileList() %&gt;% 
    setNames(
      path(.) %&gt;% 
        basename() %&gt;% 
        str_extract(&quot;DHT|Veh&quot;)
    )
) %&gt;% 
  lapply(path) %&gt;% 
  lapply(as.character) %&gt;% 
  BigWigFileList()
y_lim &lt;- here::here(&quot;data&quot;, &quot;bigwig&quot;, targets) %&gt;% 
  sapply(
    list.files, pattern = &quot;merged.*summary&quot;, full.names= TRUE, simplify = FALSE
  ) %&gt;% 
  lapply(
    function(x) lapply(x, read_tsv)
  ) %&gt;% 
  lapply(bind_rows) %&gt;% 
  lapply(function(x) c(0, max(x$score))) %&gt;% 
  setNames(basename(names(.)))
y_lim$H3K27ac &lt;- bwfl$H3K27ac %&gt;% 
  import.bw %&gt;% 
  mcols() %&gt;% 
  .[[&quot;score&quot;]] %&gt;% 
  max() %&gt;% 
  c(0) %&gt;% 
  range()</code></pre>
</div>
<div id="mapping-to-detected-genes" class="section level2 tabs">
<h2 class="tabs">Mapping To Detected Genes</h2>
<pre class="r"><code>rnaseq &lt;- here::here(&quot;data&quot;, &quot;rnaseq&quot;, &quot;dge.rds&quot;) %&gt;% 
  read_rds()
counts &lt;- here::here(&quot;data&quot;, &quot;rnaseq&quot;, &quot;counts.out.gz&quot;) %&gt;% 
  read_tsv(skip = 1) %&gt;% 
  dplyr::select(Geneid, ends_with(&quot;bam&quot;)) 
detected &lt;- counts %&gt;%
  pivot_longer(
    cols = ends_with(&quot;bam&quot;), names_to = &quot;sample&quot;, values_to = &quot;counts&quot;
  ) %&gt;% 
  mutate(detected = counts &gt; 0) %&gt;% 
  group_by(Geneid) %&gt;% 
  summarise(detected = mean(detected) &gt; 0.25, .groups = &quot;drop&quot;) %&gt;%
  dplyr::filter(detected) %&gt;% 
  pull(&quot;Geneid&quot;)
top_table &lt;- here::here(&quot;data&quot;, &quot;rnaseq&quot;, &quot;MDA-MB-453_RNASeq.tsv&quot;) %&gt;% 
  read_tsv() # Not a DHT-treatment analysis though...</code></pre>
<pre class="r"><code>dht_consensus &lt;- mapByFeature(
  dht_consensus, 
  genes = subset(all_gr$gene, gene_id %in% detected),
  prom = subset(features, feature == &quot;promoter&quot;),
  ## Given we have accuracte H3K27ac HiChIP, better mappings can be gained that way.
  ## Exclude any enhancers which overlap the HiChIP
  enh = subset(features, feature == &quot;enhancer&quot;) %&gt;% filter_by_non_overlaps(hic),
  gi = hic
) %&gt;% 
  mutate(mapped = vapply(gene_id, function(x) length(x) &gt; 0, logical(1)))
all_targets &lt;- dht_consensus %&gt;% 
  as_tibble() %&gt;% 
  dplyr::filter(if_all(all_of(targets))) %&gt;% 
  unnest(everything()) %&gt;% 
  distinct(gene_id) %&gt;% 
  pull(&quot;gene_id&quot;)</code></pre>
<p>In order to more accurately assign genes to actively transcribed
genes, the RNA-Seq dataset generated in 2013 studying DHT Vs. Vehicle in
MDA-MB-453 cells was used to define the set of genes able to be
considered as detected. As this was a polyA dataset, some ncRNA which
are not adenylated may be expressed, but will remain as undetected
despite this technological limitation. All 21,328 genes with &gt;1 read
in at least 1/4 of the samples was considered to be detected, and peaks
were only mapped to detected genes.</p>
<p>Mapping was performed using <code>mapByFeature()</code> from the
package <code>extraChIP</code> using:</p>
<ol style="list-style-type: decimal">
<li>Promoters defined externally using H3K27ac</li>
<li>Only the externally defined enhancers which contained no
H3K27ac-HiChIP overlap</li>
<li>H3K27ac HiChIP interactions for long range mapping</li>
</ol>
<p>No separate enhancers were using in the mapping steps as the
H3K27ac-HiChIP interaction were considered to provide clear associations
between peaks and target genes. By contrast, the default mapping of
<code>mapByFeature()</code> would map all genes within 100kb to any peak
considered as an enhancer</p>
<p>90% of detected genes had one or more peaks mapped to them.</p>
<p>Looking specifically at the peaks for which all four targets directly
overlapped, 10,187 of the 21,328 detected genes were mapped to at least
one directly overlapping binding region.</p>
</div>
<div id="apocrine-signature-enrichment" class="section level2">
<h2>Apocrine Signature Enrichment</h2>
<div id="subtype-gene-signatures" class="section level3">
<h3>Subtype Gene Signatures</h3>
<pre class="r"><code>hgnc &lt;- read_csv(
  here::here(&quot;data&quot;, &quot;external&quot;, &quot;hgnc-symbol-check.csv&quot;), skip = 1
) %&gt;% 
  dplyr::select(Gene = Input, gene_name = `Approved symbol`)
apo_ranks &lt;- here::here(&quot;data&quot;, &quot;external&quot;, &quot;ApoGenes.txt&quot;) %&gt;% 
  read_tsv() %&gt;% 
  left_join(hgnc) %&gt;% 
  left_join(
    as_tibble(all_gr$gene) %&gt;% 
      dplyr::select(gene_id, gene_name),
    by = &quot;gene_name&quot;
  ) %&gt;% 
  dplyr::select(starts_with(&quot;gene_&quot;), ends_with(&quot;rank&quot;)) %&gt;% 
  dplyr::filter(!is.na(gene_id))
apo_full &lt;- read_excel(
  here::here(&quot;data&quot;, &quot;external&quot;, &quot;Apocrine gene lists.xls&quot;),
  col_names = c(
    &quot;Gene&quot;, 
    paste0(&quot;ESR1&quot;, c(&quot;_logFC&quot;, &quot;_p&quot;)),
    paste0(&quot;CLCA2&quot;, c(&quot;_logFC&quot;, &quot;_p&quot;)),
    paste0(&quot;DCN&quot;, c(&quot;_logFC&quot;, &quot;_p&quot;)),
    paste0(&quot;GZMA&quot;, c(&quot;_logFC&quot;, &quot;_p&quot;)),
    paste0(&quot;MX1&quot;, c(&quot;_logFC&quot;, &quot;_p&quot;)),
    paste0(&quot;TPX2&quot;, c(&quot;_logFC&quot;, &quot;_p&quot;)),
    paste0(&quot;FABP4&quot;, c(&quot;_logFC&quot;, &quot;_p&quot;)),
    paste0(&quot;ADM&quot;, c(&quot;_logFC&quot;, &quot;_p&quot;)),
    paste0(&quot;CD83&quot;, c(&quot;_logFC&quot;, &quot;_p&quot;))
    ),
  col_types = c(&quot;text&quot;, rep(&quot;numeric&quot;, 18)),
  skip = 1
) %&gt;% 
  left_join(hgnc) %&gt;% 
  left_join(
    as_tibble(all_gr$gene) %&gt;% 
      dplyr::select(gene_id, gene_name),
    by = &quot;gene_name&quot;
  ) %&gt;% 
  dplyr::select(starts_with(&quot;gene&quot;, FALSE), matches(&quot;_(logFC|p)&quot;))
apo_genesets &lt;- colnames(apo_full) %&gt;% 
  str_subset(&quot;_p$&quot;) %&gt;% 
  sapply(
    function(x) {
      dplyr::filter(apo_full, !!sym(x) &lt; 0.05, !is.na(gene_name))$gene_name
    },
    simplify = FALSE
  ) %&gt;% 
  setNames(
    str_remove(names(.), pattern = &quot;_p&quot;)
  )
genes_by_subtype &lt;- apo_genesets %&gt;% 
  lapply(list) %&gt;% 
  as_tibble() %&gt;% 
  pivot_longer(everything(), names_to = &quot;subtype&quot;, values_to = &quot;gene_name&quot;) %&gt;% 
  unnest(everything()) %&gt;% 
  split(.$gene_name) %&gt;% 
  lapply(pull, &quot;subtype&quot;)</code></pre>
<p>The models for different subtypes, as <a
href="https://www.nature.com/articles/1208561">previously published</a>
were loaded, with a signature from each subtype being derived from genes
with an adjusted p-value &lt; 0.05, as initially published. This gave 9
different gene-sets with key marker genes denoting the subtypes:
<em>ESR1</em>, <em>CLCA2</em>, <em>DCN</em>, <em>GZMA</em>,
<em>MX1</em>, <em>TPX2</em>, <em>FABP4</em>, <em>ADM</em> and
<em>CD83</em></p>
<pre class="r"><code>apo_genesets %&gt;% 
  fromList() %&gt;% 
  upset(
    sets = rev(names(.)), 
    # keep.order = TRUE,
    order.by = &quot;freq&quot;,
    set_size.show = TRUE,
    set_size.scale_max = nrow(.) * 0.4
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/upset-signatures-1.png" alt="UpSet plot showing the association between genes contained in each signature. A relatively high degree of similarity was noted between some signature pairs, such as GZMA/CD83 and MX1/GZMA" width="960" />
<p class="caption">
UpSet plot showing the association between genes contained in each
signature. A relatively high degree of similarity was noted between some
signature pairs, such as GZMA/CD83 and MX1/GZMA
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-upset-signatures-1">
Past versions of upset-signatures-1.png
</button>
</p>
<div id="fig-upset-signatures-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1088ac1595379dd4f700c6dfad34e638f6d977b/docs/figure/comparison.Rmd/upset-signatures-1.png" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/994ec9a4dd00a58d4f3791fef5929bb29993c3f1/docs/figure/comparison.Rmd/upset-signatures-1.png" target="_blank">994ec9a</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>apo_genesets %&gt;% 
  fromList() %&gt;%
  cor() %&gt;% 
  corrplot(
    diag = FALSE, 
    addCoef.col = &quot;grey50&quot;,
    addCoefasPercent = TRUE
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/corrplot-signatures-1.png" alt="Correlation plot between signatures based on shared genes between groups" width="576" />
<p class="caption">
Correlation plot between signatures based on shared genes between groups
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-corrplot-signatures-1">
Past versions of corrplot-signatures-1.png
</button>
</p>
<div id="fig-corrplot-signatures-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1088ac1595379dd4f700c6dfad34e638f6d977b/docs/figure/comparison.Rmd/corrplot-signatures-1.png" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/994ec9a4dd00a58d4f3791fef5929bb29993c3f1/docs/figure/comparison.Rmd/corrplot-signatures-1.png" target="_blank">994ec9a</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="enrichment-testing" class="section level3 tabset">
<h3 class="tabset">Enrichment Testing</h3>
<p>The set of genes mapped to sites contain all four ChIP targets were
then tested for enrichment of any of the defined signatures. Enrichment
testing as implemented in <code>goseq</code> was used, incorporating the
number of consensus peaks mapped to a gene as the offset term for biased
sampling.</p>
<div id="all-common-sites" class="section level4">
<h4>All Common Sites</h4>
<pre class="r"><code>mapped &lt;- all_gr$gene %&gt;% 
  filter(gene_id %in% detected) %&gt;% 
  mutate(gene_width = width) %&gt;% 
  as_tibble() %&gt;% 
  dplyr::select(starts_with(&quot;gene&quot;)) %&gt;% 
  mutate(
    mapped_to_all = gene_name %in% unlist(filter(dht_consensus, !!!syms(targets))$gene_name),
    n_peaks = table(unlist(dht_consensus$gene_name))[gene_name] %&gt;% 
      as.integer(),
    n_peaks = ifelse(is.na(n_peaks), 0, n_peaks)
  ) %&gt;% 
  distinct(gene_name, .keep_all = TRUE)
goseq_all4 &lt;- structure(
  mapped$mapped_to_all, names = mapped$gene_name
) %&gt;% 
  nullp(bias.data = mapped$n_peaks) %&gt;%  
  goseq(gene2cat = genes_by_subtype) %&gt;% 
  as_tibble() %&gt;% 
  dplyr::select(
    subtype = category,
    p = over_represented_pvalue,
    n_mapped = numDEInCat,
    signature_size = numInCat
  ) %&gt;% 
  mutate(fdr = p.adjust(p, &quot;fdr&quot;))</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/pwf-all4-1.png" alt="PWF for biased sampling using the number of peaks mapped to a gene as the bias offset" width="768" />
<p class="caption">
PWF for biased sampling using the number of peaks mapped to a gene as
the bias offset
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pwf-all4-1">
Past versions of pwf-all4-1.png
</button>
</p>
<div id="fig-pwf-all4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/comparison.Rmd/pwf-all4-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1088ac1595379dd4f700c6dfad34e638f6d977b/docs/figure/comparison.Rmd/pwf-all4-1.png" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/994ec9a4dd00a58d4f3791fef5929bb29993c3f1/docs/figure/comparison.Rmd/pwf-all4-1.png" target="_blank">994ec9a</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>goseq_all4 %&gt;% 
  mutate(`%` = percent(n_mapped / signature_size, accuracy = 0.1)) %&gt;% 
  dplyr::select(
    Subtype = subtype, `Signature Size` = signature_size, `# Mapped` = n_mapped,
    `% Mapped` = `%`, p, FDR = fdr
  ) %&gt;% 
  pander(
    justify = &quot;lrrrrr&quot;,
    caption = paste(
      &quot;Results for Subtype enrichment testing using all genes mapped to any &quot;,
      &quot;consensus peak, where all four ChIP targets were considered to be present.&quot;
    ),
    emphasize.strong.rows = which(.$FDR &lt; 0.05)
  )</code></pre>
<table>
<caption>Results for Subtype enrichment testing using all genes mapped
to any consensus peak, where all four ChIP targets were considered to be
present.</caption>
<colgroup>
<col width="14%" />
<col width="20%" />
<col width="13%" />
<col width="14%" />
<col width="19%" />
<col width="19%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Subtype</th>
<th align="right">Signature Size</th>
<th align="right"># Mapped</th>
<th align="right">% Mapped</th>
<th align="right">p</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><strong>CLCA2</strong></td>
<td align="right"><strong>136</strong></td>
<td align="right"><strong>104</strong></td>
<td align="right"><strong>76.5%</strong></td>
<td align="right"><strong>0.0007285</strong></td>
<td align="right"><strong>0.006557</strong></td>
</tr>
<tr class="even">
<td align="left">ESR1</td>
<td align="right">868</td>
<td align="right">535</td>
<td align="right">61.6%</td>
<td align="right">0.05081</td>
<td align="right">0.2286</td>
</tr>
<tr class="odd">
<td align="left">ADM</td>
<td align="right">123</td>
<td align="right">74</td>
<td align="right">60.2%</td>
<td align="right">0.1779</td>
<td align="right">0.5336</td>
</tr>
<tr class="even">
<td align="left">TPX2</td>
<td align="right">333</td>
<td align="right">200</td>
<td align="right">60.1%</td>
<td align="right">0.5204</td>
<td align="right">0.946</td>
</tr>
<tr class="odd">
<td align="left">DCN</td>
<td align="right">766</td>
<td align="right">435</td>
<td align="right">56.8%</td>
<td align="right">0.8338</td>
<td align="right">0.946</td>
</tr>
<tr class="even">
<td align="left">CD83</td>
<td align="right">400</td>
<td align="right">210</td>
<td align="right">52.5%</td>
<td align="right">0.9036</td>
<td align="right">0.946</td>
</tr>
<tr class="odd">
<td align="left">GZMA</td>
<td align="right">195</td>
<td align="right">97</td>
<td align="right">49.7%</td>
<td align="right">0.9104</td>
<td align="right">0.946</td>
</tr>
<tr class="even">
<td align="left">MX1</td>
<td align="right">106</td>
<td align="right">47</td>
<td align="right">44.3%</td>
<td align="right">0.9372</td>
<td align="right">0.946</td>
</tr>
<tr class="odd">
<td align="left">FABP4</td>
<td align="right">50</td>
<td align="right">21</td>
<td align="right">42.0%</td>
<td align="right">0.946</td>
<td align="right">0.946</td>
</tr>
</tbody>
</table>
</div>
<div id="h3k27ac-associated-common-sites" class="section level4">
<h4>H3K27Ac-Associated Common Sites</h4>
<p>The same process was then performed restricting the mappings to those
sites associated with H3K27ac binding.</p>
<pre class="r"><code>mapped_h3k &lt;- all_gr$gene %&gt;% 
  filter(gene_id %in% detected) %&gt;% 
  mutate(gene_width = width) %&gt;% 
  as_tibble() %&gt;% 
  dplyr::select(starts_with(&quot;gene&quot;)) %&gt;% 
  mutate(
    mapped_to_all = gene_name %in% unlist(filter(dht_consensus, !!!syms(targets), H3K27ac)$gene_name),
    n_peaks = table(unlist(dht_consensus$gene_name))[gene_name] %&gt;% 
      as.integer(),
    n_peaks = ifelse(is.na(n_peaks), 0, n_peaks)
  ) %&gt;% 
  distinct(gene_name, .keep_all = TRUE)
goseq_h3k &lt;- structure(
  mapped_h3k$mapped_to_all, names = mapped_h3k$gene_name
) %&gt;% 
  nullp(bias.data = mapped_h3k$n_peaks) %&gt;%  
  goseq(gene2cat = genes_by_subtype) %&gt;% 
  as_tibble() %&gt;% 
  dplyr::select(
    subtype = category,
    p = over_represented_pvalue,
    n_mapped = numDEInCat,
    signature_size = numInCat
  ) %&gt;% 
  mutate(fdr = p.adjust(p, &quot;fdr&quot;))</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/pwf-h3k-1.png" alt="PWF for biased sampling using the number of peaks mapped to a gene as the bias offset" width="768" />
<p class="caption">
PWF for biased sampling using the number of peaks mapped to a gene as
the bias offset
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pwf-h3k-1">
Past versions of pwf-h3k-1.png
</button>
</p>
<div id="fig-pwf-h3k-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/comparison.Rmd/pwf-h3k-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1088ac1595379dd4f700c6dfad34e638f6d977b/docs/figure/comparison.Rmd/pwf-h3k-1.png" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/994ec9a4dd00a58d4f3791fef5929bb29993c3f1/docs/figure/comparison.Rmd/pwf-h3k-1.png" target="_blank">994ec9a</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>goseq_h3k %&gt;% 
  mutate(`%` = percent(n_mapped / signature_size, accuracy = 0.1)) %&gt;% 
  dplyr::select(
    Subtype = subtype, `Signature Size` = signature_size, `# Mapped` = n_mapped,
    `% Mapped` = `%`, p, FDR = fdr
  ) %&gt;% 
  pander(
    justify = &quot;lrrrrr&quot;,
    caption = paste(
      &quot;Results for Subtype enrichment testing using only genes mapped to a&quot;,
      &quot;consensus peak *associated with H3K27ac activity*, and where all four&quot;, 
      &quot;ChIP targets were considered to be present.&quot;
    ),
    emphasize.strong.rows = which(.$FDR &lt; 0.05)
  )</code></pre>
<table>
<caption>Results for Subtype enrichment testing using only genes mapped
to a consensus peak <em>associated with H3K27ac activity</em>, and where
all four ChIP targets were considered to be present.</caption>
<colgroup>
<col width="14%" />
<col width="20%" />
<col width="13%" />
<col width="14%" />
<col width="19%" />
<col width="19%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Subtype</th>
<th align="right">Signature Size</th>
<th align="right"># Mapped</th>
<th align="right">% Mapped</th>
<th align="right">p</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><strong>CLCA2</strong></td>
<td align="right"><strong>136</strong></td>
<td align="right"><strong>91</strong></td>
<td align="right"><strong>66.9%</strong></td>
<td align="right"><strong>5.577e-05</strong></td>
<td align="right"><strong>0.0005019</strong></td>
</tr>
<tr class="even">
<td align="left"><strong>ESR1</strong></td>
<td align="right"><strong>868</strong></td>
<td align="right"><strong>430</strong></td>
<td align="right"><strong>49.5%</strong></td>
<td align="right"><strong>0.003229</strong></td>
<td align="right"><strong>0.01453</strong></td>
</tr>
<tr class="odd">
<td align="left">ADM</td>
<td align="right">123</td>
<td align="right">55</td>
<td align="right">44.7%</td>
<td align="right">0.2842</td>
<td align="right">0.8527</td>
</tr>
<tr class="even">
<td align="left">FABP4</td>
<td align="right">50</td>
<td align="right">17</td>
<td align="right">34.0%</td>
<td align="right">0.7642</td>
<td align="right">0.9859</td>
</tr>
<tr class="odd">
<td align="left">MX1</td>
<td align="right">106</td>
<td align="right">36</td>
<td align="right">34.0%</td>
<td align="right">0.7944</td>
<td align="right">0.9859</td>
</tr>
<tr class="even">
<td align="left">TPX2</td>
<td align="right">333</td>
<td align="right">147</td>
<td align="right">44.1%</td>
<td align="right">0.8391</td>
<td align="right">0.9859</td>
</tr>
<tr class="odd">
<td align="left">GZMA</td>
<td align="right">195</td>
<td align="right">71</td>
<td align="right">36.4%</td>
<td align="right">0.8773</td>
<td align="right">0.9859</td>
</tr>
<tr class="even">
<td align="left">CD83</td>
<td align="right">400</td>
<td align="right">147</td>
<td align="right">36.8%</td>
<td align="right">0.9852</td>
<td align="right">0.9859</td>
</tr>
<tr class="odd">
<td align="left">DCN</td>
<td align="right">766</td>
<td align="right">315</td>
<td align="right">41.1%</td>
<td align="right">0.9859</td>
<td align="right">0.9859</td>
</tr>
</tbody>
</table>
</div>
<div id="h3k27ac-associated-sites-without-tfap2b"
class="section level4">
<h4>H3K27ac Associated Sites Without TFAP2B</h4>
<p>The same process was then performed restricting the mappings to those
sites associated with H3K27ac binding, where AR, FOXA1 and GATA3 are all
bound but without TFAP2B.</p>
<pre class="r"><code>mapped_noap2b &lt;- all_gr$gene %&gt;% 
  filter(gene_id %in% detected) %&gt;% 
  mutate(gene_width = width) %&gt;% 
  as_tibble() %&gt;% 
  dplyr::select(starts_with(&quot;gene&quot;)) %&gt;% 
  mutate(
    mapped_to_all = gene_name %in% unlist(filter(dht_consensus, AR, FOXA1, GATA3, !TFAP2B, H3K27ac)$gene_name),
    n_peaks = table(unlist(dht_consensus$gene_name))[gene_name] %&gt;% 
      as.integer(),
    n_peaks = ifelse(is.na(n_peaks), 0, n_peaks)
  ) %&gt;% 
  distinct(gene_name, .keep_all = TRUE)
goseq_noap2b &lt;- structure(
  mapped_noap2b$mapped_to_all, names = mapped_noap2b$gene_name
) %&gt;% 
  nullp(bias.data = mapped_noap2b$n_peaks) %&gt;%  
  goseq(gene2cat = genes_by_subtype) %&gt;% 
  as_tibble() %&gt;% 
  dplyr::select(
    subtype = category,
    p = over_represented_pvalue,
    n_mapped = numDEInCat,
    signature_size = numInCat
  ) %&gt;% 
  mutate(fdr = p.adjust(p, &quot;fdr&quot;))</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/pwf-noap2b-1.png" alt="PWF for biased sampling using the number of peaks mapped to a gene as the bias offset" width="768" />
<p class="caption">
PWF for biased sampling using the number of peaks mapped to a gene as
the bias offset
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pwf-noap2b-1">
Past versions of pwf-noap2b-1.png
</button>
</p>
<div id="fig-pwf-noap2b-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/comparison.Rmd/pwf-noap2b-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1088ac1595379dd4f700c6dfad34e638f6d977b/docs/figure/comparison.Rmd/pwf-noap2b-1.png" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/65e26b822f70462eecc2b399e344026d22517a0b/docs/figure/comparison.Rmd/pwf-noap2b-1.png" target="_blank">65e26b8</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>goseq_noap2b %&gt;% 
  mutate(`%` = percent(n_mapped / signature_size, accuracy = 0.1)) %&gt;% 
  dplyr::select(
    Subtype = subtype, `Signature Size` = signature_size, `# Mapped` = n_mapped,
    `% Mapped` = `%`, p, FDR = fdr
  ) %&gt;% 
  pander(
    justify = &quot;lrrrrr&quot;,
    caption = paste(
      &quot;Results for Subtype enrichment testing using only genes mapped to a&quot;,
      &quot;consensus peak *associated with H3K27ac activity*, and where AR, FOXA1&quot;, 
      &quot;and GATA3 were considered to be present, but in the absence of TFAP2B.&quot;
    ),
    emphasize.strong.rows = which(.$FDR &lt; 0.05)
  )</code></pre>
<table style="width:96%;">
<caption>Results for Subtype enrichment testing using only genes mapped
to a consensus peak <em>associated with H3K27ac activity</em>, and where
AR, FOXA1 and GATA3 were considered to be present, but in the absence of
TFAP2B.</caption>
<colgroup>
<col width="13%" />
<col width="23%" />
<col width="15%" />
<col width="15%" />
<col width="13%" />
<col width="13%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Subtype</th>
<th align="right">Signature Size</th>
<th align="right"># Mapped</th>
<th align="right">% Mapped</th>
<th align="right">p</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">CLCA2</td>
<td align="right">136</td>
<td align="right">46</td>
<td align="right">33.8%</td>
<td align="right">0.05437</td>
<td align="right">0.417</td>
</tr>
<tr class="even">
<td align="left">ESR1</td>
<td align="right">868</td>
<td align="right">214</td>
<td align="right">24.7%</td>
<td align="right">0.09266</td>
<td align="right">0.417</td>
</tr>
<tr class="odd">
<td align="left">DCN</td>
<td align="right">766</td>
<td align="right">168</td>
<td align="right">21.9%</td>
<td align="right">0.4691</td>
<td align="right">0.8998</td>
</tr>
<tr class="even">
<td align="left">ADM</td>
<td align="right">123</td>
<td align="right">26</td>
<td align="right">21.1%</td>
<td align="right">0.4962</td>
<td align="right">0.8998</td>
</tr>
<tr class="odd">
<td align="left">TPX2</td>
<td align="right">333</td>
<td align="right">79</td>
<td align="right">23.7%</td>
<td align="right">0.4999</td>
<td align="right">0.8998</td>
</tr>
<tr class="even">
<td align="left">FABP4</td>
<td align="right">50</td>
<td align="right">7</td>
<td align="right">14.0%</td>
<td align="right">0.7896</td>
<td align="right">0.9987</td>
</tr>
<tr class="odd">
<td align="left">MX1</td>
<td align="right">106</td>
<td align="right">16</td>
<td align="right">15.1%</td>
<td align="right">0.7964</td>
<td align="right">0.9987</td>
</tr>
<tr class="even">
<td align="left">GZMA</td>
<td align="right">195</td>
<td align="right">29</td>
<td align="right">14.9%</td>
<td align="right">0.943</td>
<td align="right">0.9987</td>
</tr>
<tr class="odd">
<td align="left">CD83</td>
<td align="right">400</td>
<td align="right">61</td>
<td align="right">15.2%</td>
<td align="right">0.9987</td>
<td align="right">0.9987</td>
</tr>
</tbody>
</table>
</div>
<div id="h3k27ac-associated-sites-without-ar" class="section level4">
<h4>H3K27ac Associated Sites Without AR</h4>
<p>The same process was then performed restricting the mappings to those
sites associated with H3K27ac binding, where FOXA1, GATA3 and TFAP2B are
all bound but without taking AR binding into account.</p>
<pre class="r"><code>mapped_noar &lt;- all_gr$gene %&gt;% 
  filter(gene_id %in% detected) %&gt;% 
  mutate(gene_width = width) %&gt;% 
  as_tibble() %&gt;% 
  dplyr::select(starts_with(&quot;gene&quot;)) %&gt;% 
  mutate(
    mapped_to_all = gene_name %in% unlist(filter(dht_consensus, FOXA1, GATA3, TFAP2B, H3K27ac)$gene_name),
    n_peaks = table(unlist(dht_consensus$gene_name))[gene_name] %&gt;% 
      as.integer(),
    n_peaks = ifelse(is.na(n_peaks), 0, n_peaks)
  ) %&gt;% 
  distinct(gene_name, .keep_all = TRUE)
goseq_noar &lt;- structure(
  mapped_noar$mapped_to_all, names = mapped_noar$gene_name
) %&gt;% 
  nullp(bias.data = mapped_noar$n_peaks) %&gt;%  
  goseq(gene2cat = genes_by_subtype) %&gt;% 
  as_tibble() %&gt;% 
  dplyr::select(
    subtype = category,
    p = over_represented_pvalue,
    n_mapped = numDEInCat,
    signature_size = numInCat
  ) %&gt;% 
  mutate(fdr = p.adjust(p, &quot;fdr&quot;))</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/pwf-noar-1.png" alt="PWF for biased sampling using the number of peaks mapped to a gene as the bias offset" width="768" />
<p class="caption">
PWF for biased sampling using the number of peaks mapped to a gene as
the bias offset
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pwf-noar-1">
Past versions of pwf-noar-1.png
</button>
</p>
<div id="fig-pwf-noar-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/comparison.Rmd/pwf-noar-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1088ac1595379dd4f700c6dfad34e638f6d977b/docs/figure/comparison.Rmd/pwf-noar-1.png" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>goseq_noar %&gt;% 
  mutate(`%` = percent(n_mapped / signature_size, accuracy = 0.1)) %&gt;% 
  dplyr::select(
    Subtype = subtype, `Signature Size` = signature_size, `# Mapped` = n_mapped,
    `% Mapped` = `%`, p, FDR = fdr
  ) %&gt;% 
  pander(
    justify = &quot;lrrrrr&quot;,
    caption = paste(
      &quot;Results for Subtype enrichment testing using only genes mapped to a&quot;,
      &quot;consensus peak *associated with H3K27ac activity*, and where FOXA1&quot;, 
      &quot;GATA3 and TFAP2B were considered to be present, but in the absence of AR.&quot;
    ),
    emphasize.strong.rows = which(.$FDR &lt; 0.05)
  )</code></pre>
<table>
<caption>Results for Subtype enrichment testing using only genes mapped
to a consensus peak <em>associated with H3K27ac activity</em>, and where
FOXA1 GATA3 and TFAP2B were considered to be present, but in the absence
of AR.</caption>
<colgroup>
<col width="14%" />
<col width="20%" />
<col width="13%" />
<col width="14%" />
<col width="19%" />
<col width="19%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Subtype</th>
<th align="right">Signature Size</th>
<th align="right"># Mapped</th>
<th align="right">% Mapped</th>
<th align="right">p</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><strong>CLCA2</strong></td>
<td align="right"><strong>136</strong></td>
<td align="right"><strong>97</strong></td>
<td align="right"><strong>71.3%</strong></td>
<td align="right"><strong>0.0004744</strong></td>
<td align="right"><strong>0.00427</strong></td>
</tr>
<tr class="even">
<td align="left">ESR1</td>
<td align="right">868</td>
<td align="right">475</td>
<td align="right">54.7%</td>
<td align="right">0.0964</td>
<td align="right">0.4338</td>
</tr>
<tr class="odd">
<td align="left">ADM</td>
<td align="right">123</td>
<td align="right">64</td>
<td align="right">52.0%</td>
<td align="right">0.2656</td>
<td align="right">0.7969</td>
</tr>
<tr class="even">
<td align="left">MX1</td>
<td align="right">106</td>
<td align="right">46</td>
<td align="right">43.4%</td>
<td align="right">0.5812</td>
<td align="right">0.9776</td>
</tr>
<tr class="odd">
<td align="left">FABP4</td>
<td align="right">50</td>
<td align="right">22</td>
<td align="right">44.0%</td>
<td align="right">0.6161</td>
<td align="right">0.9776</td>
</tr>
<tr class="even">
<td align="left">TPX2</td>
<td align="right">333</td>
<td align="right">175</td>
<td align="right">52.6%</td>
<td align="right">0.6816</td>
<td align="right">0.9776</td>
</tr>
<tr class="odd">
<td align="left">DCN</td>
<td align="right">766</td>
<td align="right">387</td>
<td align="right">50.5%</td>
<td align="right">0.7657</td>
<td align="right">0.9776</td>
</tr>
<tr class="even">
<td align="left">GZMA</td>
<td align="right">195</td>
<td align="right">83</td>
<td align="right">42.6%</td>
<td align="right">0.9199</td>
<td align="right">0.9776</td>
</tr>
<tr class="odd">
<td align="left">CD83</td>
<td align="right">400</td>
<td align="right">176</td>
<td align="right">44.0%</td>
<td align="right">0.9776</td>
<td align="right">0.9776</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="visualisations-luminal-genes" class="section level2 tabset">
<h2 class="tabset">Visualisations: Luminal Genes</h2>
<div id="myc" class="section level3">
<h3>MYC</h3>
<p>As the gene <em>MYC</em> has been identified as a key target, the
binding patterns within the gene were explored. The two binding sites 1)
with all four ChIP targets detected, 2) within 100kb of <em>MYC</em>,
and 3) which were connected using HiChIP interactions were visualised
and exported.</p>
<pre class="r"><code>gn &lt;- &quot;MYC&quot;
gr &lt;- dht_consensus %&gt;% 
  filter(!!!syms(targets), mapped, H3K27ac) %&gt;% 
  filter(
    vapply(gene_name, function(x) gn %in% x, logical(1))
  ) %&gt;% 
  mutate(d = distance(., subset(all_gr$gene, gene_name == gn))) %&gt;% 
  arrange(d) %&gt;% 
  filter(d &lt; 1e5)
fs &lt;- 14
region_plot &lt;- plotHFGC(
  gr = gr,
  hic = hic %&gt;%
    subsetByOverlaps(gr) %&gt;%
    subsetByOverlaps(
      subset(all_gr$gene, gene_name == gn)
    ) %&gt;% 
    subset(as.integer(bin_size) &lt;= 5e3),
  hicsize = 3,
  max = 5e7,
  genes = tm %&gt;%
    mutate(detected = ifelse(gene %in% detected, &quot;Detected&quot;, &quot;Not-Detected&quot;)) %&gt;%
    split(.$detected),
  genecol = c(
    Detected = &quot;forestgreen&quot;, &quot;Not-Detected&quot; = rgb(0.2, 0.2, 0.7, 0.5)
  ),
  coverage = bwfl, 
  linecol = rep(&quot;grey30&quot;, length(bwfl)),
  cytobands = grch37.cytobands,
  zoom = 1.1,
  highlight = rgb(0, 0, 1, 0.5), 
  title.width = 1, 
  col.title = &quot;black&quot;, background.title = &quot;white&quot;,
  showAxis = FALSE,
  rotation.title = 90, fontsize = fs, 
  fontface.title = 1,
  legend = FALSE,
  fontcolor.legend = &quot;black&quot;
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/plot-myc-1.png" alt="Two joint binding sites either side of *MYC* connected by HiChIP interactions to the promoter. HiChIP connections were detected at all bin sizes, however, only fine-resolutions 5kb interaction bins are shown." width="960" />
<p class="caption">
Two joint binding sites either side of <em>MYC</em> connected by HiChIP
interactions to the promoter. HiChIP connections were detected at all
bin sizes, however, only fine-resolutions 5kb interaction bins are
shown.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-myc-1">
Past versions of plot-myc-1.png
</button>
</p>
<div id="fig-plot-myc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/comparison.Rmd/plot-myc-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1088ac1595379dd4f700c6dfad34e638f6d977b/docs/figure/comparison.Rmd/plot-myc-1.png" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/994ec9a4dd00a58d4f3791fef5929bb29993c3f1/docs/figure/comparison.Rmd/plot-myc-1.png" target="_blank">994ec9a</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>region_plot[[1]]@dp@pars$fontcolor &lt;- &quot;black&quot;
region_plot$Axis@dp@pars$fontcolor &lt;- &quot;black&quot;
region_plot$Axis@dp@pars$col &lt;- &quot;grey30&quot;
if (&quot;HiC&quot; %in% names(region_plot)) {
  region_plot$HiC@name &lt;- &quot;HiChIP&quot;
  region_plot$HiC@dp@pars$fontface.title &lt;- 1
  region_plot$HiC@dp@pars$rotation.title &lt;- 0
}
region_plot$Detected@dp@pars$fontface.title &lt;- 1
region_plot$Detected@dp@pars$rotation.title &lt;- 0
region_plot$Detected@dp@pars$fontsize.group &lt;- fs
region_plot$Detected@dp@pars$cex.group &lt;- 0.8
region_plot$Detected@dp@pars$fontface.group &lt;- 1
region_plot$Detected@dp@pars$fontcolor.group &lt;- &quot;black&quot;
if (&quot;Not-Detected&quot; %in% names(region_plot)) {
  region_plot$`Not-Detected`@dp@pars$fontcolor.group &lt;- &quot;black&quot;
  region_plot$`Not-Detected`@dp@pars$fontface.title &lt;- 1
  region_plot$`Not-Detected`@dp@pars$rotation.title &lt;- 0
  region_plot$`Not-Detected`@dp@pars$fontsize.group &lt;- fs
  region_plot$`Not-Detected`@dp@pars$cex.group &lt;- 0.8
  region_plot$`Not-Detected`@dp@pars$fontface.group &lt;- 1
  region_plot$`Not-Detected`@dp@pars$just.group &lt;- &quot;right&quot;
}
region_plot$AR@dp@pars$rotation.title &lt;- 0
region_plot$FOXA1@dp@pars$rotation.title &lt;- 0
region_plot$GATA3@dp@pars$rotation.title &lt;- 0
region_plot$TFAP2B@dp@pars$rotation.title &lt;- 0
region_plot$H3K27ac@dp@pars$rotation.title &lt;- 0
region_plot$H3K27ac@dp@pars$size &lt;- 5
ax &lt;- region_plot[1:2]
top_panels &lt;- region_plot[names(region_plot) %in% c(&quot;HiC&quot;, &quot;Detected&quot;, &quot;Not-Detected&quot;)]
hl &lt;- HighlightTrack(
  region_plot[c(targets, &quot;H3K27ac&quot;)],
  range = resize(gr, width = 3*width(gr), fix = &#39;center&#39;)
)
hl@dp@pars$fill &lt;- rgb(1, 1, 1, 0)
hl@dp@pars$col &lt;- &quot;blue&quot;
hl@dp@pars$lwd &lt;- 0.8
pdf(here::here(&quot;output&quot;, glue(&quot;{gn}.pdf&quot;)), width = 10, height = 10)
plot_range &lt;- c(gr, subset(all_gr$gene, gene_name == gn)) %&gt;% 
  range(ignore.strand = TRUE) %&gt;% 
  resize(width = 1.2*width(.), fix = &#39;center&#39;)
Gviz::plotTracks(
  c(ax, top_panels, hl),
  from = start(plot_range), to = end(plot_range),
  title.width = 1,
  margin = 40,
  fontface.main = 1
  )
dev.off()</code></pre>
</div>
<div id="gata3" class="section level3">
<h3>GATA3</h3>
<pre class="r"><code>gn &lt;- &quot;GATA3&quot;
gr &lt;- dht_consensus %&gt;% 
  filter(!!!syms(targets), mapped, H3K27ac) %&gt;% 
  filter(
    vapply(gene_name, function(x) gn %in% x, logical(1))
  ) %&gt;% 
  mutate(d = distance(., subset(all_gr$gene, gene_name == gn))) %&gt;% 
  arrange(d) %&gt;% 
  subsetByOverlaps(
    hic %&gt;% 
      subset(as.integer(bin_size) == 5e3) %&gt;% 
      subsetByOverlaps(
        subset(all_gr$gene, gene_name == gn)
      )
  )
fs &lt;- 14
region_plot &lt;- plotHFGC(
  gr = gr,
  hic = hic %&gt;%
    subsetByOverlaps(gr) %&gt;%
    subsetByOverlaps(
      subset(all_gr$gene, gene_name == gn)
    ) %&gt;% 
    subset(as.integer(bin_size) &lt;= 5e3),
  hicsize = 3,
  genes = tm %&gt;%
    mutate(detected = ifelse(gene %in% detected, &quot;Detected&quot;, &quot;Not-Detected&quot;)) %&gt;%
    split(.$detected),
  genecol = c(
    Detected = &quot;forestgreen&quot;, &quot;Not-Detected&quot; = rgb(0.2, 0.2, 0.7, 0.5)
  ),
  coverage = bwfl, 
  linecol = rep(&quot;grey30&quot;, length(bwfl)),
  cytobands = grch37.cytobands,
  zoom = 1.1,
  highlight = rgb(0, 0, 1, 0.5), 
  title.width = 1, 
  col.title = &quot;black&quot;, background.title = &quot;white&quot;,
  showAxis = FALSE,
  rotation.title = 90, fontsize = fs, 
  fontface.title = 1,
  legend = FALSE,
  fontcolor.legend = &quot;black&quot;
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/plot-gata3-1.png" alt="All H3K27ac-associated peaks where all four targets were detected, and which map to *GATA3*. Only the high-resolution 5kb bins are shown." width="960" />
<p class="caption">
All H3K27ac-associated peaks where all four targets were detected, and
which map to <em>GATA3</em>. Only the high-resolution 5kb bins are
shown.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-gata3-1">
Past versions of plot-gata3-1.png
</button>
</p>
<div id="fig-plot-gata3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/comparison.Rmd/plot-gata3-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1088ac1595379dd4f700c6dfad34e638f6d977b/docs/figure/comparison.Rmd/plot-gata3-1.png" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/65e26b822f70462eecc2b399e344026d22517a0b/docs/figure/comparison.Rmd/plot-gata3-1.png" target="_blank">65e26b8</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>region_plot[[1]]@dp@pars$fontcolor &lt;- &quot;black&quot;
region_plot$Axis@dp@pars$fontcolor &lt;- &quot;black&quot;
region_plot$Axis@dp@pars$col &lt;- &quot;grey30&quot;
if (&quot;HiC&quot; %in% names(region_plot)) {
  region_plot$HiC@name &lt;- &quot;HiChIP&quot;
  region_plot$HiC@dp@pars$fontface.title &lt;- 1
  region_plot$HiC@dp@pars$rotation.title &lt;- 0
}
region_plot$Detected@dp@pars$fontface.title &lt;- 1
region_plot$Detected@dp@pars$rotation.title &lt;- 0
region_plot$Detected@dp@pars$fontsize.group &lt;- fs
region_plot$Detected@dp@pars$cex.group &lt;- 0.8
region_plot$Detected@dp@pars$fontface.group &lt;- 1
region_plot$Detected@dp@pars$fontcolor.group &lt;- &quot;black&quot;
region_plot$Detected@dp@pars$just.group &lt;- &quot;right&quot;
if (&quot;Not-Detected&quot; %in% names(region_plot)) {
  region_plot$`Not-Detected`@dp@pars$fontcolor.group &lt;- &quot;black&quot;
  region_plot$`Not-Detected`@dp@pars$fontface.title &lt;- 1
  region_plot$`Not-Detected`@dp@pars$rotation.title &lt;- 0
  region_plot$`Not-Detected`@dp@pars$fontsize.group &lt;- fs
  region_plot$`Not-Detected`@dp@pars$cex.group &lt;- 0.8
  region_plot$`Not-Detected`@dp@pars$fontface.group &lt;- 1
  region_plot$`Not-Detected`@dp@pars$just.group &lt;- &quot;right&quot;
}
region_plot$AR@dp@pars$rotation.title &lt;- 0
region_plot$FOXA1@dp@pars$rotation.title &lt;- 0
region_plot$GATA3@dp@pars$rotation.title &lt;- 0
region_plot$TFAP2B@dp@pars$rotation.title &lt;- 0
region_plot$H3K27ac@dp@pars$rotation.title &lt;- 0
region_plot$H3K27ac@dp@pars$size &lt;- 5
ax &lt;- region_plot[1:2]
top_panels &lt;- region_plot[names(region_plot) %in% c(&quot;HiC&quot;, &quot;Detected&quot;, &quot;Not-Detected&quot;)]
hl &lt;- HighlightTrack(
  region_plot[c(targets, &quot;H3K27ac&quot;)],
  range = resize(gr, width = 3*width(gr), fix = &#39;center&#39;)
)
hl@dp@pars$fill &lt;- rgb(1, 1, 1, 0)
hl@dp@pars$col &lt;- &quot;blue&quot;
hl@dp@pars$lwd &lt;- 0.8
pdf(here::here(&quot;output&quot;, glue(&quot;{gn}.pdf&quot;)), width = 10, height = 10)
plot_range &lt;- c(gr, subset(all_gr$gene, gene_name == gn)) %&gt;% 
  range(ignore.strand = TRUE) %&gt;% 
  resize(width = 1.2*width(.), fix = &#39;center&#39;)
Gviz::plotTracks(
  c(ax, top_panels, hl),
  from = start(plot_range), to = end(plot_range),
  title.width = 1,
  margin = 40,
  fontface.main = 1
  )
dev.off()</code></pre>
</div>
<div id="foxa1" class="section level3">
<h3>FOXA1</h3>
<pre class="r"><code>gn &lt;- &quot;FOXA1&quot;
gr &lt;- dht_consensus %&gt;% 
  filter(!!!syms(targets), mapped, H3K27ac) %&gt;% 
  filter(
    vapply(gene_name, function(x) gn %in% x, logical(1))
  ) %&gt;% 
  mutate(d = distance(., subset(all_gr$gene, gene_name == gn))) %&gt;% 
  arrange(d) %&gt;%
  subsetByOverlaps(
    hic %&gt;%
      subset(as.integer(bin_size) == 5e3) %&gt;%
      subsetByOverlaps(
        subset(all_gr$gene, gene_name == gn)
      )
  )
fs &lt;- 14
region_plot &lt;- plotHFGC(
  gr = gr,
  hic = hic %&gt;%
    subsetByOverlaps(gr) %&gt;%
    subsetByOverlaps(
      subset(all_gr$gene, gene_name == gn)
    ) %&gt;% 
    subset(as.integer(bin_size) &lt;= 5e3),
  max = 1e5,
  hicsize = 3,
  genes = tm %&gt;%
    mutate(detected = ifelse(gene %in% detected, &quot;Detected&quot;, &quot;Not-Detected&quot;)) %&gt;%
    split(.$detected),
  genecol = c(
    Detected = &quot;forestgreen&quot;, &quot;Not-Detected&quot; = rgb(0.2, 0.2, 0.7, 0.5)
  ),
  coverage = bwfl, 
  linecol = rep(&quot;grey30&quot;, length(bwfl)),
  cytobands = grch37.cytobands,
  zoom = 1.2,
  highlight = rgb(0, 0, 1, 0.5), 
  title.width = 1, 
  col.title = &quot;black&quot;, background.title = &quot;white&quot;,
  showAxis = FALSE,
  rotation.title = 90, fontsize = fs, 
  fontface.title = 1,
  legend = FALSE,
  fontcolor.legend = &quot;black&quot;
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/plot-foxa1-1.png" alt="All H3K27ac-associated peaks where all four targets were detected, and which map to *FOXA1*. Only the high-resolution 5kb bins are shown." width="960" />
<p class="caption">
All H3K27ac-associated peaks where all four targets were detected, and
which map to <em>FOXA1</em>. Only the high-resolution 5kb bins are
shown.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-foxa1-1">
Past versions of plot-foxa1-1.png
</button>
</p>
<div id="fig-plot-foxa1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/comparison.Rmd/plot-foxa1-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1088ac1595379dd4f700c6dfad34e638f6d977b/docs/figure/comparison.Rmd/plot-foxa1-1.png" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/65e26b822f70462eecc2b399e344026d22517a0b/docs/figure/comparison.Rmd/plot-foxa1-1.png" target="_blank">65e26b8</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>region_plot[[1]]@dp@pars$fontcolor &lt;- &quot;black&quot;
region_plot$Axis@dp@pars$fontcolor &lt;- &quot;black&quot;
region_plot$Axis@dp@pars$col &lt;- &quot;grey30&quot;
if (&quot;HiC&quot; %in% names(region_plot)) {
  region_plot$HiC@name &lt;- &quot;HiChIP&quot;
  region_plot$HiC@dp@pars$fontface.title &lt;- 1
  region_plot$HiC@dp@pars$rotation.title &lt;- 0
}
region_plot$Detected@dp@pars$fontface.title &lt;- 1
region_plot$Detected@dp@pars$rotation.title &lt;- 0
region_plot$Detected@dp@pars$fontsize.group &lt;- fs
region_plot$Detected@dp@pars$cex.group &lt;- 0.8
region_plot$Detected@dp@pars$fontface.group &lt;- 1
region_plot$Detected@dp@pars$fontcolor.group &lt;- &quot;black&quot;
if (&quot;Not-Detected&quot; %in% names(region_plot)) {
  region_plot$`Not-Detected`@dp@pars$fontcolor.group &lt;- &quot;black&quot;
  region_plot$`Not-Detected`@dp@pars$fontface.title &lt;- 1
  region_plot$`Not-Detected`@dp@pars$rotation.title &lt;- 0
  region_plot$`Not-Detected`@dp@pars$fontsize.group &lt;- fs
  region_plot$`Not-Detected`@dp@pars$cex.group &lt;- 0.8
  region_plot$`Not-Detected`@dp@pars$fontface.group &lt;- 1
  region_plot$`Not-Detected`@dp@pars$just.group &lt;- &quot;right&quot;
}
region_plot$AR@dp@pars$rotation.title &lt;- 0
region_plot$FOXA1@dp@pars$rotation.title &lt;- 0
region_plot$GATA3@dp@pars$rotation.title &lt;- 0
region_plot$TFAP2B@dp@pars$rotation.title &lt;- 0
region_plot$H3K27ac@dp@pars$rotation.title &lt;- 0
region_plot$H3K27ac@dp@pars$size &lt;- 5
ax &lt;- region_plot[1:2]
top_panels &lt;- region_plot[names(region_plot) %in% c(&quot;HiC&quot;, &quot;Detected&quot;, &quot;Not-Detected&quot;)]
hl &lt;- HighlightTrack(
  region_plot[c(targets, &quot;H3K27ac&quot;)],
  range = resize(gr, width = 1.5 * width(gr), fix = &#39;center&#39;)
)
hl@dp@pars$fill &lt;- rgb(1, 1, 1, 0)
hl@dp@pars$col &lt;- &quot;blue&quot;
hl@dp@pars$lwd &lt;- 0.8
pdf(here::here(&quot;output&quot;, glue(&quot;{gn}.pdf&quot;)), width = 10, height = 10)
plot_range &lt;- c(gr, subset(all_gr$gene, gene_name == gn)) %&gt;% 
  range(ignore.strand = TRUE) %&gt;% 
  resize(width = 1.2*width(.), fix = &#39;center&#39;)
Gviz::plotTracks(
  c(ax, top_panels, hl),
  from = start(plot_range), to = end(plot_range),
  title.width = 1,
  margin = 40,
  fontface.main = 1
  )
dev.off()</code></pre>
</div>
<div id="xbp1" class="section level3">
<h3>XBP1</h3>
<pre class="r"><code>gn &lt;- &quot;XBP1&quot;
gr &lt;- dht_consensus %&gt;% 
  filter(!!!syms(targets), mapped, H3K27ac) %&gt;% 
  filter(
    vapply(gene_name, function(x) gn %in% x, logical(1))
  ) %&gt;% 
  mutate(d = distance(., subset(all_gr$gene, gene_name == gn))) %&gt;% 
  arrange(d) %&gt;% 
  filter(d &lt; 100e3)
fs &lt;- 14
region_plot &lt;- plotHFGC(
  gr = gr,
  hic = hic %&gt;%
    subsetByOverlaps(gr) %&gt;%
    subsetByOverlaps(
      subset(all_gr$gene, gene_name == gn)
    ) %&gt;% 
    subset(as.integer(bin_size) == 5e3),
  hicsize = 3,
  genes = tm %&gt;%
    mutate(detected = ifelse(gene %in% detected, &quot;Detected&quot;, &quot;Not-Detected&quot;)) %&gt;%
    split(.$detected),
  genecol = c(
    Detected = &quot;forestgreen&quot;, &quot;Not-Detected&quot; = rgb(0.2, 0.2, 0.7, 0.5)
  ),
  coverage = bwfl, 
  linecol = rep(&quot;grey30&quot;, length(bwfl)),
  cytobands = grch37.cytobands,
  zoom = 1.1,
  highlight = rgb(0, 0, 1, 0.5), 
  title.width = 1, 
  col.title = &quot;black&quot;, background.title = &quot;white&quot;,
  showAxis = FALSE,
  rotation.title = 90, fontsize = fs, 
  fontface.title = 1,
  legend = FALSE,
  fontcolor.legend = &quot;black&quot;
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/plot-xbp1-1.png" alt="All H3K27ac-associated peaks where all four targets were detected, and which are within 100kb of *XBP1*. Only the high-resolution 5kb interaction bins are shown. The two additional peaks were not considered to show evidence of TFAP2B binding by macs2 callpeak and are not highlighted for this reason" width="960" />
<p class="caption">
All H3K27ac-associated peaks where all four targets were detected, and
which are within 100kb of <em>XBP1</em>. Only the high-resolution 5kb
interaction bins are shown. The two additional peaks were not considered
to show evidence of TFAP2B binding by macs2 callpeak and are not
highlighted for this reason
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-xbp1-1">
Past versions of plot-xbp1-1.png
</button>
</p>
<div id="fig-plot-xbp1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/comparison.Rmd/plot-xbp1-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1088ac1595379dd4f700c6dfad34e638f6d977b/docs/figure/comparison.Rmd/plot-xbp1-1.png" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/65e26b822f70462eecc2b399e344026d22517a0b/docs/figure/comparison.Rmd/plot-xbp1-1.png" target="_blank">65e26b8</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>region_plot[[1]]@dp@pars$fontcolor &lt;- &quot;black&quot;
region_plot$Axis@dp@pars$fontcolor &lt;- &quot;black&quot;
region_plot$Axis@dp@pars$col &lt;- &quot;grey30&quot;
if (&quot;HiC&quot; %in% names(region_plot)) {
  region_plot$HiC@name &lt;- &quot;HiChIP&quot;
  region_plot$HiC@dp@pars$fontface.title &lt;- 1
  region_plot$HiC@dp@pars$rotation.title &lt;- 0
}
region_plot$Detected@dp@pars$fontface.title &lt;- 1
region_plot$Detected@dp@pars$rotation.title &lt;- 0
region_plot$Detected@dp@pars$fontsize.group &lt;- fs
region_plot$Detected@dp@pars$cex.group &lt;- 0.8
region_plot$Detected@dp@pars$fontface.group &lt;- 1
region_plot$Detected@dp@pars$fontcolor.group &lt;- &quot;black&quot;
if (&quot;Not-Detected&quot; %in% names(region_plot)) {
  region_plot$`Not-Detected`@dp@pars$fontcolor.group &lt;- &quot;black&quot;
  region_plot$`Not-Detected`@dp@pars$fontface.title &lt;- 1
  region_plot$`Not-Detected`@dp@pars$rotation.title &lt;- 0
  region_plot$`Not-Detected`@dp@pars$fontsize.group &lt;- fs
  region_plot$`Not-Detected`@dp@pars$cex.group &lt;- 0.8
  region_plot$`Not-Detected`@dp@pars$fontface.group &lt;- 1
  region_plot$`Not-Detected`@dp@pars$just.group &lt;- &quot;right&quot;
}
region_plot$AR@dp@pars$rotation.title &lt;- 0
region_plot$FOXA1@dp@pars$rotation.title &lt;- 0
region_plot$GATA3@dp@pars$rotation.title &lt;- 0
region_plot$TFAP2B@dp@pars$rotation.title &lt;- 0
region_plot$H3K27ac@dp@pars$rotation.title &lt;- 0
region_plot$H3K27ac@dp@pars$size &lt;- 5
ax &lt;- region_plot[1:2]
top_panels &lt;- region_plot[names(region_plot) %in% c(&quot;HiC&quot;, &quot;Detected&quot;, &quot;Not-Detected&quot;)]
hl &lt;- HighlightTrack(
  region_plot[c(targets, &quot;H3K27ac&quot;)],
  range = resize(gr, width = 1.5 * width(gr), fix = &#39;center&#39;)
)
hl@dp@pars$fill &lt;- rgb(1, 1, 1, 0)
hl@dp@pars$col &lt;- &quot;blue&quot;
hl@dp@pars$lwd &lt;- 0.8
pdf(here::here(&quot;output&quot;, glue(&quot;{gn}.pdf&quot;)), width = 10, height = 10)
plot_range &lt;- c(gr, subset(all_gr$gene, gene_name == gn)) %&gt;% 
  range(ignore.strand = TRUE) %&gt;% 
  resize(width = 1.2*width(.), fix = &#39;center&#39;)
Gviz::plotTracks(
  c(ax, top_panels, hl),
  from = start(plot_range), to = end(plot_range),
  title.width = 1,
  margin = 40,
  fontface.main = 1
  )
dev.off()</code></pre>
</div>
<div id="myb" class="section level3">
<h3>MYB</h3>
<pre class="r"><code>gn &lt;- &quot;MYB&quot;
gr &lt;- dht_consensus %&gt;% 
  filter(!!!syms(targets), mapped, H3K27ac) %&gt;% 
  filter(
    vapply(gene_name, function(x) gn %in% x, logical(1))
  ) %&gt;% 
  mutate(d = distance(., subset(all_gr$gene, gene_name == gn))) %&gt;% 
  arrange(d) 
fs &lt;- 14
region_plot &lt;- plotHFGC(
  gr = gr,
  hic = hic %&gt;%
    subsetByOverlaps(gr) %&gt;%
    subsetByOverlaps(
      subset(all_gr$gene, gene_name == gn)
    ) %&gt;% 
    subset(as.integer(bin_size) == 20e3),
  hicsize = 3,
  genes = tm %&gt;%
    mutate(detected = ifelse(gene %in% detected, &quot;Detected&quot;, &quot;Not-Detected&quot;)) %&gt;%
    split(.$detected),
  genecol = c(
    Detected = &quot;forestgreen&quot;, &quot;Not-Detected&quot; = rgb(0.2, 0.2, 0.7, 0.5)
  ),
  coverage = bwfl, 
  linecol = rep(&quot;grey30&quot;, length(bwfl)),
  cytobands = grch37.cytobands,
  zoom = 1.1,
  highlight = rgb(0, 0, 1, 0.5), 
  title.width = 1, 
  col.title = &quot;black&quot;, background.title = &quot;white&quot;,
  showAxis = FALSE,
  rotation.title = 90, fontsize = fs, 
  fontface.title = 1,
  legend = FALSE,
  fontcolor.legend = &quot;black&quot;
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/plot-myb-1.png" alt="All H3K27ac-associated peaks where all four targets were detected, and which map to *MYB*. Only the low-resolution 20kb interaction bins are shown." width="960" />
<p class="caption">
All H3K27ac-associated peaks where all four targets were detected, and
which map to <em>MYB</em>. Only the low-resolution 20kb interaction bins
are shown.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-myb-1">
Past versions of plot-myb-1.png
</button>
</p>
<div id="fig-plot-myb-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/comparison.Rmd/plot-myb-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1088ac1595379dd4f700c6dfad34e638f6d977b/docs/figure/comparison.Rmd/plot-myb-1.png" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/65e26b822f70462eecc2b399e344026d22517a0b/docs/figure/comparison.Rmd/plot-myb-1.png" target="_blank">65e26b8</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>region_plot[[1]]@dp@pars$fontcolor &lt;- &quot;black&quot;
region_plot$Axis@dp@pars$fontcolor &lt;- &quot;black&quot;
region_plot$Axis@dp@pars$col &lt;- &quot;grey30&quot;
if (&quot;HiC&quot; %in% names(region_plot)) {
  region_plot$HiC@name &lt;- &quot;HiChIP&quot;
  region_plot$HiC@dp@pars$fontface.title &lt;- 1
  region_plot$HiC@dp@pars$rotation.title &lt;- 0
}
region_plot$Detected@dp@pars$fontface.title &lt;- 1
region_plot$Detected@dp@pars$rotation.title &lt;- 0
region_plot$Detected@dp@pars$fontsize.group &lt;- fs
region_plot$Detected@dp@pars$cex.group &lt;- 0.8
region_plot$Detected@dp@pars$fontface.group &lt;- 1
region_plot$Detected@dp@pars$fontcolor.group &lt;- &quot;black&quot;
if (&quot;Not-Detected&quot; %in% names(region_plot)) {
  region_plot$`Not-Detected`@dp@pars$fontcolor.group &lt;- &quot;black&quot;
  region_plot$`Not-Detected`@dp@pars$fontface.title &lt;- 1
  region_plot$`Not-Detected`@dp@pars$rotation.title &lt;- 0
  region_plot$`Not-Detected`@dp@pars$fontsize.group &lt;- fs
  region_plot$`Not-Detected`@dp@pars$cex.group &lt;- 0.8
  region_plot$`Not-Detected`@dp@pars$fontface.group &lt;- 1
  region_plot$`Not-Detected`@dp@pars$just.group &lt;- &quot;right&quot;
}
region_plot$AR@dp@pars$rotation.title &lt;- 0
region_plot$FOXA1@dp@pars$rotation.title &lt;- 0
region_plot$GATA3@dp@pars$rotation.title &lt;- 0
region_plot$TFAP2B@dp@pars$rotation.title &lt;- 0
region_plot$H3K27ac@dp@pars$rotation.title &lt;- 0
region_plot$H3K27ac@dp@pars$size &lt;- 5
ax &lt;- region_plot[1:2]
top_panels &lt;- region_plot[names(region_plot) %in% c(&quot;HiC&quot;, &quot;Detected&quot;, &quot;Not-Detected&quot;)]
hl &lt;- HighlightTrack(
  region_plot[c(targets, &quot;H3K27ac&quot;)],
  range = resize(gr, width = 1.5 * width(gr), fix = &#39;center&#39;)
)
hl@dp@pars$fill &lt;- rgb(1, 1, 1, 0)
hl@dp@pars$col &lt;- &quot;blue&quot;
hl@dp@pars$lwd &lt;- 0.8
pdf(here::here(&quot;output&quot;, glue(&quot;{gn}.pdf&quot;)), width = 10, height = 10)
plot_range &lt;- c(gr, subset(all_gr$gene, gene_name == gn)) %&gt;% 
  range(ignore.strand = TRUE) %&gt;% 
  resize(width = 1.2*width(.), fix = &#39;center&#39;)
Gviz::plotTracks(
  c(ax, top_panels, hl),
  from = start(plot_range), to = end(plot_range),
  title.width = 1,
  margin = 40,
  fontface.main = 1
  )
dev.off()</code></pre>
</div>
<div id="ar" class="section level3">
<h3>AR</h3>
<pre class="r"><code>gn &lt;- &quot;AR&quot;
gr &lt;- dht_consensus %&gt;% 
  filter(!!!syms(targets), mapped, H3K27ac) %&gt;% 
  filter(
    vapply(gene_name, function(x) gn %in% x, logical(1))
  ) %&gt;% 
  mutate(d = distance(., subset(all_gr$gene, gene_name == gn))) %&gt;% 
  filter(d &lt; 1e5)
fs &lt;- 14
region_plot &lt;- plotHFGC(
  gr = gr,
  hic = hic %&gt;%
    subsetByOverlaps(gr) %&gt;%
    subsetByOverlaps(
      subset(all_gr$gene, gene_name == gn)
    ) %&gt;%
    subset(as.integer(bin_size) == 20e3),
  hicsize = 3,
  max = 5e4,
  genes = tm %&gt;%
    mutate(detected = ifelse(gene %in% detected, &quot;Detected&quot;, &quot;Not-Detected&quot;)) %&gt;%
    split(.$detected),
  genecol = c(
    Detected = &quot;forestgreen&quot;, &quot;Not-Detected&quot; = rgb(0.2, 0.2, 0.7, 0.5)
  ),
  coverage = bwfl, 
  linecol = rep(&quot;grey30&quot;, length(bwfl)),
  cytobands = grch37.cytobands,
  zoom = 6,
  shift = -6e4,
  highlight = rgb(0, 0, 1, 0.5), 
  title.width = 1, 
  col.title = &quot;black&quot;, background.title = &quot;white&quot;,
  showAxis = FALSE,
  rotation.title = 90, fontsize = fs, 
  fontface.title = 1,
  legend = FALSE,
  fontcolor.legend = &quot;black&quot;
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/plot-ar-1.png" alt="All H3K27ac-associated peaks where all four targets were detected, which map to *AR* and are within 100kb of the gene. Only the low-resolution 20kb HiC interaction bins are shown." width="960" />
<p class="caption">
All H3K27ac-associated peaks where all four targets were detected, which
map to <em>AR</em> and are within 100kb of the gene. Only the
low-resolution 20kb HiC interaction bins are shown.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-ar-1">
Past versions of plot-ar-1.png
</button>
</p>
<div id="fig-plot-ar-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/comparison.Rmd/plot-ar-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1088ac1595379dd4f700c6dfad34e638f6d977b/docs/figure/comparison.Rmd/plot-ar-1.png" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/a6276af47cce84f4875cecc503bf745b668b606c/docs/figure/comparison.Rmd/plot-ar-1.png" target="_blank">a6276af</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>region_plot[[1]]@dp@pars$fontcolor &lt;- &quot;black&quot;
region_plot$Axis@dp@pars$fontcolor &lt;- &quot;black&quot;
region_plot$Axis@dp@pars$col &lt;- &quot;grey30&quot;
if (&quot;HiC&quot; %in% names(region_plot)) {
  region_plot$HiC@name &lt;- &quot;HiChIP&quot;
  region_plot$HiC@dp@pars$fontface.title &lt;- 1
  region_plot$HiC@dp@pars$rotation.title &lt;- 0
}
region_plot$Detected@dp@pars$fontface.title &lt;- 1
region_plot$Detected@dp@pars$rotation.title &lt;- 0
region_plot$Detected@dp@pars$fontsize.group &lt;- fs
region_plot$Detected@dp@pars$cex.group &lt;- 0.8
region_plot$Detected@dp@pars$fontface.group &lt;- 1
region_plot$Detected@dp@pars$fontcolor.group &lt;- &quot;black&quot;
if (&quot;Not-Detected&quot; %in% names(region_plot)) {
  region_plot$`Not-Detected`@dp@pars$fontcolor.group &lt;- &quot;black&quot;
  region_plot$`Not-Detected`@dp@pars$fontface.title &lt;- 1
  region_plot$`Not-Detected`@dp@pars$rotation.title &lt;- 0
  region_plot$`Not-Detected`@dp@pars$fontsize.group &lt;- fs
  region_plot$`Not-Detected`@dp@pars$cex.group &lt;- 0.8
  region_plot$`Not-Detected`@dp@pars$fontface.group &lt;- 1
  region_plot$`Not-Detected`@dp@pars$just.group &lt;- &quot;right&quot;
}
region_plot$AR@dp@pars$rotation.title &lt;- 0
region_plot$FOXA1@dp@pars$rotation.title &lt;- 0
region_plot$GATA3@dp@pars$rotation.title &lt;- 0
region_plot$TFAP2B@dp@pars$rotation.title &lt;- 0
region_plot$H3K27ac@dp@pars$rotation.title &lt;- 0
region_plot$H3K27ac@dp@pars$size &lt;- 5
ax &lt;- region_plot[1:2]
top_panels &lt;- region_plot[names(region_plot) %in% c(&quot;HiC&quot;, &quot;Detected&quot;, &quot;Not-Detected&quot;)]
hl &lt;- HighlightTrack(
  region_plot[c(targets, &quot;H3K27ac&quot;)],
  range = resize(gr, width = 3 * width(gr), fix = &#39;center&#39;) %&gt;% 
    range()
)
hl@dp@pars$fill &lt;- rgb(1, 1, 1, 0)
hl@dp@pars$col &lt;- &quot;blue&quot;
hl@dp@pars$lwd &lt;- 0.8
pdf(here::here(&quot;output&quot;, glue(&quot;{gn}.pdf&quot;)), width = 10, height = 10)
plot_range &lt;- c(gr, subset(all_gr$gene, gene_name == gn)) %&gt;% 
  range(ignore.strand = TRUE) %&gt;% 
  resize(width = 1.2*width(.), fix = &#39;center&#39;)
Gviz::plotTracks(
  c(ax, top_panels, hl),
  from = start(plot_range), to = end(plot_range),
  title.width = 1,
  margin = 40,
  fontface.main = 1
  )
dev.off()</code></pre>
</div>
</div>
<div id="visualisations-apocrine-genes" class="section level2 tabset">
<h2 class="tabset">Visualisations:: Apocrine Genes</h2>
<div id="clca2" class="section level3">
<h3>CLCA2</h3>
<pre class="r"><code>gn &lt;- &quot;CLCA2&quot;
gr &lt;- dht_consensus %&gt;% 
  filter(!!!syms(targets), mapped, H3K27ac) %&gt;% 
  filter(
    vapply(gene_name, function(x) gn %in% x, logical(1))
  ) %&gt;% 
  mutate(d = distance(., subset(all_gr$gene, gene_name == gn))) %&gt;% 
  arrange(d) 
fs &lt;- 14
region_plot &lt;- plotHFGC(
  gr = gr,
  genes = tm %&gt;%
    mutate(detected = ifelse(gene %in% detected, &quot;Detected&quot;, &quot;Not-Detected&quot;)) %&gt;%
    split(.$detected),
  genecol = c(
    Detected = &quot;forestgreen&quot;, &quot;Not-Detected&quot; = rgb(0.2, 0.2, 0.7, 0.5)
  ),
  collapseTranscripts = &quot;meta&quot;,
  coverage = bwfl, 
  linecol = rep(&quot;grey30&quot;, length(bwfl)),
  cytobands = grch37.cytobands,
  zoom = 4,
  shift = 1e4,
  highlight = rgb(0, 0, 1, 0.5), 
  title.width = 1, 
  col.title = &quot;black&quot;, background.title = &quot;white&quot;,
  showAxis = FALSE,
  rotation.title = 90, fontsize = fs, 
  fontface.title = 1,
  legend = FALSE,
  fontcolor.legend = &quot;black&quot;
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/plot-clca2-1.png" alt="All H3K27ac-associated peaks where all four targets were detected, and which within the transcribed region for *CLCA2*. No HiC data is shown." width="960" />
<p class="caption">
All H3K27ac-associated peaks where all four targets were detected, and
which within the transcribed region for <em>CLCA2</em>. No HiC data is
shown.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-clca2-1">
Past versions of plot-clca2-1.png
</button>
</p>
<div id="fig-plot-clca2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/comparison.Rmd/plot-clca2-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1088ac1595379dd4f700c6dfad34e638f6d977b/docs/figure/comparison.Rmd/plot-clca2-1.png" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/65e26b822f70462eecc2b399e344026d22517a0b/docs/figure/comparison.Rmd/plot-clca2-1.png" target="_blank">65e26b8</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>region_plot[[1]]@dp@pars$fontcolor &lt;- &quot;black&quot;
region_plot$Axis@dp@pars$fontcolor &lt;- &quot;black&quot;
region_plot$Axis@dp@pars$col &lt;- &quot;grey30&quot;
if (&quot;HiC&quot; %in% names(region_plot)) {
  region_plot$HiC@name &lt;- &quot;HiChIP&quot;
  region_plot$HiC@dp@pars$fontface.title &lt;- 1
  region_plot$HiC@dp@pars$rotation.title &lt;- 0
}
region_plot$Detected@dp@pars$fontface.title &lt;- 1
region_plot$Detected@dp@pars$rotation.title &lt;- 0
region_plot$Detected@dp@pars$fontsize.group &lt;- fs
region_plot$Detected@dp@pars$cex.group &lt;- 0.8
region_plot$Detected@dp@pars$fontface.group &lt;- 1
region_plot$Detected@dp@pars$fontcolor.group &lt;- &quot;black&quot;
if (&quot;Not-Detected&quot; %in% names(region_plot)) {
  region_plot$`Not-Detected`@dp@pars$fontcolor.group &lt;- &quot;black&quot;
  region_plot$`Not-Detected`@dp@pars$fontface.title &lt;- 1
  region_plot$`Not-Detected`@dp@pars$rotation.title &lt;- 0
  region_plot$`Not-Detected`@dp@pars$fontsize.group &lt;- fs
  region_plot$`Not-Detected`@dp@pars$cex.group &lt;- 0.8
  region_plot$`Not-Detected`@dp@pars$fontface.group &lt;- 1
  region_plot$`Not-Detected`@dp@pars$just.group &lt;- &quot;right&quot;
}
region_plot$AR@dp@pars$rotation.title &lt;- 0
region_plot$FOXA1@dp@pars$rotation.title &lt;- 0
region_plot$GATA3@dp@pars$rotation.title &lt;- 0
region_plot$TFAP2B@dp@pars$rotation.title &lt;- 0
region_plot$H3K27ac@dp@pars$rotation.title &lt;- 0
region_plot$H3K27ac@dp@pars$size &lt;- 5
ax &lt;- region_plot[1:2]
top_panels &lt;- region_plot[names(region_plot) %in% c(&quot;HiC&quot;, &quot;Detected&quot;, &quot;Not-Detected&quot;)]
hl &lt;- HighlightTrack(
  region_plot[c(targets, &quot;H3K27ac&quot;)],
  range = resize(gr, width = 1.2 * width(gr), fix = &#39;center&#39;)
)
hl@dp@pars$fill &lt;- rgb(1, 1, 1, 0)
hl@dp@pars$col &lt;- &quot;blue&quot;
hl@dp@pars$lwd &lt;- 0.8
pdf(here::here(&quot;output&quot;, glue(&quot;{gn}.pdf&quot;)), width = 10, height = 10)
plot_range &lt;- c(gr, subset(all_gr$gene, gene_name == gn)) %&gt;% 
  range(ignore.strand = TRUE) %&gt;% 
  resize(width = 1.2*width(.), fix = &#39;center&#39;)
Gviz::plotTracks(
  c(ax, top_panels, hl),
  from = start(plot_range), to = end(plot_range),
  title.width = 1,
  margin = 40,
  fontface.main = 1
  )
dev.off()</code></pre>
</div>
<div id="tfap2b" class="section level3">
<h3>TFAP2B</h3>
<pre class="r"><code>gn &lt;- &quot;TFAP2B&quot;
gr &lt;- dht_consensus %&gt;% 
  filter(!!!syms(targets), mapped, H3K27ac) %&gt;% 
  filter(
    vapply(gene_name, function(x) gn %in% x, logical(1))
  ) %&gt;% 
  mutate(d = distance(., subset(all_gr$gene, gene_name == gn))) %&gt;% 
  filter(d &lt; 5e5) 
fs &lt;- 14
region_plot &lt;- plotHFGC(
  gr = gr,
  hic = hic %&gt;%
    subsetByOverlaps(gr) %&gt;%
    subsetByOverlaps(
      subset(all_gr$gene, gene_name == gn)
    ) %&gt;% 
    subset(as.integer(bin_size) == 10e3),
  hicsize = 3,
  genes = tm %&gt;%
    mutate(detected = ifelse(gene %in% detected, &quot;Detected&quot;, &quot;Not-Detected&quot;)) %&gt;%
    split(.$detected),
  genecol = c(
    Detected = &quot;forestgreen&quot;, &quot;Not-Detected&quot; = rgb(0.2, 0.2, 0.7, 0.5)
  ),
  coverage = bwfl, 
  linecol = rep(&quot;grey30&quot;, length(bwfl)),
  cytobands = grch37.cytobands,
  zoom = 1.1,
  highlight = rgb(0, 0, 1, 0.5), 
  title.width = 1, 
  col.title = &quot;black&quot;, background.title = &quot;white&quot;,
  showAxis = FALSE,
  rotation.title = 90, fontsize = fs, 
  fontface.title = 1,
  legend = FALSE,
  fontcolor.legend = &quot;black&quot;
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/plot-tfap2b-1.png" alt="All H3K27ac-associated peaks where all four targets were detected, which map to *TFAP2B* and are within 500kb of *TFAP2B*. Only the medium-resolution 10kb interaction bins are shown." width="960" />
<p class="caption">
All H3K27ac-associated peaks where all four targets were detected, which
map to <em>TFAP2B</em> and are within 500kb of <em>TFAP2B</em>. Only the
medium-resolution 10kb interaction bins are shown.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-tfap2b-1">
Past versions of plot-tfap2b-1.png
</button>
</p>
<div id="fig-plot-tfap2b-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/comparison.Rmd/plot-tfap2b-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1088ac1595379dd4f700c6dfad34e638f6d977b/docs/figure/comparison.Rmd/plot-tfap2b-1.png" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/65e26b822f70462eecc2b399e344026d22517a0b/docs/figure/comparison.Rmd/plot-tfap2b-1.png" target="_blank">65e26b8</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>region_plot[[1]]@dp@pars$fontcolor &lt;- &quot;black&quot;
region_plot$Axis@dp@pars$fontcolor &lt;- &quot;black&quot;
region_plot$Axis@dp@pars$col &lt;- &quot;grey30&quot;
if (&quot;HiC&quot; %in% names(region_plot)) {
  region_plot$HiC@name &lt;- &quot;HiChIP&quot;
  region_plot$HiC@dp@pars$fontface.title &lt;- 1
  region_plot$HiC@dp@pars$rotation.title &lt;- 0
}
region_plot$Detected@dp@pars$fontface.title &lt;- 1
region_plot$Detected@dp@pars$rotation.title &lt;- 0
region_plot$Detected@dp@pars$fontsize.group &lt;- fs
region_plot$Detected@dp@pars$cex.group &lt;- 0.8
region_plot$Detected@dp@pars$fontface.group &lt;- 1
region_plot$Detected@dp@pars$fontcolor.group &lt;- &quot;black&quot;
if (&quot;Not-Detected&quot; %in% names(region_plot)) {
  region_plot$`Not-Detected`@dp@pars$fontcolor.group &lt;- &quot;black&quot;
  region_plot$`Not-Detected`@dp@pars$fontface.title &lt;- 1
  region_plot$`Not-Detected`@dp@pars$rotation.title &lt;- 0
  region_plot$`Not-Detected`@dp@pars$fontsize.group &lt;- fs
  region_plot$`Not-Detected`@dp@pars$cex.group &lt;- 0.8
  region_plot$`Not-Detected`@dp@pars$fontface.group &lt;- 1
  region_plot$`Not-Detected`@dp@pars$just.group &lt;- &quot;right&quot;
}
region_plot$AR@dp@pars$rotation.title &lt;- 0
region_plot$FOXA1@dp@pars$rotation.title &lt;- 0
region_plot$GATA3@dp@pars$rotation.title &lt;- 0
region_plot$TFAP2B@dp@pars$rotation.title &lt;- 0
region_plot$H3K27ac@dp@pars$rotation.title &lt;- 0
region_plot$H3K27ac@dp@pars$size &lt;- 5
ax &lt;- region_plot[1:2]
top_panels &lt;- region_plot[names(region_plot) %in% c(&quot;HiC&quot;, &quot;Detected&quot;, &quot;Not-Detected&quot;)]
hl &lt;- HighlightTrack(
  region_plot[c(targets, &quot;H3K27ac&quot;)],
  range = resize(gr, width = 3 * width(gr), fix = &#39;center&#39;) %&gt;% 
    reduce()
)
hl@dp@pars$fill &lt;- rgb(1, 1, 1, 0)
hl@dp@pars$col &lt;- &quot;blue&quot;
hl@dp@pars$lwd &lt;- 0.8
pdf(here::here(&quot;output&quot;, glue(&quot;{gn}.pdf&quot;)), width = 10, height = 10)
plot_range &lt;- c(gr, subset(all_gr$gene, gene_name == gn)) %&gt;% 
  range(ignore.strand = TRUE) %&gt;% 
  resize(width = 1.2*width(.), fix = &#39;center&#39;)
Gviz::plotTracks(
  c(ax, top_panels, hl),
  from = start(plot_range), to = end(plot_range),
  title.width = 1,
  margin = 40,
  fontface.main = 1
  )
dev.off()</code></pre>
</div>
<div id="kynu" class="section level3">
<h3>KYNU</h3>
<pre class="r"><code>gn &lt;- &quot;KYNU&quot;
gr &lt;- dht_consensus %&gt;% 
  filter(!!!syms(targets), mapped, H3K27ac) %&gt;% 
  filter(
    vapply(gene_name, function(x) gn %in% x, logical(1))
  ) %&gt;% 
  mutate(d = distance(., subset(all_gr$gene, gene_name == gn))) 
fs &lt;- 14
region_plot &lt;- plotHFGC(
  gr = gr,
  hic = hic %&gt;%
    subsetByOverlaps(gr) %&gt;%
    subsetByOverlaps(
      subset(all_gr$gene, gene_name == gn)
    ) %&gt;% 
    subset(as.integer(bin_size) &lt;= 40e3),
  hicsize = 3,
  genes = tm %&gt;%
    mutate(detected = ifelse(gene %in% detected, &quot;Detected&quot;, &quot;Not-Detected&quot;)) %&gt;%
    split(.$detected),
  genecol = c(
    Detected = &quot;forestgreen&quot;, &quot;Not-Detected&quot; = rgb(0.2, 0.2, 0.7, 0.5)
  ),
  coverage = bwfl, 
  linecol = rep(&quot;grey30&quot;, length(bwfl)),
  cytobands = grch37.cytobands,
  zoom = 5,
  shift = 3e4,
  highlight = rgb(0, 0, 1, 0.5), 
  title.width = 1, 
  col.title = &quot;black&quot;, background.title = &quot;white&quot;,
  showAxis = FALSE,
  rotation.title = 90, fontsize = fs, 
  fontface.title = 1,
  legend = FALSE,
  fontcolor.legend = &quot;black&quot;
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/plot-kynu-1.png" alt="All H3K27ac-associated peaks where all four targets were detected, and which map to *KYNU*. No HiC interaction bins were found in this region." width="960" />
<p class="caption">
All H3K27ac-associated peaks where all four targets were detected, and
which map to <em>KYNU</em>. No HiC interaction bins were found in this
region.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-kynu-1">
Past versions of plot-kynu-1.png
</button>
</p>
<div id="fig-plot-kynu-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/comparison.Rmd/plot-kynu-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1088ac1595379dd4f700c6dfad34e638f6d977b/docs/figure/comparison.Rmd/plot-kynu-1.png" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/65e26b822f70462eecc2b399e344026d22517a0b/docs/figure/comparison.Rmd/plot-kynu-1.png" target="_blank">65e26b8</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>region_plot[[1]]@dp@pars$fontcolor &lt;- &quot;black&quot;
region_plot$Axis@dp@pars$fontcolor &lt;- &quot;black&quot;
region_plot$Axis@dp@pars$col &lt;- &quot;grey30&quot;
if (&quot;HiC&quot; %in% names(region_plot)) {
  region_plot$HiC@name &lt;- &quot;HiChIP&quot;
  region_plot$HiC@dp@pars$fontface.title &lt;- 1
  region_plot$HiC@dp@pars$rotation.title &lt;- 0
}
region_plot$Detected@dp@pars$fontface.title &lt;- 1
region_plot$Detected@dp@pars$rotation.title &lt;- 0
region_plot$Detected@dp@pars$fontsize.group &lt;- fs
region_plot$Detected@dp@pars$cex.group &lt;- 0.8
region_plot$Detected@dp@pars$fontface.group &lt;- 1
region_plot$Detected@dp@pars$fontcolor.group &lt;- &quot;black&quot;
if (&quot;Not-Detected&quot; %in% names(region_plot)) {
  region_plot$`Not-Detected`@dp@pars$fontcolor.group &lt;- &quot;black&quot;
  region_plot$`Not-Detected`@dp@pars$fontface.title &lt;- 1
  region_plot$`Not-Detected`@dp@pars$rotation.title &lt;- 0
  region_plot$`Not-Detected`@dp@pars$fontsize.group &lt;- fs
  region_plot$`Not-Detected`@dp@pars$cex.group &lt;- 0.8
  region_plot$`Not-Detected`@dp@pars$fontface.group &lt;- 1
  region_plot$`Not-Detected`@dp@pars$just.group &lt;- &quot;right&quot;
}
region_plot$AR@dp@pars$rotation.title &lt;- 0
region_plot$FOXA1@dp@pars$rotation.title &lt;- 0
region_plot$GATA3@dp@pars$rotation.title &lt;- 0
region_plot$TFAP2B@dp@pars$rotation.title &lt;- 0
region_plot$H3K27ac@dp@pars$rotation.title &lt;- 0
region_plot$H3K27ac@dp@pars$size &lt;- 5
ax &lt;- region_plot[1:2]
top_panels &lt;- region_plot[names(region_plot) %in% c(&quot;HiC&quot;, &quot;Detected&quot;, &quot;Not-Detected&quot;)]
hl &lt;- HighlightTrack(
  region_plot[c(targets, &quot;H3K27ac&quot;)],
  range = resize(gr, width = 1.5 * width(gr), fix = &#39;center&#39;)
)
hl@dp@pars$fill &lt;- rgb(1, 1, 1, 0)
hl@dp@pars$col &lt;- &quot;blue&quot;
hl@dp@pars$lwd &lt;- 0.8
pdf(here::here(&quot;output&quot;, glue(&quot;{gn}.pdf&quot;)), width = 10, height = 10)
plot_range &lt;- c(gr, subset(all_gr$gene, gene_name == gn)) %&gt;% 
  range(ignore.strand = TRUE) %&gt;% 
  resize(width = 1.1*width(.), fix = &#39;center&#39;) %&gt;% 
  resize(width = 0.4 * width(.), fix = &#39;start&#39;)
Gviz::plotTracks(
  c(ax, top_panels, hl),
  from = start(plot_range), to = end(plot_range),
  title.width = 1,
  margin = 40,
  fontface.main = 1
  )
dev.off()</code></pre>
</div>
<div id="dusp6" class="section level3">
<h3>DUSP6</h3>
<pre class="r"><code>gn &lt;- &quot;DUSP6&quot;
gr &lt;- dht_consensus %&gt;% 
  filter(!!!syms(targets), mapped, H3K27ac) %&gt;% 
  filter(
    vapply(gene_name, function(x) gn %in% x, logical(1))
  ) %&gt;% 
  mutate(d = distance(., subset(all_gr$gene, gene_name == gn))) %&gt;% 
  filter(d &lt; 1e5)
fs &lt;- 14
region_plot &lt;- plotHFGC(
  gr = gr,
  hic = hic %&gt;%
    subsetByOverlaps(gr) %&gt;%
    subsetByOverlaps(
      subset(all_gr$gene, gene_name == gn)
    ) %&gt;% 
    subset(as.integer(bin_size) &lt;= 5e3),
  hicsize = 3,
  genes = tm %&gt;%
    mutate(detected = ifelse(gene %in% detected, &quot;Detected&quot;, &quot;Not-Detected&quot;)) %&gt;%
    split(.$detected),
  genecol = c(
    Detected = &quot;forestgreen&quot;, &quot;Not-Detected&quot; = rgb(0.2, 0.2, 0.7, 0.5)
  ),
  coverage = bwfl, 
  linecol = rep(&quot;grey30&quot;, length(bwfl)),
  cytobands = grch37.cytobands,
  zoom = 1.1,
  shift = 3e4,
  highlight = rgb(0, 0, 1, 0.5), 
  title.width = 1, 
  col.title = &quot;black&quot;, background.title = &quot;white&quot;,
  showAxis = FALSE,
  rotation.title = 90, fontsize = fs, 
  fontface.title = 1,
  legend = FALSE,
  fontcolor.legend = &quot;black&quot;
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/plot-dusp6-1.png" alt="All H3K27ac-associated peaks where all four targets were detected, which map to *DUSP6* and are within 100kb of the gene. Only the high-resolution 5kb HiC interaction bins are shown." width="960" />
<p class="caption">
All H3K27ac-associated peaks where all four targets were detected, which
map to <em>DUSP6</em> and are within 100kb of the gene. Only the
high-resolution 5kb HiC interaction bins are shown.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-dusp6-1">
Past versions of plot-dusp6-1.png
</button>
</p>
<div id="fig-plot-dusp6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/comparison.Rmd/plot-dusp6-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1088ac1595379dd4f700c6dfad34e638f6d977b/docs/figure/comparison.Rmd/plot-dusp6-1.png" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/65e26b822f70462eecc2b399e344026d22517a0b/docs/figure/comparison.Rmd/plot-dusp6-1.png" target="_blank">65e26b8</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>region_plot[[1]]@dp@pars$fontcolor &lt;- &quot;black&quot;
region_plot$Axis@dp@pars$fontcolor &lt;- &quot;black&quot;
region_plot$Axis@dp@pars$col &lt;- &quot;grey30&quot;
if (&quot;HiC&quot; %in% names(region_plot)) {
  region_plot$HiC@name &lt;- &quot;HiChIP&quot;
  region_plot$HiC@dp@pars$fontface.title &lt;- 1
  region_plot$HiC@dp@pars$rotation.title &lt;- 0
}
region_plot$Detected@dp@pars$fontface.title &lt;- 1
region_plot$Detected@dp@pars$rotation.title &lt;- 0
region_plot$Detected@dp@pars$fontsize.group &lt;- fs
region_plot$Detected@dp@pars$cex.group &lt;- 0.8
region_plot$Detected@dp@pars$fontface.group &lt;- 1
region_plot$Detected@dp@pars$fontcolor.group &lt;- &quot;black&quot;
if (&quot;Not-Detected&quot; %in% names(region_plot)) {
  region_plot$`Not-Detected`@dp@pars$fontcolor.group &lt;- &quot;black&quot;
  region_plot$`Not-Detected`@dp@pars$fontface.title &lt;- 1
  region_plot$`Not-Detected`@dp@pars$rotation.title &lt;- 0
  region_plot$`Not-Detected`@dp@pars$fontsize.group &lt;- fs
  region_plot$`Not-Detected`@dp@pars$cex.group &lt;- 0.8
  region_plot$`Not-Detected`@dp@pars$fontface.group &lt;- 1
  region_plot$`Not-Detected`@dp@pars$just.group &lt;- &quot;right&quot;
}
region_plot$AR@dp@pars$rotation.title &lt;- 0
region_plot$FOXA1@dp@pars$rotation.title &lt;- 0
region_plot$GATA3@dp@pars$rotation.title &lt;- 0
region_plot$TFAP2B@dp@pars$rotation.title &lt;- 0
region_plot$H3K27ac@dp@pars$rotation.title &lt;- 0
region_plot$H3K27ac@dp@pars$size &lt;- 5
ax &lt;- region_plot[1:2]
top_panels &lt;- region_plot[names(region_plot) %in% c(&quot;HiC&quot;, &quot;Detected&quot;, &quot;Not-Detected&quot;)]
hl &lt;- HighlightTrack(
  region_plot[c(targets, &quot;H3K27ac&quot;)],
  range = resize(gr, width = 3 * width(gr), fix = &#39;center&#39;) %&gt;% 
    reduce()
)
hl@dp@pars$fill &lt;- rgb(1, 1, 1, 0)
hl@dp@pars$col &lt;- &quot;blue&quot;
hl@dp@pars$lwd &lt;- 0.8
pdf(here::here(&quot;output&quot;, glue(&quot;{gn}.pdf&quot;)), width = 10, height = 10)
plot_range &lt;- c(gr, subset(all_gr$gene, gene_name == gn)) %&gt;% 
  range(ignore.strand = TRUE) %&gt;% 
  resize(width = 0.8 * width(.), fix = &#39;center&#39;) %&gt;% 
  shift(1e4)
Gviz::plotTracks(
  c(ax, top_panels, hl),
  from = start(plot_range), to = end(plot_range),
  title.width = 1,
  margin = 40,
  fontface.main = 1
  )
dev.off()</code></pre>
</div>
<div id="fgfr4" class="section level3">
<h3>FGFR4</h3>
<pre class="r"><code>gn &lt;- &quot;FGFR4&quot;
gr &lt;- dht_consensus %&gt;% 
  filter(!!!syms(targets), mapped, H3K27ac) %&gt;% 
  filter(
    vapply(gene_name, function(x) gn %in% x, logical(1))
  ) %&gt;% 
  mutate(d = distance(., subset(all_gr$gene, gene_name == gn))) %&gt;% 
  filter(d &lt; 1e5)
fs &lt;- 14
region_plot &lt;- plotHFGC(
  gr = gr,
  hic = hic %&gt;%
    subsetByOverlaps(gr) %&gt;%
    subsetByOverlaps(
      subset(all_gr$gene, gene_name == gn)
    ) %&gt;% 
    subset(as.integer(bin_size) &lt;= 5e3),
  hicsize = 3,
  genes = tm %&gt;%
    mutate(detected = ifelse(gene %in% detected, &quot;Detected&quot;, &quot;Not-Detected&quot;)) %&gt;%
    split(.$detected),
  genecol = c(
    Detected = &quot;forestgreen&quot;, &quot;Not-Detected&quot; = rgb(0.2, 0.2, 0.7, 0.5)
  ),
  coverage = bwfl, 
  linecol = rep(&quot;grey30&quot;, length(bwfl)),
  cytobands = grch37.cytobands,
  zoom = 1.1,
  highlight = rgb(0, 0, 1, 0.5), 
  title.width = 1, 
  col.title = &quot;black&quot;, background.title = &quot;white&quot;,
  showAxis = FALSE,
  rotation.title = 90, fontsize = fs, 
  fontface.title = 1,
  legend = FALSE,
  fontcolor.legend = &quot;black&quot;
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/plot-fgfr4-1.png" alt="All H3K27ac-associated peaks where all four targets were detected, which map to *FGFR4* and are within 100kb of the gene. Only the high-resolution 5kb HiC interaction bins are shown." width="960" />
<p class="caption">
All H3K27ac-associated peaks where all four targets were detected, which
map to <em>FGFR4</em> and are within 100kb of the gene. Only the
high-resolution 5kb HiC interaction bins are shown.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-fgfr4-1">
Past versions of plot-fgfr4-1.png
</button>
</p>
<div id="fig-plot-fgfr4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/comparison.Rmd/plot-fgfr4-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1088ac1595379dd4f700c6dfad34e638f6d977b/docs/figure/comparison.Rmd/plot-fgfr4-1.png" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/65e26b822f70462eecc2b399e344026d22517a0b/docs/figure/comparison.Rmd/plot-fgfr4-1.png" target="_blank">65e26b8</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>region_plot[[1]]@dp@pars$fontcolor &lt;- &quot;black&quot;
region_plot$Axis@dp@pars$fontcolor &lt;- &quot;black&quot;
region_plot$Axis@dp@pars$col &lt;- &quot;grey30&quot;
if (&quot;HiC&quot; %in% names(region_plot)) {
  region_plot$HiC@name &lt;- &quot;HiChIP&quot;
  region_plot$HiC@dp@pars$fontface.title &lt;- 1
  region_plot$HiC@dp@pars$rotation.title &lt;- 0
}
region_plot$Detected@dp@pars$fontface.title &lt;- 1
region_plot$Detected@dp@pars$rotation.title &lt;- 0
region_plot$Detected@dp@pars$fontsize.group &lt;- fs
region_plot$Detected@dp@pars$cex.group &lt;- 0.8
region_plot$Detected@dp@pars$fontface.group &lt;- 1
region_plot$Detected@dp@pars$fontcolor.group &lt;- &quot;black&quot;
if (&quot;Not-Detected&quot; %in% names(region_plot)) {
  region_plot$`Not-Detected`@dp@pars$fontcolor.group &lt;- &quot;black&quot;
  region_plot$`Not-Detected`@dp@pars$fontface.title &lt;- 1
  region_plot$`Not-Detected`@dp@pars$rotation.title &lt;- 0
  region_plot$`Not-Detected`@dp@pars$fontsize.group &lt;- fs
  region_plot$`Not-Detected`@dp@pars$cex.group &lt;- 0.8
  region_plot$`Not-Detected`@dp@pars$fontface.group &lt;- 1
  region_plot$`Not-Detected`@dp@pars$just.group &lt;- &quot;right&quot;
}
region_plot$AR@dp@pars$rotation.title &lt;- 0
region_plot$FOXA1@dp@pars$rotation.title &lt;- 0
region_plot$GATA3@dp@pars$rotation.title &lt;- 0
region_plot$TFAP2B@dp@pars$rotation.title &lt;- 0
region_plot$H3K27ac@dp@pars$rotation.title &lt;- 0
region_plot$H3K27ac@dp@pars$size &lt;- 5
ax &lt;- region_plot[1:2]
top_panels &lt;- region_plot[names(region_plot) %in% c(&quot;HiC&quot;, &quot;Detected&quot;, &quot;Not-Detected&quot;)]
hl &lt;- HighlightTrack(
  region_plot[c(targets, &quot;H3K27ac&quot;)],
  range = resize(gr, width = 2 * width(gr), fix = &#39;center&#39;)
)
hl@dp@pars$fill &lt;- rgb(1, 1, 1, 0)
hl@dp@pars$col &lt;- &quot;blue&quot;
hl@dp@pars$lwd &lt;- 0.8
pdf(here::here(&quot;output&quot;, glue(&quot;{gn}.pdf&quot;)), width = 10, height = 10)
plot_range &lt;- c(gr, subset(all_gr$gene, gene_name == gn)) %&gt;% 
  range(ignore.strand = TRUE) %&gt;% 
  resize(width = 1.2*width(.), fix = &#39;center&#39;)
Gviz::plotTracks(
  c(ax, top_panels, hl),
  from = start(plot_range), to = end(plot_range),
  title.width = 1,
  margin = 40,
  fontface.main = 1
  )
dev.off()</code></pre>
</div>
<div id="pgap3" class="section level3">
<h3>PGAP3</h3>
<pre class="r"><code>gn &lt;- &quot;PGAP3&quot;
gr &lt;- dht_consensus %&gt;% 
  filter(!!!syms(targets), mapped, H3K27ac) %&gt;% 
  filter(
    vapply(gene_name, function(x) gn %in% x, logical(1))
  ) %&gt;% 
  mutate(d = distance(., subset(all_gr$gene, gene_name == gn))) %&gt;% 
  filter(d &lt; 1e5)
fs &lt;- 14
region_plot &lt;- plotHFGC(
  gr = gr,
  hic = hic %&gt;%
    subsetByOverlaps(gr) %&gt;%
    subsetByOverlaps(
      subset(all_gr$gene, gene_name == gn)
    ) %&gt;% 
    subset(as.integer(bin_size) &lt;= 5e3),
  hicsize = 3,
  max = 5e4,
  genes = tm %&gt;%
    mutate(detected = ifelse(gene %in% detected, &quot;Detected&quot;, &quot;Not-Detected&quot;)) %&gt;%
    split(.$detected),
  genecol = c(
    Detected = &quot;forestgreen&quot;, &quot;Not-Detected&quot; = rgb(0.2, 0.2, 0.7, 0.5)
  ),
  coverage = bwfl, 
  linecol = rep(&quot;grey30&quot;, length(bwfl)),
  cytobands = grch37.cytobands,
  zoom = 1.1,
  highlight = rgb(0, 0, 1, 0.5), 
  title.width = 1, 
  col.title = &quot;black&quot;, background.title = &quot;white&quot;,
  showAxis = FALSE,
  rotation.title = 90, fontsize = fs, 
  fontface.title = 1,
  legend = FALSE,
  fontcolor.legend = &quot;black&quot;
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/plot-pgap3-1.png" alt="All H3K27ac-associated peaks where all four targets were detected, which map to *PGAP3* and are within 100kb of the gene. Only the high-resolution 5kb HiC interaction bins are shown." width="960" />
<p class="caption">
All H3K27ac-associated peaks where all four targets were detected, which
map to <em>PGAP3</em> and are within 100kb of the gene. Only the
high-resolution 5kb HiC interaction bins are shown.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-pgap3-1">
Past versions of plot-pgap3-1.png
</button>
</p>
<div id="fig-plot-pgap3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/comparison.Rmd/plot-pgap3-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1088ac1595379dd4f700c6dfad34e638f6d977b/docs/figure/comparison.Rmd/plot-pgap3-1.png" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/65e26b822f70462eecc2b399e344026d22517a0b/docs/figure/comparison.Rmd/plot-pgap3-1.png" target="_blank">65e26b8</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>region_plot[[1]]@dp@pars$fontcolor &lt;- &quot;black&quot;
region_plot$Axis@dp@pars$fontcolor &lt;- &quot;black&quot;
region_plot$Axis@dp@pars$col &lt;- &quot;grey30&quot;
if (&quot;HiC&quot; %in% names(region_plot)) {
  region_plot$HiC@name &lt;- &quot;HiChIP&quot;
  region_plot$HiC@dp@pars$fontface.title &lt;- 1
  region_plot$HiC@dp@pars$rotation.title &lt;- 0
}
region_plot$Detected@dp@pars$fontface.title &lt;- 1
region_plot$Detected@dp@pars$rotation.title &lt;- 0
region_plot$Detected@dp@pars$fontsize.group &lt;- fs
region_plot$Detected@dp@pars$cex.group &lt;- 0.8
region_plot$Detected@dp@pars$fontface.group &lt;- 1
region_plot$Detected@dp@pars$fontcolor.group &lt;- &quot;black&quot;
if (&quot;Not-Detected&quot; %in% names(region_plot)) {
  region_plot$`Not-Detected`@dp@pars$fontcolor.group &lt;- &quot;black&quot;
  region_plot$`Not-Detected`@dp@pars$fontface.title &lt;- 1
  region_plot$`Not-Detected`@dp@pars$rotation.title &lt;- 0
  region_plot$`Not-Detected`@dp@pars$fontsize.group &lt;- fs
  region_plot$`Not-Detected`@dp@pars$cex.group &lt;- 0.8
  region_plot$`Not-Detected`@dp@pars$fontface.group &lt;- 1
  region_plot$`Not-Detected`@dp@pars$just.group &lt;- &quot;right&quot;
}
region_plot$AR@dp@pars$rotation.title &lt;- 0
region_plot$FOXA1@dp@pars$rotation.title &lt;- 0
region_plot$GATA3@dp@pars$rotation.title &lt;- 0
region_plot$TFAP2B@dp@pars$rotation.title &lt;- 0
region_plot$H3K27ac@dp@pars$rotation.title &lt;- 0
region_plot$H3K27ac@dp@pars$size &lt;- 5
ax &lt;- region_plot[1:2]
top_panels &lt;- region_plot[names(region_plot) %in% c(&quot;HiC&quot;, &quot;Detected&quot;, &quot;Not-Detected&quot;)]
hl &lt;- HighlightTrack(
  region_plot[c(targets, &quot;H3K27ac&quot;)],
  range = resize(gr, width = 1.5 * width(gr), fix = &#39;center&#39;)
)
hl@dp@pars$fill &lt;- rgb(1, 1, 1, 0)
hl@dp@pars$col &lt;- &quot;blue&quot;
hl@dp@pars$lwd &lt;- 0.8
pdf(here::here(&quot;output&quot;, glue(&quot;{gn}.pdf&quot;)), width = 10, height = 10)
plot_range &lt;- c(gr, subset(all_gr$gene, gene_name == gn)) %&gt;% 
  range(ignore.strand = TRUE) %&gt;% 
  resize(width = 1.2*width(.), fix = &#39;center&#39;)
Gviz::plotTracks(
  c(ax, top_panels, hl),
  from = start(plot_range), to = end(plot_range),
  title.width = 1,
  margin = 40,
  fontface.main = 1
  )
dev.off()</code></pre>
</div>
</div>
<div id="data-export" class="section level2">
<h2>Data Export</h2>
<div id="sequence-export" class="section level3">
<h3>Sequence Export</h3>
<pre class="r"><code>hg19 &lt;- BSgenome.Hsapiens.UCSC.hg19
sq19 &lt;- seqinfo(hg19)
consenus_hg19 &lt;- dht_consensus
seqinfo(consenus_hg19) &lt;- sq19
features_hg19 &lt;- features
seqinfo(features_hg19) &lt;- sq19

getSeq(
  hg19,
  consenus_hg19 %&gt;% 
    filter(AR, FOXA1, GATA3, !TFAP2B) %&gt;% 
    resize(width = 500, fix = &#39;center&#39;) %&gt;% 
    setNames(as.character(.))
) %&gt;% 
  writeXStringSet(
    here::here(&quot;output&quot;, &quot;ar_foxa1_gata3_no_tfap2b.fa&quot;)
  )
getSeq(
  hg19,
  consenus_hg19 %&gt;% 
    filter(AR, FOXA1, GATA3, TFAP2B) %&gt;% 
    resize(width = 500, fix = &#39;center&#39;) %&gt;% 
    setNames(as.character(.))
) %&gt;% 
  writeXStringSet(
    here::here(&quot;output&quot;, &quot;ar_foxa1_gata3_tfap2b.fa&quot;)
  )
getSeq(
  hg19,
  consenus_hg19 %&gt;% 
    filter(AR, FOXA1, GATA3, TFAP2B, H3K27ac) %&gt;% 
    resize(width = 500, fix = &#39;center&#39;) %&gt;% 
    setNames(as.character(.))
) %&gt;% 
  writeXStringSet(
    here::here(&quot;output&quot;, &quot;all4_h3k27ac.fa&quot;)
  )
getSeq(
  hg19,
  consenus_hg19 %&gt;% 
    filter(AR, FOXA1, GATA3, TFAP2B, !H3K27ac) %&gt;% 
    resize(width = 500, fix = &#39;center&#39;) %&gt;% 
    setNames(as.character(.))
) %&gt;% 
  writeXStringSet(
    here::here(&quot;output&quot;, &quot;all4_no_h3k27ac.fa&quot;)
  )</code></pre>
</div>
<div id="peaks" class="section level3">
<h3>Peaks</h3>
<pre class="r"><code>write_rds(dht_consensus, here::here(&quot;output/dht_consensus.rds&quot;), compress = &quot;gz&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] BSgenome.Hsapiens.UCSC.hg19_1.4.3 BSgenome_1.64.0                  
 [3] Biostrings_2.64.0                 XVector_0.36.0                   
 [5] scico_1.3.0                       cowplot_1.1.1                    
 [7] goseq_1.48.0                      geneLenDataBase_1.32.0           
 [9] BiasedUrn_1.07                    corrplot_0.92                    
[11] readxl_1.4.0                      Gviz_1.40.1                      
[13] effects_4.2-1                     carData_3.0-5                    
[15] nnet_7.3-17                       glue_1.6.2                       
[17] multcomp_1.4-19                   TH.data_1.1-1                    
[19] MASS_7.3-56                       survival_3.2-13                  
[21] mvtnorm_1.1-3                     GenomicInteractions_1.30.0       
[23] InteractionSet_1.24.0             rtracklayer_1.56.0               
[25] UpSetR_1.4.0                      htmltools_0.5.2                  
[27] reactable_0.3.0                   scales_1.2.0                     
[29] pander_0.6.5                      plyranges_1.16.0                 
[31] extraChIPs_1.0.1                  SummarizedExperiment_1.26.1      
[33] Biobase_2.56.0                    MatrixGenerics_1.8.0             
[35] matrixStats_0.62.0                GenomicRanges_1.48.0             
[37] GenomeInfoDb_1.32.2               IRanges_2.30.0                   
[39] S4Vectors_0.34.0                  BiocGenerics_0.42.0              
[41] BiocParallel_1.30.2               magrittr_2.0.3                   
[43] forcats_0.5.1                     stringr_1.4.0                    
[45] dplyr_1.0.9                       purrr_0.3.4                      
[47] readr_2.1.2                       tidyr_1.2.0                      
[49] tibble_3.1.7                      ggplot2_3.3.6                    
[51] tidyverse_1.3.1                   workflowr_1.7.0                  

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3           csaw_1.30.1              bit64_4.0.5             
  [4] knitr_1.39               DelayedArray_0.22.0      data.table_1.14.2       
  [7] rpart_4.1.16             KEGGREST_1.36.0          RCurl_1.98-1.6          
 [10] AnnotationFilter_1.20.0  doParallel_1.0.17        generics_0.1.2          
 [13] GenomicFeatures_1.48.3   callr_3.7.0              EnrichedHeatmap_1.26.0  
 [16] RSQLite_2.2.14           bit_4.0.4                tzdb_0.3.0              
 [19] xml2_1.3.3               lubridate_1.8.0          httpuv_1.6.5            
 [22] assertthat_0.2.1         xfun_0.31                hms_1.1.1               
 [25] jquerylib_0.1.4          evaluate_0.15            promises_1.2.0.1        
 [28] fansi_1.0.3              restfulr_0.0.13          progress_1.2.2          
 [31] dbplyr_2.1.1             igraph_1.3.1             DBI_1.1.2               
 [34] htmlwidgets_1.5.4        ellipsis_0.3.2           crosstalk_1.2.0         
 [37] backports_1.4.1          insight_0.17.1           survey_4.1-1            
 [40] biomaRt_2.52.0           vctrs_0.4.1              here_1.0.1              
 [43] ensembldb_2.20.1         cachem_1.0.6             withr_2.5.0             
 [46] ggforce_0.3.3            vroom_1.5.7              checkmate_2.1.0         
 [49] GenomicAlignments_1.32.0 prettyunits_1.1.1        cluster_2.1.3           
 [52] lazyeval_0.2.2           crayon_1.5.1             labeling_0.4.2          
 [55] edgeR_3.38.1             pkgconfig_2.0.3          tweenr_1.0.2            
 [58] nlme_3.1-157             ProtGenerics_1.28.0      rlang_1.0.2             
 [61] lifecycle_1.0.1          sandwich_3.0-1           filelock_1.0.2          
 [64] BiocFileCache_2.4.0      modelr_0.1.8             dichromat_2.0-0.1       
 [67] cellranger_1.1.0         rprojroot_2.0.3          polyclip_1.10-0         
 [70] reactR_0.4.4             Matrix_1.4-1             boot_1.3-28             
 [73] zoo_1.8-10               reprex_2.0.1             base64enc_0.1-3         
 [76] whisker_0.4              GlobalOptions_0.1.2      processx_3.5.3          
 [79] viridisLite_0.4.0        png_0.1-7                rjson_0.2.21            
 [82] bitops_1.0-7             getPass_0.2-2            blob_1.2.3              
 [85] shape_1.4.6              jpeg_0.1-9               memoise_2.0.1           
 [88] plyr_1.8.7               zlibbioc_1.42.0          compiler_4.2.0          
 [91] scatterpie_0.1.7         BiocIO_1.6.0             RColorBrewer_1.1-3      
 [94] clue_0.3-61              lme4_1.1-29              Rsamtools_2.12.0        
 [97] cli_3.3.0                ps_1.7.0                 htmlTable_2.4.0         
[100] Formula_1.2-4            mgcv_1.8-40              ggside_0.2.0.9990       
[103] tidyselect_1.1.2         stringi_1.7.6            highr_0.9               
[106] mitools_2.4              yaml_2.3.5               locfit_1.5-9.5          
[109] latticeExtra_0.6-29      ggrepel_0.9.1            sass_0.4.1              
[112] VariantAnnotation_1.42.1 tools_4.2.0              parallel_4.2.0          
[115] circlize_0.4.15          rstudioapi_0.13          foreach_1.5.2           
[118] foreign_0.8-82           git2r_0.30.1             metapod_1.4.0           
[121] gridExtra_2.3            farver_2.1.0             digest_0.6.29           
[124] Rcpp_1.0.8.3             broom_0.8.0              later_1.3.0             
[127] httr_1.4.3               AnnotationDbi_1.58.0     biovizBase_1.44.0       
[130] ComplexHeatmap_2.12.0    colorspace_2.0-3         rvest_1.0.2             
[133] XML_3.99-0.9             fs_1.5.2                 splines_4.2.0           
[136] jsonlite_1.8.0           nloptr_2.0.3             ggfun_0.0.6             
[139] R6_2.5.1                 Hmisc_4.7-0              pillar_1.7.0            
[142] fastmap_1.1.0            minqa_1.2.4              codetools_0.2-18        
[145] utf8_1.2.2               lattice_0.20-45          bslib_0.3.1             
[148] curl_4.3.2               GO.db_3.15.0             limma_3.52.1            
[151] rmarkdown_2.14           munsell_0.5.0            GetoptLong_1.0.5        
[154] GenomeInfoDbData_1.2.8   iterators_1.0.14         haven_2.5.0             
[157] gtable_0.3.0            </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
