<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Stephen Pederson" />


<title>Comparison of all ChIP Targets</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/core-js-2.5.3/shim.min.js"></script>
<script src="site_libs/react-17.0.0/react.min.js"></script>
<script src="site_libs/react-17.0.0/react-dom.min.js"></script>
<script src="site_libs/reactwidget-1.0.0/react-tools.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="site_libs/reactable-binding-0.2.3/reactable.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MDA-MB-453: Apocrine Signature</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="comparison.html">Comparison</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Comparison of all ChIP Targets</h1>
<h4 class="author">Stephen Pederson</h4>
<h4 class="date">13 May, 2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-05-13
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>apocrine_signature_mdamb453/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220427code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220427)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220427code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220427)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDRMCRLapocrinesignaturemdamb453tree9288752656ff464f7c730b06b43f0cc66f91bb0ftargetblank9288752a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/tree/9288752656ff464f7c730b06b43f0cc66f91bb0f" target="_blank">9288752</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomDRMCRLapocrinesignaturemdamb453tree9288752656ff464f7c730b06b43f0cc66f91bb0ftargetblank9288752a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/tree/9288752656ff464f7c730b06b43f0cc66f91bb0f" target="_blank">9288752</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/
    Ignored:    data/bigwig/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/comparison.Rmd</code>) and HTML
(<code>docs/comparison.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/9288752656ff464f7c730b06b43f0cc66f91bb0f/analysis/comparison.Rmd" target="_blank">9288752</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-13
</td>
<td>
Added 95% CIs for targets
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1582607b19db6df592097b193ebcd67fd14a8328/analysis/comparison.Rmd" target="_blank">1582607</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-12
</td>
<td>
Started looking at H3K27ac
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/8217e8a21e6dfffb0e5597e720e3baea1a6f2398/analysis/comparison.Rmd" target="_blank">8217e8a</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-10
</td>
<td>
Started explorations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/apocrine_signature_mdamb453/8217e8a21e6dfffb0e5597e720e3baea1a6f2398/docs/comparison.html" target="_blank">8217e8a</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-10
</td>
<td>
Started explorations
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<pre class="r"><code>library(tidyverse)
library(magrittr)
library(extraChIPs)
library(plyranges)
library(pander)
library(scales)
library(reactable)
library(htmltools)
library(UpSetR)
library(rtracklayer)
library(GenomicInteractions)
library(multcomp)
theme_set(theme_bw())</code></pre>
<pre class="r"><code>dht_peaks &lt;- here::here(&quot;data&quot;, &quot;peaks&quot;) %&gt;% 
  list.files(recursive = TRUE, pattern = &quot;oracle&quot;, full.names = TRUE) %&gt;% 
  sapply(read_rds, simplify = FALSE) %&gt;% 
  lapply(function(x) x[[&quot;DHT&quot;]]) %&gt;% 
  lapply(setNames, nm = c()) %&gt;% 
  setNames(str_extract_all(names(.), &quot;AR|FOXA1|GATA3|TFAP2B&quot;))
dht_consensus &lt;- dht_peaks %&gt;%
  lapply(granges) %&gt;% 
  GRangesList() %&gt;% 
  unlist() %&gt;% 
  reduce() %&gt;% 
  mutate(
    AR = overlapsAny(., dht_peaks$AR),
    FOXA1 = overlapsAny(., dht_peaks$FOXA1),
    GATA3 = overlapsAny(., dht_peaks$GATA3),
    TFAP2B = overlapsAny(., dht_peaks$TFAP2B),
  )
targets &lt;- names(dht_peaks)
sq &lt;- seqinfo(dht_consensus)</code></pre>
</div>
<div id="relationship-between-binding-regions" class="section level2">
<h2>Relationship Between Binding Regions</h2>
<p>Oracle peaks from the DHT-treated samples in each ChIP target were
obtained previously using the GRAVI workflow. AR and GATA3 peaks were
derived from the same samples/passages, whilst FOXA1 and TFAP2B ChIP-Seq
experiments were performed separately.</p>
<pre class="r"><code>cp &lt;- htmltools::tags$em(
  &quot;Summary of all oracle peaks from DHT-treated samples. FOXA1 clearly showed the most binding activity.&quot;
)
tbl &lt;- dht_peaks %&gt;% 
  lapply(
    function(x) {
      tibble(
        n = length(x),
        w = median(width(x)),
        kb = sum(width(x)) / 1e3
      )
    }
  ) %&gt;% 
  lapply(list) %&gt;% 
  as_tibble() %&gt;% 
  pivot_longer(cols = everything(), names_to = &quot;target&quot;) %&gt;% 
  unnest(everything()) %&gt;% 
  reactable(
    filterable = FALSE, searchable = FALSE,
    columns = list(
      target = colDef(name = &quot;ChIP Target&quot;),
      n = colDef(name = &quot;Total Peaks&quot;),
      w = colDef(name = &quot;Median Width&quot;),
      kb = colDef(name = &quot;Total Width (kb)&quot;, format = colFormat(digits = 1))
    ),
    defaultColDef = colDef(
      format = colFormat(separators = TRUE, digits = 0)
    )
  )
div(class = &quot;table&quot;,
  div(class = &quot;table-header&quot;,
      div(class = &quot;caption&quot;, cp),
      tbl
  )
)</code></pre>
<div class="table">
<div class="table-header">
<div class="caption">
<em>Summary of all oracle peaks from DHT-treated samples. FOXA1 clearly showed the most binding activity.</em>
</div>
<div id="htmlwidget-9929d7c7769fe1c64f8c" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-9929d7c7769fe1c64f8c">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"target":["AR","FOXA1","GATA3","TFAP2B"],"n":[19406,92751,29666,21008],"w":[329,374,297,310],"kb":[7156.449,41675.084,10059.128,7661.389]},"columns":[{"accessor":"target","name":"ChIP Target","type":"character","format":{"cell":{"digits":0,"separators":true},"aggregated":{"digits":0,"separators":true}}},{"accessor":"n","name":"Total Peaks","type":"numeric","format":{"cell":{"digits":0,"separators":true},"aggregated":{"digits":0,"separators":true}}},{"accessor":"w","name":"Median Width","type":"numeric","format":{"cell":{"digits":0,"separators":true},"aggregated":{"digits":0,"separators":true}}},{"accessor":"kb","name":"Total Width (kb)","type":"numeric","format":{"cell":{"digits":1},"aggregated":{"digits":1}}}],"defaultPageSize":10,"paginationType":"numbers","showPageInfo":true,"minRows":1,"dataKey":"dde42d4b31286b0fbee5937965e1c5a2","key":"dde42d4b31286b0fbee5937965e1c5a2"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>A set of target-agnostic set of binding regions was then defined as
the union of all DHT-treat peaks across all targets.</p>
<pre class="r"><code>dht_consensus %&gt;% 
  as_tibble() %&gt;% 
  pivot_longer(cols = all_of(targets), names_to = &quot;target&quot;, values_to = &quot;bound&quot;) %&gt;% 
  dplyr::filter(bound) %&gt;% 
  split(.$target) %&gt;% 
  lapply(pull, &quot;range&quot;) %&gt;% 
  fromList() %&gt;% 
  upset(
    sets = rev(targets), keep.order = TRUE, 
    order.by = &quot;freq&quot;, 
    set_size.show = TRUE, set_size.scale_max = nrow(.)
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/upset-dht-consensus-1.png" alt="Using the union of all binding regions, those which overlapped an oracle peak from each ChIP target are shown" width="672" />
<p class="caption">
Using the union of all binding regions, those which overlapped an oracle
peak from each ChIP target are shown
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-upset-dht-consensus-1">
Past versions of upset-dht-consensus-1.png
</button>
</p>
<div id="fig-upset-dht-consensus-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/8217e8a21e6dfffb0e5597e720e3baea1a6f2398/docs/figure/comparison.Rmd/upset-dht-consensus-1.png" target="_blank">8217e8a</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="h3k27ac-signal" class="section level2">
<h2>H3K27ac Signal</h2>
<p>Given the hypothesis that the co-ocurrence of all 4 ChIP targets
should be associated with increased transcriptional or regulatory
activity, the set of all DHT peaks were then compared to promoters and
enahncers derived H2K27ac binding, as detected in the existing GATA3/AR
dataset produced by Leil Hosseinzadeh. Any H3K27ac peak detected in this
dataset was classified either as a promoter or enhancer, and thus these
features can be considered as the complete set of regions with
detectable H3K27ac signal from this experiment.</p>
<pre class="r"><code>features &lt;- here::here(&quot;data&quot;, &quot;h3k27ac&quot;) %&gt;% 
  list.files(full.names = TRUE, pattern = &quot;bed$&quot;) %&gt;% 
  sapply(import.bed, seqinfo = sq) %&gt;% 
  lapply(granges) %&gt;% 
  setNames(basename(names(.))) %&gt;% 
  setNames(str_remove_all(names(.), &quot;s.bed&quot;)) %&gt;% 
  GRangesList() %&gt;% 
  unlist() %&gt;% 
  names_to_column(&quot;feature&quot;) %&gt;% 
  sort()</code></pre>
<pre class="r"><code>grp_h3k27ac &lt;- dht_consensus %&gt;% 
  mutate(h3k27ac = overlapsAny(., features)) %&gt;% 
  as_tibble() %&gt;% 
  dplyr::select(range, all_of(targets), h3k27ac) %&gt;% 
  pivot_longer(cols = all_of(targets), names_to = &quot;targets&quot;) %&gt;% 
  dplyr::filter(value) %&gt;% 
  group_by(range, h3k27ac) %&gt;% 
  summarise(targets = paste(targets, collapse = &quot;+&quot;), .groups = &quot;drop&quot;)
glm_h3k27ac &lt;- glm(h3k27ac ~ 0 + targets, family = &quot;binomial&quot;, data = grp_h3k27ac)</code></pre>
<pre class="r"><code>inv.logit &lt;- binomial()$linkinv
glm_h3k27ac %&gt;% 
  glht() %&gt;% 
  confint() %&gt;% 
  .[[&quot;confint&quot;]] %&gt;% 
  as_tibble(rownames = &quot;targets&quot;) %&gt;% 
  mutate(
    n_targets = str_count(targets, &quot;_&quot;) + 1,
    targets = fct_inorder(targets) %&gt;% 
      fct_relabel(str_remove_all, pattern = &quot;targets&quot;),
    p = inv.logit(Estimate),
    TFAP2B = ifelse(str_detect(targets, &quot;TFAP2B&quot;), &quot;TFAP2B&quot;, &quot;No TFAP2B&quot;),
    across(all_of(c(&quot;upr&quot;, &quot;lwr&quot;)), inv.logit)
  ) %&gt;% 
  ggplot(aes(p, targets)) + 
  geom_point() +
  geom_errorbarh(aes(xmin = lwr, xmax = upr), height = 0.4, colour = &quot;grey20&quot;) +
  facet_grid(TFAP2B ~ ., scales = &quot;free_y&quot;) +
  labs(x = &quot;Probability of Feature Overlap&quot;, y = &quot;Detected Targets&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/comparison.Rmd/confint-glm-h3k27ac-1.png" alt="Family-wise 95% Confidence Intervals for the probability of overlapping an H3K27ac-derived feature, based on the combinations of detected ChIP targets." width="672" />
<p class="caption">
Family-wise 95% Confidence Intervals for the probability of overlapping
an H3K27ac-derived feature, based on the combinations of detected ChIP
targets.
</p>
</div>
<div id="comparison-with-h3k27ac-hichip-data" class="section level3">
<h3>Comparison With H3K27ac HiChIP Data</h3>
<p>To perform this step, externally-sourced data defining promoters
&amp; enhancers (H3K27ac ChIP-Seq), and H3K27ac-HiChIP obtained from <a
href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE157381">SRA</a>
and <a
href="https://github.com/DRMCRL/MDA-MB-453-H3K27Ac-HiChIP">analysed
previously</a>, were included. The H3K27ac-derived features were
obtained from the same passages/experiments as GATA3 and AR. Conversely,
the HiChIP data was obtained from a public dataset, not produced with
the DRMCRL. HiChIP interactions were obtained using only the Vehicle
controls from an Abemaciclib Vs. Vehicle experiment.</p>
<pre class="r"><code>fl &lt;- here::here(&quot;data&quot;, &quot;hichip&quot;) %&gt;% 
  list.files(full.names = TRUE, pattern = &quot;gi.+rds&quot;) 
hic &lt;- GInteractions()
for (f in fl) {
  hic &lt;- c(hic, read_rds(f))
}
hic &lt;- sort(hic)</code></pre>
<p>Before proceeding, the comparability of the H3K27ac-HiChIP and
H3K27ac-derived features was checked. 93% of HiChIP long-range
interactions overlapped a promoter or enhancer derived from H3K27ac
ChIP-seq. Conversely, 86% or ChIP-Seq features mapped to a long-range
interaction.</p>
</div>
</div>
<div id="mapping-to-genes" class="section level2">
<h2>Mapping To Genes</h2>
<pre class="r"><code>all_gr &lt;- here::here(&quot;data&quot;, &quot;annotations&quot;, &quot;all_gr.rds&quot;) %&gt;% 
  read_rds() 
rnaseq &lt;- here::here(&quot;data&quot;, &quot;rnaseq&quot;, &quot;dge.rds&quot;) %&gt;% 
  read_rds()
counts &lt;- here::here(&quot;data&quot;, &quot;rnaseq&quot;, &quot;counts.out.gz&quot;) %&gt;% 
  read_tsv(skip = 1) %&gt;% 
  dplyr::select(Geneid, ends_with(&quot;bam&quot;)) 
detected &lt;- counts %&gt;%
  pivot_longer(
    cols = ends_with(&quot;bam&quot;), names_to = &quot;sample&quot;, values_to = &quot;counts&quot;
  ) %&gt;% 
  mutate(detected = counts &gt; 0) %&gt;% 
  group_by(Geneid) %&gt;% 
  summarise(detected = mean(detected) &gt; 0.25, .groups = &quot;drop&quot;) %&gt;%
  dplyr::filter(detected) %&gt;% 
  pull(&quot;Geneid&quot;)</code></pre>
<p>In order to more accurately assign genes to actively transcribed
genes, the RNA-Seq dataset generated in 2013 studying DHT Vs. Vehicle in
MDA-MB-453 cells was used. All 21,328 genes with &gt;1 read in at least
1/4 of the samples was considered to be detected, and peaks were only
mapped to detected genes.</p>
<pre class="r"><code>dht_consensus &lt;- mapByFeature(
  dht_consensus, 
  genes = subset(all_gr$gene, gene_id %in% detected),
  prom = subset(features, feature == &quot;promoter&quot;),
  enh = subset(features, feature == &quot;enhancer&quot;),
  gi = hic
)
all_targets &lt;- dht_consensus %&gt;% 
  as_tibble() %&gt;% 
  dplyr::filter(if_all(targets)) %&gt;% 
  unnest(everything()) %&gt;% 
  distinct(gene_id) %&gt;% 
  pull(&quot;gene_id&quot;)</code></pre>
<p>92% of detected genes had one or more peaks mapped to them.</p>
<p>Looking specifically at the peaks for which all four targets directly
overlapped, 10,743 of the 21,328 detected genes were mapped to at least
one directly overlapping binding region.</p>
</div>
<div id="apocrine-enrichment" class="section level2">
<h2>Apocrine Enrichment</h2>
<pre class="r"><code>hgnc &lt;- read_csv(
  here::here(&quot;data&quot;, &quot;external&quot;, &quot;hgnc-symbol-check.csv&quot;), skip = 1
) %&gt;% 
  dplyr::select(Gene = Input, gene_name = `Approved symbol`)
apo_ranks &lt;- here::here(&quot;data&quot;, &quot;external&quot;, &quot;ApoGenes.txt&quot;) %&gt;% 
  read_tsv() %&gt;% 
  left_join(hgnc) %&gt;% 
  left_join(
    as_tibble(all_gr$gene) %&gt;% 
      dplyr::select(gene_id, gene_name),
    by = &quot;gene_name&quot;
  ) %&gt;% 
  dplyr::select(starts_with(&quot;gene_&quot;), ends_with(&quot;rank&quot;)) %&gt;% 
  dplyr::filter(!is.na(gene_id))</code></pre>
<p>The Apocrine genes and ranks from <a
href="https://www.nature.com/articles/1208561#Sec10">Farmer <em>et
al</em></a> were loaded, updating gene names using the latest release
from <a
href="https://www.genenames.org/tools/multi-symbol-checker/">HGNC</a>.
10,669 of the original genes were able to be mapped to gene identifiers
matching Gencode release 33.</p>
<div id="apocrine-ranks" class="section level3">
<h3>Apocrine Ranks</h3>
<ul>
<li>Need to cut by H3K27Ac</li>
<li>Check UpSet plot by promoter/enhancer/detected genes</li>
</ul>
<p>For the purposes of simple exploration, the provided list was sorted
by <code>apo_rank</code> and the proportion of genes mapped to at least
one set of overlapping peaks was compared to these ranks.</p>
<pre class="r"><code>apo_ranks %&gt;%
  arrange(apo_rank) %&gt;% 
  mutate(
    all = gene_id %in% all_targets,
    n = cumsum(all), 
    p = n / apo_rank
  ) %&gt;% 
  ggplot(aes(apo_rank, p)) + 
  geom_line() + 
  coord_cartesian(xlim= c(0, 1000), ylim = c(0.6, 1))</code></pre>
<p><img src="figure/comparison.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/8217e8a21e6dfffb0e5597e720e3baea1a6f2398/docs/figure/comparison.Rmd/unnamed-chunk-2-1.png" target="_blank">8217e8a</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="ab-ranks" class="section level3">
<h3>AB Ranks</h3>
<p>The same approach was applied to AB ranks</p>
<pre class="r"><code>apo_ranks %&gt;%
  dplyr::filter(!is.na(ABrank)) %&gt;% 
  arrange(ABrank) %&gt;% 
  mutate(
    ABrank = seq_along(ABrank),
    all = gene_id %in% all_targets,
    n = cumsum(all), 
    p = n / ABrank
  ) %&gt;% 
  ggplot(aes(ABrank, p)) + 
  geom_line() </code></pre>
<p><img src="figure/comparison.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/8217e8a21e6dfffb0e5597e720e3baea1a6f2398/docs/figure/comparison.Rmd/unnamed-chunk-3-1.png" target="_blank">8217e8a</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="al-ranks" class="section level3">
<h3>AL Ranks</h3>
<p>The same approach was applied to AL ranks</p>
<pre class="r"><code>apo_ranks %&gt;%
  dplyr::filter(!is.na(ALrank)) %&gt;% 
  arrange(ALrank) %&gt;% 
  mutate(
    ALrank = seq_along(ALrank),
    all = gene_id %in% all_targets,
    n = cumsum(all), 
    p = n / ALrank
  ) %&gt;% 
  ggplot(aes(ALrank, p)) + 
  geom_line() </code></pre>
<p><img src="figure/comparison.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/8217e8a21e6dfffb0e5597e720e3baea1a6f2398/docs/figure/comparison.Rmd/unnamed-chunk-4-1.png" target="_blank">8217e8a</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] multcomp_1.4-19             TH.data_1.1-1              
 [3] MASS_7.3-56                 survival_3.2-13            
 [5] mvtnorm_1.1-3               GenomicInteractions_1.30.0 
 [7] InteractionSet_1.24.0       rtracklayer_1.56.0         
 [9] UpSetR_1.4.0                htmltools_0.5.2            
[11] reactable_0.2.3             scales_1.2.0               
[13] pander_0.6.5                plyranges_1.16.0           
[15] extraChIPs_1.0.0            SummarizedExperiment_1.26.1
[17] Biobase_2.56.0              MatrixGenerics_1.8.0       
[19] matrixStats_0.62.0          GenomicRanges_1.48.0       
[21] GenomeInfoDb_1.32.1         IRanges_2.30.0             
[23] S4Vectors_0.34.0            BiocGenerics_0.42.0        
[25] BiocParallel_1.30.0         magrittr_2.0.3             
[27] forcats_0.5.1               stringr_1.4.0              
[29] dplyr_1.0.9                 purrr_0.3.4                
[31] readr_2.1.2                 tidyr_1.2.0                
[33] tibble_3.1.7                ggplot2_3.3.6              
[35] tidyverse_1.3.1             workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] utf8_1.2.2               tidyselect_1.1.2         RSQLite_2.2.14          
  [4] AnnotationDbi_1.58.0     htmlwidgets_1.5.4        grid_4.2.0              
  [7] scatterpie_0.1.7         munsell_0.5.0            codetools_0.2-18        
 [10] withr_2.5.0              colorspace_2.0-3         filelock_1.0.2          
 [13] highr_0.9                knitr_1.39               rstudioapi_0.13         
 [16] ggside_0.2.0.9990        labeling_0.4.2           git2r_0.30.1            
 [19] GenomeInfoDbData_1.2.8   polyclip_1.10-0          farver_2.1.0            
 [22] bit64_4.0.5              rprojroot_2.0.3          vctrs_0.4.1             
 [25] generics_0.1.2           xfun_0.31                biovizBase_1.44.0       
 [28] csaw_1.30.0              BiocFileCache_2.4.0      R6_2.5.1                
 [31] doParallel_1.0.17        clue_0.3-60              locfit_1.5-9.5          
 [34] AnnotationFilter_1.20.0  bitops_1.0-7             cachem_1.0.6            
 [37] DelayedArray_0.22.0      assertthat_0.2.1         vroom_1.5.7             
 [40] promises_1.2.0.1         BiocIO_1.6.0             nnet_7.3-17             
 [43] gtable_0.3.0             processx_3.5.3           sandwich_3.0-1          
 [46] ensembldb_2.20.1         rlang_1.0.2              GlobalOptions_0.1.2     
 [49] splines_4.2.0            lazyeval_0.2.2           dichromat_2.0-0.1       
 [52] broom_0.8.0              checkmate_2.1.0          yaml_2.3.5              
 [55] modelr_0.1.8             crosstalk_1.2.0          GenomicFeatures_1.48.0  
 [58] backports_1.4.1          httpuv_1.6.5             Hmisc_4.7-0             
 [61] EnrichedHeatmap_1.26.0   tools_4.2.0              ellipsis_0.3.2          
 [64] jquerylib_0.1.4          RColorBrewer_1.1-3       plyr_1.8.7              
 [67] Rcpp_1.0.8.3             base64enc_0.1-3          progress_1.2.2          
 [70] zlibbioc_1.42.0          RCurl_1.98-1.6           ps_1.7.0                
 [73] prettyunits_1.1.1        rpart_4.1.16             GetoptLong_1.0.5        
 [76] zoo_1.8-10               reactR_0.4.4             ggrepel_0.9.1           
 [79] haven_2.5.0              cluster_2.1.3            here_1.0.1              
 [82] fs_1.5.2                 data.table_1.14.2        circlize_0.4.15         
 [85] reprex_2.0.1             whisker_0.4              ProtGenerics_1.28.0     
 [88] hms_1.1.1                evaluate_0.15            XML_3.99-0.9            
 [91] jpeg_0.1-9               readxl_1.4.0             gridExtra_2.3           
 [94] shape_1.4.6              compiler_4.2.0           biomaRt_2.52.0          
 [97] crayon_1.5.1             ggfun_0.0.6              later_1.3.0             
[100] tzdb_0.3.0               Formula_1.2-4            lubridate_1.8.0         
[103] DBI_1.1.2                tweenr_1.0.2             dbplyr_2.1.1            
[106] ComplexHeatmap_2.12.0    rappdirs_0.3.3           Matrix_1.4-1            
[109] cli_3.3.0                parallel_4.2.0           Gviz_1.40.1             
[112] metapod_1.4.0            igraph_1.3.1             pkgconfig_2.0.3         
[115] getPass_0.2-2            GenomicAlignments_1.32.0 foreign_0.8-82          
[118] xml2_1.3.3               foreach_1.5.2            bslib_0.3.1             
[121] XVector_0.36.0           rvest_1.0.2              VariantAnnotation_1.42.0
[124] callr_3.7.0              digest_0.6.29            Biostrings_2.64.0       
[127] rmarkdown_2.14           cellranger_1.1.0         htmlTable_2.4.0         
[130] edgeR_3.38.0             restfulr_0.0.13          curl_4.3.2              
[133] Rsamtools_2.12.0         rjson_0.2.21             lifecycle_1.0.1         
[136] jsonlite_1.8.0           limma_3.52.0             BSgenome_1.64.0         
[139] fansi_1.0.3              pillar_1.7.0             lattice_0.20-45         
[142] KEGGREST_1.36.0          fastmap_1.1.0            httr_1.4.3              
[145] glue_1.6.2               png_0.1-7                iterators_1.0.14        
[148] bit_4.0.4                ggforce_0.3.3            stringi_1.7.6           
[151] sass_0.4.1               blob_1.2.3               latticeExtra_0.6-29     
[154] memoise_2.0.1           </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
