<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Steve Pederson" />


<title>Motif Analysis</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/core-js-2.5.3/shim.min.js"></script>
<script src="site_libs/react-17.0.0/react.min.js"></script>
<script src="site_libs/react-17.0.0/react-dom.min.js"></script>
<script src="site_libs/reactwidget-1.0.0/react-tools.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="site_libs/reactable-binding-0.3.0/reactable.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MDA-MB-453: Apocrine Signature</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="comparison.html">Comparison</a>
</li>
<li>
  <a href="motif_analysis.html">Motif Analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Motif Analysis</h1>
<h4 class="author">Steve Pederson</h4>
<h4 class="date">23 June, 2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-06-23
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>apocrine_signature_mdamb453/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220427code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220427)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220427code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220427)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDRMCRLapocrinesignaturemdamb453treee7f4eded530569ef88b0e9a75af9f09d536b101etargetblanke7f4edea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/tree/e7f4eded530569ef88b0e9a75af9f09d536b101e" target="_blank">e7f4ede</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomDRMCRLapocrinesignaturemdamb453treee7f4eded530569ef88b0e9a75af9f09d536b101etargetblanke7f4edea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/tree/e7f4eded530569ef88b0e9a75af9f09d536b101e" target="_blank">e7f4ede</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/bigwig/

Unstaged changes:
    Modified:   analysis/motif_analysis.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/motif_analysis.Rmd</code>) and
HTML (<code>docs/motif_analysis.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/analysis/motif_analysis.Rmd" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
<td>
Finished checking motif distances
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/apocrine_signature_mdamb453/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/motif_analysis.html" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
<td>
Finished checking motif distances
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/b882cb8ccca0c67325724df3bf74f151f8b8c2f1/analysis/motif_analysis.Rmd" target="_blank">b882cb8</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-07
</td>
<td>
Ranges are no longer centred
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/apocrine_signature_mdamb453/b882cb8ccca0c67325724df3bf74f151f8b8c2f1/docs/motif_analysis.html" target="_blank">b882cb8</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-07
</td>
<td>
Ranges are no longer centred
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/0c127407bf36122be17950abac95eb8d8ddfc9b8/analysis/motif_analysis.Rmd" target="_blank">0c12740</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-06
</td>
<td>
Finished regression analysis for motifs
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/apocrine_signature_mdamb453/0c127407bf36122be17950abac95eb8d8ddfc9b8/docs/motif_analysis.html" target="_blank">0c12740</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-06
</td>
<td>
Finished regression analysis for motifs
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/apocrine_signature_mdamb453/e893c08d9d68b4c4565ed064d68ad0e8dba3e0e6/docs/motif_analysis.html" target="_blank">e893c08</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-06
</td>
<td>
Build page
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/8dde309446d2e562a920f7fa7c87e825ff5cb64d/analysis/motif_analysis.Rmd" target="_blank">8dde309</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-06
</td>
<td>
Updated motif analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/apocrine_signature_mdamb453/8dde309446d2e562a920f7fa7c87e825ff5cb64d/docs/motif_analysis.html" target="_blank">8dde309</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-06
</td>
<td>
Updated motif analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/e1088ac1595379dd4f700c6dfad34e638f6d977b/analysis/motif_analysis.Rmd" target="_blank">e1088ac</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-03
</td>
<td>
revised comparison
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/16615bf0cb46725ffb951b12c670812d5dadaee9/analysis/motif_analysis.Rmd" target="_blank">16615bf</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-28
</td>
<td>
Finished motif analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/apocrine_signature_mdamb453/16615bf0cb46725ffb951b12c670812d5dadaee9/docs/motif_analysis.html" target="_blank">16615bf</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-28
</td>
<td>
Finished motif analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1bf46f725877100a2a8b4a7da76611f3be92a29f/analysis/motif_analysis.Rmd" target="_blank">1bf46f7</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-27
</td>
<td>
Added motifs distance plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/apocrine_signature_mdamb453/1bf46f725877100a2a8b4a7da76611f3be92a29f/docs/motif_analysis.html" target="_blank">1bf46f7</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-27
</td>
<td>
Added motifs distance plots
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/7cfa8e707d2a52f7e5c4b23c23c59c4fc7873cff/analysis/motif_analysis.Rmd" target="_blank">7cfa8e7</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-26
</td>
<td>
Added TOMTOM
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/apocrine_signature_mdamb453/7cfa8e707d2a52f7e5c4b23c23c59c4fc7873cff/docs/motif_analysis.html" target="_blank">7cfa8e7</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-26
</td>
<td>
Added TOMTOM
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/88360904ce5089346cccb1f5c4c86a011fe28cd4/analysis/motif_analysis.Rmd" target="_blank">8836090</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-26
</td>
<td>
Added motif centrality
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DRMCRL/apocrine_signature_mdamb453/88360904ce5089346cccb1f5c4c86a011fe28cd4/docs/motif_analysis.html" target="_blank">8836090</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-26
</td>
<td>
Added motif centrality
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>This step takes the peaks defined previously and</p>
<ol style="list-style-type: decimal">
<li>Assesses motif enrichment for all ChIP targets within the peaks</li>
<li>Detect additional binding motifs</li>
</ol>
<p>As AR is only found bound in the presence of DHT, all peaks are taken
using DHT-treatment as the reference condition.</p>
<pre class="r"><code>library(memes)
library(universalmotif)
library(tidyverse)
library(plyranges)
library(extraChIPs)
library(rtracklayer)
library(magrittr)
library(BSgenome.Hsapiens.UCSC.hg19)
library(scales)
library(pander)
library(glue)
library(cowplot)
library(parallel)
library(gt)
library(parallel)
library(readxl)
library(UpSetR)
library(reactable)
library(htmltools)
library(multcomp)</code></pre>
<pre class="r"><code>options(meme_bin = &quot;/opt/meme/bin/&quot;)
check_meme_install()
theme_set(theme_bw())
panderOptions(&quot;big.mark&quot;, &quot;,&quot;)
n_cores &lt;- 12</code></pre>
<pre class="r"><code>getBestMatch &lt;- function(x, subject, min.score = &quot;80%&quot;, rc = TRUE, ...) {
  
  pwm &lt;- NULL
  if (is.matrix(x)) pwm &lt;- x
  if (is(x, &quot;universalmotif&quot;)) pwm &lt;- x@motif
  stopifnot(!is.null(pwm))
  stopifnot(is(subject, &quot;XStringViews&quot;))
  
  matches &lt;- matchPWM(
    pwm, subject, min.score = min.score, with.score = TRUE, ...
  )
  df &lt;- as.data.frame(matches)
  df$score &lt;- mcols(matches)$score
  df$strand &lt;- &quot;+&quot;
  
  if (rc) {
    rc_matches &lt;-  matchPWM(
      reverseComplement(pwm), subject ,
      min.score = min.score, with.score = TRUE, 
      ...
    )
    rc_df &lt;- as.data.frame(rc_matches)
    rc_df$score &lt;- mcols(rc_matches)$score
    rc_df$strand &lt;- &quot;-&quot;
    df &lt;- bind_rows(df, rc_df)
  }
  
  grp_cols &lt;- c(&quot;start&quot;, &quot;end&quot;, &quot;width&quot;, &quot;seq&quot;, &quot;score&quot;)
  grp_df &lt;- group_by(df, !!!syms(grp_cols))
  summ_df &lt;- summarise(
    grp_df, strand = paste(strand, collapse = &quot; &quot;), .groups=  &quot;drop&quot;
  )
  summ_df$strand &lt;- str_replace_all(summ_df$strand, &quot;. .&quot;, &quot;*&quot;)
  summ_df$strand &lt;- factor(summ_df$strand, levels = c(&quot;+&quot;, &quot;-&quot;, &quot;*&quot;))
  summ_df$view &lt;- findInterval(summ_df$start, start(subject))
  summ_df &lt;- arrange(summ_df, view, desc(score))
  
  best_df &lt;- distinct(summ_df, view, .keep_all = TRUE)
  best_df &lt;- mutate(
    best_df,
    start_in_view = start - start(subject)[view] + 1,
    end_in_view = start_in_view + width
  )
  ## This will only be added if the views have names
  best_df$view_name &lt;- names(subject[best_df$view])
  
  best_df
  
}
bar_style &lt;- function(width = 1, fill = &quot;#e6e6e6&quot;, height = &quot;75%&quot;, align = c(&quot;left&quot;, &quot;right&quot;), color = NULL) {
  align &lt;- match.arg(align)
  if (align == &quot;left&quot;) {
    position &lt;- paste0(width * 100, &quot;%&quot;)
    image &lt;- sprintf(&quot;linear-gradient(90deg, %1$s %2$s, transparent %2$s)&quot;, fill, position)
  } else {
    position &lt;- paste0(100 - width * 100, &quot;%&quot;)
    image &lt;- sprintf(&quot;linear-gradient(90deg, transparent %1$s, %2$s %1$s)&quot;, position, fill)
  }
  list(
    backgroundImage = image,
    backgroundSize = paste(&quot;100%&quot;, height),
    backgroundRepeat = &quot;no-repeat&quot;,
    backgroundPosition = &quot;center&quot;,
    color = color
  )
}</code></pre>
<pre class="r"><code>hg19 &lt;- BSgenome.Hsapiens.UCSC.hg19
sq19 &lt;- seqinfo(hg19)</code></pre>
<pre class="r"><code>all_gr &lt;- here::here(&quot;data&quot;, &quot;annotations&quot;, &quot;all_gr.rds&quot;) %&gt;%
  read_rds() %&gt;% 
  lapply(
    function(x) {
      seqinfo(x) &lt;- sq19
      x
    }
  )
counts &lt;- here::here(&quot;data&quot;, &quot;rnaseq&quot;, &quot;counts.out.gz&quot;) %&gt;%
  read_tsv(skip = 1) %&gt;%
  dplyr::select(Geneid, ends_with(&quot;bam&quot;)) %&gt;%
  dplyr::rename(gene_id = Geneid)
detected &lt;- all_gr$gene %&gt;%
  as_tibble() %&gt;%
  distinct(gene_id, gene_name) %&gt;%
  left_join(counts) %&gt;%
  pivot_longer(
    cols = ends_with(&quot;bam&quot;), names_to = &quot;sample&quot;, values_to = &quot;counts&quot;
  ) %&gt;%
  mutate(detected = counts &gt; 0) %&gt;%
  group_by(gene_id, gene_name) %&gt;%
  summarise(detected = mean(detected) &gt; 0.25, .groups = &quot;drop&quot;) %&gt;%
  dplyr::filter(detected) %&gt;%
  dplyr::select(starts_with(&quot;gene&quot;))</code></pre>
<p>Pre-existing RNA-seq data was again loaded and the set of 21,173
detected genes was defined as those with <span
class="math inline">\(\geq 1\)</span> aligned read in at least 25% of
samples. This dataset represented the DHT response in MDA-MB-453 cells,
however, given the existence of both Vehicle and DHT conditions,
represents a good representative measure of the polyadenylated
components of the transcriptome.</p>
<pre class="r"><code>features &lt;- here::here(&quot;data&quot;, &quot;h3k27ac&quot;) %&gt;%
  list.files(full.names = TRUE, pattern = &quot;bed$&quot;) %&gt;%
  sapply(import.bed, seqinfo = sq19) %&gt;%
  lapply(granges) %&gt;%
  setNames(basename(names(.))) %&gt;%
  setNames(str_remove_all(names(.), &quot;s.bed&quot;)) %&gt;%
  GRangesList() %&gt;%
  unlist() %&gt;%
  names_to_column(&quot;feature&quot;) %&gt;%
  sort()</code></pre>
<p>The set of H3K27ac-defined features was again used as a marker of
regulatory activity, with <em>16,544</em> and <em>15,286</em> regions
defined as enhancers or promoters respectively.</p>
<pre class="r"><code>uniprot_to_ensg &lt;- here::here(&quot;data&quot;, &quot;external&quot;, &quot;uniprot_to_ensg.txt.gz&quot;) %&gt;% 
  read_tsv(col_names = c(&quot;uniprot_id&quot;, &quot;protein&quot;, &quot;gene_id&quot;)) %&gt;% 
  mutate(
    gene_id = lapply(gene_id, str_split, pattern = &quot;; &quot;) %&gt;% 
      lapply(unlist)
  ) %&gt;% 
  unnest(gene_id) %&gt;% 
  mutate(gene_id = str_replace_all(gene_id, &quot;(ENSG[0-9]+)\\..*&quot;, &quot;\\1&quot;)) %&gt;% 
  left_join(
    as_tibble(all_gr$gene) %&gt;% 
      dplyr::select(gene_id, gene_name)
  ) %&gt;% 
  dplyr::filter(!is.na(gene_id)) %&gt;% 
  dplyr::select(-protein)
rime_detected &lt;- here::here(&quot;data&quot;, &quot;external&quot;, &quot;MDA-MB-453_RIME_processed.xlsx&quot;) %&gt;% 
  read_excel() %&gt;% 
  dplyr::select(-starts_with(&quot;AVG&quot;), -ends_with(&quot;Coverage2&quot;), -contains(&quot;delta&quot;)) %&gt;% 
  pivot_longer(cols = contains(&quot;_&quot;)) %&gt;% 
  separate(name, c(&quot;sample&quot;, &quot;measure&quot;), sep = &quot;_&quot;, extra = &quot;merge&quot;) %&gt;% 
  mutate(
    IP = str_remove(sample, &quot;[0-9]$&quot;),
    replicate = str_extract(sample, &quot;[0-9]$&quot;),
    measure = paste(measure, IP, sep = &quot;_&quot;)
  ) %&gt;% 
  pivot_wider(
    names_from = &quot;measure&quot;, values_from = &quot;value&quot;, 
    id_cols = c(&quot;Accession&quot;, &quot;Description&quot;, &quot;Protein&quot;, &quot;replicate&quot;)
  ) %&gt;% 
  mutate(
    logfc_cov = log2(Coverage_AR + 1) - log2(Coverage_IgG + 1),
    logfc_peptides = log2(Unique_peptides_AR + 1) - log2(Unique_peptides_IgG + 1)
  ) %&gt;% 
  dplyr::filter(logfc_cov &gt; 0) %&gt;% 
  group_by(Accession, Description, Protein) %&gt;% 
  summarise(
    n_detected = dplyr::n(),
    mn_logfc_coverage = mean(logfc_cov),
    mn_coverage_AR = mean(Coverage_AR),
    mn_coverage_IgG = mean(Coverage_IgG),
    mn_peptides_AR = mean(Unique_peptides_AR),
    mn_peptides_IgG = mean(Unique_peptides_IgG),
    .groups = &quot;drop&quot;
  ) %&gt;% 
  dplyr::filter(n_detected &gt; 1) %&gt;% 
  arrange(desc(mn_logfc_coverage)) %&gt;% 
  left_join(uniprot_to_ensg, by = c(&quot;Accession&quot; = &quot;uniprot_id&quot;)) %&gt;% 
  dplyr::select(
    Accession, Description, Protein, starts_with(&quot;gene&quot;), everything()
  ) %&gt;% 
  dplyr::filter(!is.na(gene_name))</code></pre>
<p>AR-RIME data for MDA-MB-453 cells was also used, with protein to
transcript ID mappings performed using the mappings obtained using the
following <code>bash</code> command</p>
<pre><code>curl https://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/idmapping/by_organism/HUMAN_9606_idmapping_selected.tab.gz |\
  zcat |\
  cut -f1-2,19 |\
  gzip -c &gt; data/external/uniprot_to_ensg.txt.gz</code></pre>
<p>For RIME data, the criteria for detection within a sample was simply
defined as the coverage being greater when using AR as the IP-target,
than when using IgG as the IP-target. Under this approach, 241 proteins
were considered as detected in <span
class="math inline">\(&gt;2\)</span> of 3 samples and were retained for
inclusion with motif analysis. When further comparing to the RNA-Seq
data only <em>KRT9</em> was found in the RIME data without supporting
RNA evidence. Any proteins with no gene_name associated (i.e. only an
Ensembl Gene ID) were also excluded.</p>
<pre class="r"><code>meme_db &lt;- here::here(&quot;data&quot;, &quot;external&quot;, &quot;JASPAR2022_CORE_Hsapiens.meme&quot;) %&gt;%
  read_meme() %&gt;%
  to_df() %&gt;%
  as_tibble() %&gt;%
  mutate(
    across(everything(), vctrs::vec_proxy),
    altname = str_remove(altname, &quot;MA[0-9]+\\.[0-9]*\\.&quot;),
    organism = &quot;Homo sapiens&quot;
  )
meme_altname &lt;- setNames(meme_db$altname, meme_db$name)
meme_db_detected &lt;- meme_db %&gt;%
  mutate(
    gene_name = str_split(altname, pattern = &quot;::&quot;, n = 2)
  ) %&gt;%
  dplyr::filter(
    vapply(
      gene_name,
      function(x) all(x %in% detected$gene_name),
      logical(1)
    )
  )
meme_info &lt;- meme_db_detected %&gt;% 
  mutate(motif_length = vapply(motif, function(x){ncol(x@motif)}, integer(1))) %&gt;% 
  dplyr::select(name, altname, icscore, motif_length) </code></pre>
<p>The motif database was initially defined as all 727 CORE motifs found
in <em>Homo sapiens</em>, as contained in <a
href="https://jaspar2022.genereg.net/">JASPAR2022</a>. In order to
minimise irrelevant results, only the 465 motifs associated with a
detected gene, using presence within the RNA-Seq dataset as evidence of
detection.</p>
<pre class="r"><code>dht_consensus &lt;- read_rds(
  here::here(&quot;output/dht_consensus.rds&quot;)
)
seqinfo(dht_consensus) &lt;- sq19
dht_peaks &lt;- here::here(&quot;data&quot;, &quot;peaks&quot;) %&gt;%
  list.files(recursive = TRUE, pattern = &quot;oracle&quot;, full.names = TRUE) %&gt;%
  sapply(read_rds, simplify = FALSE) %&gt;%
  lapply(function(x) x[[&quot;DHT&quot;]]) %&gt;%
  lapply(setNames, nm = c()) %&gt;%
  setNames(str_extract_all(names(.), &quot;AR|FOXA1|GATA3|TFAP2B&quot;)) %&gt;% 
  lapply(
    function(x) {
      seqinfo(x) &lt;- sq19
      subsetByOverlaps(x, dht_consensus)
    }
  )
targets &lt;- names(dht_peaks)</code></pre>
</div>
<div id="analysis-of-motif-enrichment-ame" class="section level1">
<h1>Analysis of Motif Enrichment (AME)</h1>
<pre class="r"><code>grl_by_targets &lt;- dht_consensus %&gt;% 
  as_tibble() %&gt;% 
  pivot_longer(
    all_of(targets), names_to = &quot;target&quot;, values_to = &quot;detected&quot;
  ) %&gt;% 
  dplyr::filter(detected) %&gt;% 
  group_by(range, H3K27ac) %&gt;% 
  summarise(
    targets = paste(sort(target), collapse = &quot; &quot;),
    .groups = &quot;drop&quot;
  ) %&gt;% 
  colToRanges(var = &quot;range&quot;, seqinfo = sq19) %&gt;% 
  sort() %&gt;% 
  splitAsList(.$targets)
dht_consensus$targets &lt;- bestOverlap(dht_consensus, grl_by_targets)</code></pre>
<p>Union ranges were defined by grouping by the combination of detected
AR, FOXA1, GATA3 and TFAP2B binding, then sequences obtained
corresponding to each union range.</p>
<pre class="r"><code>grl_by_targets %&gt;% 
  lapply(
    function(x) {
      tibble(
        `N Sites` = length(x), 
        `% H3K27ac` = mean(x$H3K27ac),
        `Median Width` = floor(median(width(x)))
      )
    }
  ) %&gt;% 
  bind_rows(.id = &quot;Targets&quot;) %&gt;% 
  mutate(
    `N Targets` = str_count(Targets, &quot; &quot;) + 1,
    `% H3K27ac` = percent(`% H3K27ac`, 0.1)
  ) %&gt;% 
  dplyr::select(contains(&quot;Targets&quot;), everything()) %&gt;% 
  arrange(`N Targets`, Targets) %&gt;% 
  pander(
    justify = &quot;lrrrr&quot;,
    caption = &quot;Summary of union ranges and which combinations of the targets were detected.&quot;
  )</code></pre>
<table>
<caption>Summary of union ranges and which combinations of the targets
were detected.</caption>
<colgroup>
<col width="32%" />
<col width="16%" />
<col width="13%" />
<col width="16%" />
<col width="20%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Targets</th>
<th align="right">N Targets</th>
<th align="right">N Sites</th>
<th align="right">% H3K27ac</th>
<th align="right">Median Width</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AR</td>
<td align="right">1</td>
<td align="right">344</td>
<td align="right">10.5%</td>
<td align="right">223</td>
</tr>
<tr class="even">
<td align="left">FOXA1</td>
<td align="right">1</td>
<td align="right">57,755</td>
<td align="right">9.8%</td>
<td align="right">291</td>
</tr>
<tr class="odd">
<td align="left">GATA3</td>
<td align="right">1</td>
<td align="right">4,019</td>
<td align="right">16.0%</td>
<td align="right">216</td>
</tr>
<tr class="even">
<td align="left">TFAP2B</td>
<td align="right">1</td>
<td align="right">6,412</td>
<td align="right">74.7%</td>
<td align="right">278</td>
</tr>
<tr class="odd">
<td align="left">AR FOXA1</td>
<td align="right">2</td>
<td align="right">3,580</td>
<td align="right">11.6%</td>
<td align="right">454</td>
</tr>
<tr class="even">
<td align="left">AR GATA3</td>
<td align="right">2</td>
<td align="right">157</td>
<td align="right">12.7%</td>
<td align="right">281</td>
</tr>
<tr class="odd">
<td align="left">AR TFAP2B</td>
<td align="right">2</td>
<td align="right">134</td>
<td align="right">49.3%</td>
<td align="right">374</td>
</tr>
<tr class="even">
<td align="left">FOXA1 GATA3</td>
<td align="right">2</td>
<td align="right">9,906</td>
<td align="right">11.9%</td>
<td align="right">430</td>
</tr>
<tr class="odd">
<td align="left">FOXA1 TFAP2B</td>
<td align="right">2</td>
<td align="right">3,967</td>
<td align="right">54.0%</td>
<td align="right">454</td>
</tr>
<tr class="even">
<td align="left">GATA3 TFAP2B</td>
<td align="right">2</td>
<td align="right">411</td>
<td align="right">53.3%</td>
<td align="right">324</td>
</tr>
<tr class="odd">
<td align="left">AR FOXA1 GATA3</td>
<td align="right">3</td>
<td align="right">6,553</td>
<td align="right">15.3%</td>
<td align="right">620</td>
</tr>
<tr class="even">
<td align="left">AR FOXA1 TFAP2B</td>
<td align="right">3</td>
<td align="right">1,800</td>
<td align="right">47.8%</td>
<td align="right">585</td>
</tr>
<tr class="odd">
<td align="left">AR GATA3 TFAP2B</td>
<td align="right">3</td>
<td align="right">63</td>
<td align="right">47.6%</td>
<td align="right">354</td>
</tr>
<tr class="even">
<td align="left">FOXA1 GATA3 TFAP2B</td>
<td align="right">3</td>
<td align="right">1,434</td>
<td align="right">38.4%</td>
<td align="right">525</td>
</tr>
<tr class="odd">
<td align="left">AR FOXA1 GATA3 TFAP2B</td>
<td align="right">4</td>
<td align="right">6,321</td>
<td align="right">42.3%</td>
<td align="right">746</td>
</tr>
</tbody>
</table>
<pre class="r"><code>seq_by_targets &lt;- grl_by_targets %&gt;% 
  lapply(function(x) split(x, x$H3K27ac)) %&gt;% 
  lapply(GRangesList) %&gt;% 
  lapply(get_sequence, genome = hg19)

## All bound sites
rds_ame_by_targets &lt;- here::here(&quot;output&quot;, &quot;ame_by_targets.rds&quot;)
if (!file.exists(rds_ame_by_targets)) {
  ame_by_targets &lt;- seq_by_targets %&gt;% 
    lapply(unlist) %&gt;% 
    mclapply(
      runAme, 
      database = to_list(meme_db_detected), 
      mc.cores = min(n_cores, length(.))
    )
  nulls &lt;- ame_by_targets %&gt;% 
    vapply(is, logical(1), &quot;data.frame&quot;) %&gt;% 
    not() %&gt;% 
    which() %&gt;% 
    names()
  if (length(nulls) &gt; 0) {
    ame_by_targets[nulls] &lt;- seq_by_targets[nulls] %&gt;% 
      lapply(unlist) %&gt;% 
      lapply(
        runAme, 
        database = to_list(meme_db_detected)
      )
  }
  write_rds(ame_by_targets, rds_ame_by_targets, compress = &quot;gz&quot;)
} else {
  ame_by_targets &lt;- read_rds(rds_ame_by_targets)
}

## Only sites marked as active by H3K27ac
rds_ame_by_targets_active &lt;- here::here(&quot;output&quot;, &quot;ame_by_targets_active.rds&quot;)
if (!file.exists(rds_ame_by_targets_active)) {
  ame_by_targets_active &lt;- seq_by_targets %&gt;% 
    lapply(function(x) x[[&quot;TRUE&quot;]]) %&gt;% 
    mclapply(
      runAme, 
      database = to_list(meme_db_detected), 
      mc.cores = min(n_cores, length(.))
    )
  nulls &lt;- ame_by_targets_active %&gt;% 
    vapply(is, logical(1), &quot;data.frame&quot;) %&gt;% 
    not() %&gt;% 
    which() %&gt;% 
    names()
  if (length(nulls) &gt; 0) {
    ame_by_targets_active[nulls] &lt;- seq_by_targets[nulls] %&gt;% 
      lapply(function(x) x[[&quot;TRUE&quot;]]) %&gt;% 
      lapply(
        runAme, 
        database = to_list(meme_db_detected)
      )
  }
  write_rds(ame_by_targets_active, rds_ame_by_targets_active, compress = &quot;gz&quot;)
} else {
  ame_by_targets_active &lt;- read_rds(rds_ame_by_targets_active)
}

## Compare the active and inactive sites for each set of targets
rds_ame_active_vs_inactive &lt;- here::here(&quot;output&quot;, &quot;ame_active_vs_inactive.rds&quot;)
if (!file.exists(rds_ame_active_vs_inactive)) {
  ame_active_vs_inactive &lt;- seq_by_targets %&gt;% 
    mclapply(
      function(x) {
        runAme(x[[&quot;TRUE&quot;]], x[[&quot;FALSE&quot;]], database = to_list(meme_db_detected))
      },
      mc.cores = min(n_cores, length(.))
    )
  nulls &lt;- ame_active_vs_inactive %&gt;% 
    vapply(is, logical(1), &quot;data.frame&quot;) %&gt;% 
    not() %&gt;% 
    which() %&gt;% 
    names()
  if (length(nulls) &gt; 0) {
    ame_active_vs_inactive[nulls] &lt;- seq_by_targets[nulls] %&gt;% 
      lapply(
        function(x) {
          runAme(x[[&quot;TRUE&quot;]], x[[&quot;FALSE&quot;]], database = to_list(meme_db_detected))
        }
      )
  }
  write_rds(ame_active_vs_inactive, rds_ame_active_vs_inactive, compress = &quot;gz&quot;)
} else {
  ame_active_vs_inactive &lt;- read_rds(rds_ame_active_vs_inactive)
}

## Compare the inactive and active sites for each set of targets
rds_ame_inactive_vs_active &lt;- here::here(&quot;output&quot;, &quot;ame_inactive_vs_active.rds&quot;)
if (!file.exists(rds_ame_inactive_vs_active)) {
  ame_inactive_vs_active &lt;- seq_by_targets %&gt;% 
    mclapply(
      function(x) {
        runAme(x[[&quot;FALSE&quot;]], x[[&quot;TRUE&quot;]], database = to_list(meme_db_detected))
      },
      mc.cores = min(n_cores, length(.))
    )
  nulls &lt;- ame_inactive_vs_active %&gt;% 
    vapply(is, logical(1), &quot;data.frame&quot;) %&gt;% 
    not() %&gt;% 
    which() %&gt;% 
    names()
  if (length(nulls) &gt; 0) {
    ame_inactive_vs_active[nulls] &lt;- seq_by_targets[nulls] %&gt;% 
      lapply(
        function(x) {
          runAme(x[[&quot;FALSE&quot;]], x[[&quot;TRUE&quot;]], database = to_list(meme_db_detected))
        }
      )
  }
  write_rds(ame_inactive_vs_active, rds_ame_inactive_vs_active, compress = &quot;gz&quot;)
} else {
  ame_inactive_vs_active &lt;- read_rds(rds_ame_inactive_vs_active)
}</code></pre>
<p><a href="https://meme-suite.org/meme/doc/ame.html?man_type=web">AME
from the meme-suite</a> was then run on each sequence to detect the
presence of motifs associated with each target within each set of
sequences. Under this approach the optimal score is estimated for
considering a PWM match to be a true positive. Matches above this value
are classed as true positives, whilst those below are considered as
false positives. Significance is determined using using Fisher’s Exact
Test against the set of control sequences, which are either shuffled
input sequences, or a discrete set provided. Adjusted p-values are
determined within each motif based on resampling, with the e-values
representing a Bonferroni adjustment of the within-motif adjusted
p-values.</p>
<p>Motif enrichment inevitably produces hits to highly-related motifs
For unsupervised enrichment, the following strategy was used to aid
interpretation</p>
<ol style="list-style-type: decimal">
<li>Find enriched motifs</li>
<li>Calculate similarity of enriched motifs as a correlation matrix
<ul>
<li>Reduce correlations &lt; <span class="math inline">\(\rho\)</span>
to zero, commonly using <span class="math inline">\(\rho=\)</span>
0.8.</li>
<li>Converted to a distance matrix for clustering</li>
</ul></li>
<li>Cluster the motifs using <code>hclust</code> and a height of
0.9</li>
<li>Report the results for all motifs within a cluster as a single
result</li>
</ol>
<div id="enrichment-of-chip-target-motifs" class="section level2">
<h2>Enrichment Of ChIP Target Motifs</h2>
<pre class="r"><code>plot_grid(
  plot_grid(
    plotlist = lapply(
      meme_db_detected %&gt;% 
        dplyr::filter(altname %in% targets) %&gt;% 
        arrange(altname) %&gt;% 
        dplyr::filter(altname != &quot;TFAP2B&quot;) %&gt;% 
        to_list() %&gt;% 
        lapply(
          function(x){
            x@name &lt;- paste0(
              x@name, &quot; (&quot;, x@altname, &quot;)&quot;
            )
            x
          }
        ),
      function(x) {
        view_motifs(x) +
          ggtitle(x@name) +
          theme(plot.title = element_text(hjust = 0.5, size = 11))
      }
    ),
    ncol = 1
  ),
  plot_grid(
    meme_db_detected %&gt;% 
      dplyr::filter(altname == &quot;TFAP2B&quot;) %&gt;% 
      to_list() %&gt;% 
      lapply(
          function(x){
            x@name &lt;- paste0(
              x@name, &quot; (&quot;, x@altname, &quot;)&quot;
            )
            x
          }
      ) %&gt;% 
      view_motifs() +
      theme(plot.title = element_text(hjust = 0.5, size = 9))
  ),
  ncol = 2
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/motif_analysis.Rmd/plot-motifs-1.png" alt="All motifs matching the set of ChIP targets. AR, FOXA1 and GATA3 are not aligned with each other, whilst all TFAP2B motifs are shown using the best alignment to each other." width="960" />
<p class="caption">
All motifs matching the set of ChIP targets. AR, FOXA1 and GATA3 are not
aligned with each other, whilst all TFAP2B motifs are shown using the
best alignment to each other.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-motifs-1">
Past versions of plot-motifs-1.png
</button>
</p>
<div id="fig-plot-motifs-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/8dde309446d2e562a920f7fa7c87e825ff5cb64d/docs/figure/motif_analysis.Rmd/plot-motifs-1.png" target="_blank">8dde309</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/7cfa8e707d2a52f7e5c4b23c23c59c4fc7873cff/docs/figure/motif_analysis.Rmd/plot-motifs-1.png" target="_blank">7cfa8e7</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="all-sequences" class="section level3">
<h3>All Sequences</h3>
<p>AME was used to assess enrichment of motifs within the sequences
derived from each combination of ChIP targets. Enrichment was considered
relative to <em>shuffled sequences</em> as the control set, with AME
only returning matches considered to be statistically significant.</p>
<pre class="r"><code>ame_by_targets %&gt;% 
  bind_rows(.id = &quot;targets&quot;) %&gt;% 
  dplyr::filter(motif_alt_id %in% globalenv()$targets) %&gt;% 
  mutate(
    motif_id = as.factor(motif_id),
    targets = as.factor(targets),
    TFAP2B = ifelse(
      str_detect(targets, &quot;TFAP2B&quot;), &quot;TFAP2B&quot;, &quot;No TFAP2B&quot;
    )
  ) %&gt;% 
  plot_ame_heatmap(
    id = motif_id,
    value = tp_percent,
    group = fct_rev(targets)
  ) +
  geom_label(
    aes(label = percent(tp_percent/100, 0.1)),
    size = 3.5,
    show.legend = FALSE
  ) +
  facet_grid(TFAP2B ~ motif_alt_id, scales = &quot;free&quot;, space = &quot;free&quot;) +
  scale_x_discrete(expand = expansion(c(0, 0))) +
  scale_y_discrete(expand = expansion(c(0, 0))) +
  labs(
    x = &quot;Detected Motif&quot;,
    y = &quot;Detected ChIP Targets&quot;,
    fill = expression(widehat(&quot;TP%&quot;))
  ) +
  theme(
    axis.text.x = element_text(angle = 0, hjust = 0.5, size = 10),
    axis.text.y = element_text(angle = 0, hjust = 1, size = 10),
    panel.grid = element_blank()
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/motif_analysis.Rmd/ame-heatmap-1.png" alt="Estimated percentage of true positive matches to all binding motifs associated with each ChIP target within all sequences. Only motifs considered to be enriched are shown, such that missing values are able to be confidently assumed as not enriched for the motif. All provide supporting evidence for direct binding of each target to it's motif." width="960" />
<p class="caption">
Estimated percentage of true positive matches to all binding motifs
associated with each ChIP target within all sequences. Only motifs
considered to be enriched are shown, such that missing values are able
to be confidently assumed as not enriched for the motif. All provide
supporting evidence for direct binding of each target to it’s motif.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-ame-heatmap-1">
Past versions of ame-heatmap-1.png
</button>
</p>
<div id="fig-ame-heatmap-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/motif_analysis.Rmd/ame-heatmap-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/8dde309446d2e562a920f7fa7c87e825ff5cb64d/docs/figure/motif_analysis.Rmd/ame-heatmap-1.png" target="_blank">8dde309</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/88360904ce5089346cccb1f5c4c86a011fe28cd4/docs/figure/motif_analysis.Rmd/ame-heatmap-1.png" target="_blank">8836090</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="h3k27ac-associated-sequences" class="section level3">
<h3>H3K27ac-Associated Sequences</h3>
<p>Restricting the test for motif enrichment to all combinations of ChIP
targets which overlap an active H3K27ac peak, similar patterns were
observed. Of possible note is reduced enrichment for TFAP2<span
class="math inline">\(\beta\)</span> motifs amongst the sites where
TFAP2<span class="math inline">\(\beta\)</span> was not observed.</p>
<p>The set of H3K27-associated peaks where AR, GATA3 and TFAP2<span
class="math inline">\(\beta\)</span> were found in combination were
notably enriched for one specific TFAP2<span
class="math inline">\(\beta\)</span> motif (MA0812) to the exclusion of
others suggesting that FOXA1 independent formation of the regulatory
complex may have a distinct mechanism. In conjunction with the increased
occurrence of this motif, this set of peaks also contained the lowest
enrichment for AR motifs amongst the AR-associated peaks.</p>
<p>Another clear association between the presence of FOXA1 and possible
increased enrichment for the MA0811.1 TFAP2<span
class="math inline">\(\beta\)</span> motif may also be of interest.</p>
<pre class="r"><code>ame_by_targets_active %&gt;% 
  bind_rows(.id = &quot;targets&quot;) %&gt;% 
  dplyr::filter(motif_alt_id %in% globalenv()$targets) %&gt;% 
  mutate(
    motif_id = as.factor(motif_id),
    targets = as.factor(targets),
    TFAP2B = ifelse(
      str_detect(targets, &quot;TFAP2B&quot;), &quot;TFAP2B&quot;, &quot;No TFAP2B&quot;
    )
  ) %&gt;% 
  plot_ame_heatmap(
    id = motif_id,
    value = tp_percent,
    group = fct_rev(targets)
  ) +
  geom_label(
    aes(label = percent(tp_percent/100, 0.1)),
    size = 3.5,
    show.legend = FALSE
  ) +
  facet_grid(TFAP2B ~ motif_alt_id, scales = &quot;free&quot;, space = &quot;free&quot;) +
  scale_x_discrete(expand = expansion(c(0, 0))) +
  scale_y_discrete(expand = expansion(c(0, 0))) +
  labs(
    x = &quot;Detected Motif&quot;,
    y = &quot;Detected ChIP Targets&quot;,
    fill = expression(widehat(&quot;TP%&quot;))
  ) +
  theme(
    axis.text.x = element_text(angle = 0, hjust = 0.5, size = 10),
    axis.text.y = element_text(angle = 0, hjust = 1, size = 10),
    panel.grid = element_blank()
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/motif_analysis.Rmd/ame-heatmap-active-1.png" alt="Estimated percentage of true positive matches to all binding motifs associated with each ChIP target, restricting sequences to **those marked by overlapping H3K27ac signal**. Only motif considered to be enriched are shown, such that missing values are able to be confidently assumed as not enriched for the motif. All provide supporting evidence for direct binding of each target to it's motif. Interestingly FOXA1 in combination with TFAP2B appears to favour the motif MA0811.1, whilst AR and GATA3 in combination with TFAP2B appears to favour MA0812.1" width="960" />
<p class="caption">
Estimated percentage of true positive matches to all binding motifs
associated with each ChIP target, restricting sequences to <strong>those
marked by overlapping H3K27ac signal</strong>. Only motif considered to
be enriched are shown, such that missing values are able to be
confidently assumed as not enriched for the motif. All provide
supporting evidence for direct binding of each target to it’s motif.
Interestingly FOXA1 in combination with TFAP2B appears to favour the
motif MA0811.1, whilst AR and GATA3 in combination with TFAP2B appears
to favour MA0812.1
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-ame-heatmap-active-1">
Past versions of ame-heatmap-active-1.png
</button>
</p>
<div id="fig-ame-heatmap-active-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/motif_analysis.Rmd/ame-heatmap-active-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/8dde309446d2e562a920f7fa7c87e825ff5cb64d/docs/figure/motif_analysis.Rmd/ame-heatmap-active-1.png" target="_blank">8dde309</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="enrichment-of-other-motifs" class="section level2">
<h2>Enrichment of Other Motifs</h2>
<p>Motifs associated with any detected transcription factor besides the
ChIP targets were also assessed for enrichment. Motifs which matched
GATA[0-9], Forkhead (FOX[A-Z][0-9]) or TFAP2[A-Z] were omitted due to
the high-level of motif redundancy and these being likely to represent
the ChIP target, instead of an additional protein. Given that an
extremely large number of motifs were found, only those found in &gt;50%
of sites in at least one group of ChIP targets were retained</p>
<div id="all-sequences-1" class="section level3">
<h3>All Sequences</h3>
<p>The set of motifs associated with detectable genes (using RNA) was
tested for enrichment against all sequences from all combinations of
ChIP target binding, using shuffled sequences as controls. Given that
376 motifs were considered to be enriched in at least one set of peaks,
the plot below only shows those motifs which were found in &gt;50% of
sequences in one or more sets of peaks.</p>
<p>The <a href="https://jaspar.genereg.net/matrix/MA0671.1/">NFIX
binding motif</a> was strongly associated with most sets of peaks which
were not bound by TFAP2<span class="math inline">\(\beta\)</span>,
particularly with FOXA1 in the absence of TFAP2<span
class="math inline">\(\beta\)</span>. However, a very strong association
with all four bound targets was also observed. Given the short (6nt)
nature of this motif, this may be artefactual.</p>
<p>The TRPS1 motif was also found to be strongly enriched where GATA3
was bound, however, this motif bears a very strong resemblance to the
GATA3 motif.</p>
<pre class="r"><code>meme_db %&gt;% 
  dplyr::filter(str_detect(altname, &quot;GATA3|TRPS1&quot;)) %&gt;% 
  to_list() %&gt;%
  lapply(function(x) {x@name &lt;- glue(&quot;{x@name}\n({x@altname})&quot;); x}) %&gt;% 
  view_motifs(names.pos = &quot;right&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/motif_analysis.Rmd/motif-gata3-trps1-1.png" alt="Comparison between GATA3 and TRPS1 motifs" width="768" />
<p class="caption">
Comparison between GATA3 and TRPS1 motifs
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-motif-gata3-trps1-1">
Past versions of motif-gata3-trps1-1.png
</button>
</p>
<div id="fig-motif-gata3-trps1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/b882cb8ccca0c67325724df3bf74f151f8b8c2f1/docs/figure/motif_analysis.Rmd/motif-gata3-trps1-1.png" target="_blank">b882cb8</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/8dde309446d2e562a920f7fa7c87e825ff5cb64d/docs/figure/motif_analysis.Rmd/motif-gata3-trps1-1.png" target="_blank">8dde309</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>x &lt;- ame_by_targets %&gt;% 
  bind_rows(.id = &quot;targets&quot;) %&gt;% 
  dplyr::filter(
    !motif_alt_id %in% targets,
    !str_detect(motif_alt_id, &quot;(GATA[0-9]|FOX[A-Z][0-9]|TFAP2[A-Z])&quot;),
    tp_percent &gt; 50
  ) %&gt;% 
  distinct(motif_id) %&gt;% 
  pull(&quot;motif_id&quot;)
cormat &lt;- meme_db %&gt;% 
  dplyr::filter(name %in% x) %&gt;% 
  to_list() %&gt;% 
  compare_motifs()
cormat[cormat &lt; 0.9] &lt;- 0
cormat[is.na(cormat)] &lt;- 0
cl &lt;- as.dist(1 - cormat) %&gt;% 
  hclust() %&gt;% 
  cutree(h= 0.9)
ame_by_targets %&gt;% 
  bind_rows(.id = &quot;targets&quot;) %&gt;% 
  dplyr::filter(motif_id %in% names(cl)) %&gt;% 
  mutate(
    cluster = cl[motif_id],
    motif_id = as.factor(motif_id),
    targets = as.factor(targets),
    TFAP2B = ifelse(
      str_detect(targets, &quot;TFAP2B&quot;), &quot;TFAP2B&quot;, &quot;No TFAP2B&quot;
    )
  ) %&gt;% 
  group_by(cluster) %&gt;% 
  mutate(
    best_tp = max(tp_percent)
  ) %&gt;% 
  ungroup() %&gt;% 
  group_by(motif_id) %&gt;% 
  dplyr::filter(any(tp_percent == best_tp)) %&gt;% 
  ungroup() %&gt;% 
  plot_ame_heatmap(
    id = motif_id,
    value = tp_percent,
    group = fct_rev(targets)
  ) +
  geom_label(
    aes(label = percent(tp_percent/100, 0.1)),
    size = 3.5,
    show.legend = FALSE
  ) +
  facet_grid(TFAP2B ~ motif_alt_id, scales = &quot;free&quot;, space = &quot;free&quot;) +
  scale_x_discrete(expand = expansion(c(0, 0))) +
  scale_y_discrete(expand = expansion(c(0, 0))) +
  labs(
    x = &quot;Detected Motif&quot;,
    y = &quot;Detected ChIP Targets&quot;,
    fill = expression(widehat(&quot;TP%&quot;))
  ) +
  theme(
    axis.text.x = element_text(angle = 0, hjust = 0.5, size = 10),
    axis.text.y = element_text(angle = 0, hjust = 1, size = 10),
    panel.grid = element_blank()
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/motif_analysis.Rmd/plot-ame-other-all-1.png" alt="Enriched motifs when analysing all sequences associated with one or more ChIP targets. Motifs are only shown if found in &gt;50% of sequences from at least one set of peaks." width="960" />
<p class="caption">
Enriched motifs when analysing all sequences associated with one or more
ChIP targets. Motifs are only shown if found in &gt;50% of sequences
from at least one set of peaks.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-ame-other-all-1">
Past versions of plot-ame-other-all-1.png
</button>
</p>
<div id="fig-plot-ame-other-all-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/motif_analysis.Rmd/plot-ame-other-all-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/8dde309446d2e562a920f7fa7c87e825ff5cb64d/docs/figure/motif_analysis.Rmd/plot-ame-other-all-1.png" target="_blank">8dde309</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="h3k27ac-active-sequences" class="section level4">
<h4>H3K27ac Active Sequences</h4>
<p>When restricting the peaks to being any ChIP-target binding peak
associated with H3K27ac significantly more motifs were considered to be
enriched.</p>
<p>Notably:</p>
<ul>
<li><a href="https://jaspar.genereg.net/matrix/MA0671.1/">NFIX</a>
enrichment was mostly lost, especially in the set of peaks associated
with all four targets. Significance was retained for FOXA1 and GATA3 in
combination, but only in the absence of TFAP2<span
class="math inline">\(\beta\)</span></li>
<li><a href="https://jaspar.genereg.net/matrix/MA0738.1/">HIC2</a> was
enriched in all combinations where AR and FOXA1 were bound in
combination. However, HIC2 was not found in combination with AR in the
RIME dataset.</li>
</ul>
<p>No other striking patterns were observed.</p>
<pre class="r"><code>x &lt;- ame_by_targets_active %&gt;% 
  bind_rows(.id = &quot;targets&quot;) %&gt;% 
  dplyr::filter(
    !motif_alt_id %in% targets,
    !str_detect(motif_alt_id, &quot;(GATA[0-9]|FOX[A-Z][0-9]|TFAP2[A-Z])&quot;),
    tp_percent &gt; 50
  ) %&gt;% 
  distinct(motif_id) %&gt;% 
  pull(&quot;motif_id&quot;)
cormat &lt;- meme_db %&gt;% 
  dplyr::filter(name %in% x) %&gt;% 
  to_list() %&gt;% 
  compare_motifs()
cormat[cormat &lt; 0.9] &lt;- 0
cormat[is.na(cormat)] &lt;- 0
cl &lt;- as.dist(1 - cormat) %&gt;% 
  hclust() %&gt;% 
  cutree(h= 0.9)
ame_by_targets_active %&gt;% 
  bind_rows(.id = &quot;targets&quot;) %&gt;% 
  dplyr::filter(motif_id %in% names(cl)) %&gt;% 
  mutate(
    cluster = cl[motif_id],
    motif_id = as.factor(motif_id),
    targets = as.factor(targets),
    TFAP2B = ifelse(
      str_detect(targets, &quot;TFAP2B&quot;), &quot;TFAP2B&quot;, &quot;No TFAP2B&quot;
    )
  ) %&gt;% 
  group_by(cluster) %&gt;% 
  mutate(
    best_tp = max(tp_percent)
  ) %&gt;% 
  ungroup() %&gt;% 
  group_by(motif_id) %&gt;% 
  dplyr::filter(any(tp_percent == best_tp)) %&gt;% 
  ungroup() %&gt;% 
  plot_ame_heatmap(
    id = motif_id,
    value = tp_percent,
    group = fct_rev(targets)
  ) +
  geom_label(
    aes(label = percent(tp_percent/100, 0.1)),
    size = 3,
    show.legend = FALSE
  ) +
  facet_grid(TFAP2B ~ motif_alt_id, scales = &quot;free&quot;, space = &quot;free&quot;) +
  scale_x_discrete(expand = expansion(c(0, 0))) +
  scale_y_discrete(expand = expansion(c(0, 0))) +
  labs(
    x = &quot;Detected Motif&quot;,
    y = &quot;Detected ChIP Targets&quot;,
    fill = expression(widehat(&quot;TP%&quot;))
  ) +
  theme(
    axis.text.x = element_text(angle = 0, hjust = 0.5, size = 8),
    axis.text.y = element_text(angle = 0, hjust = 1, size = 8),
    panel.grid = element_blank()
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/motif_analysis.Rmd/plot-ame-active-all-1.png" alt="Enriched motifs when analysing all sequences associated with one or more ChIP targets **and which overlap an H3K27ac peak**. Motifs are only shown if found in &gt;50% of sequences from at least one set of peaks." width="960" />
<p class="caption">
Enriched motifs when analysing all sequences associated with one or more
ChIP targets <strong>and which overlap an H3K27ac peak</strong>. Motifs
are only shown if found in &gt;50% of sequences from at least one set of
peaks.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-ame-active-all-1">
Past versions of plot-ame-active-all-1.png
</button>
</p>
<div id="fig-plot-ame-active-all-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/motif_analysis.Rmd/plot-ame-active-all-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/8dde309446d2e562a920f7fa7c87e825ff5cb64d/docs/figure/motif_analysis.Rmd/plot-ame-active-all-1.png" target="_blank">8dde309</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="h3k27ac-active-vs-inactive" class="section level3">
<h3>H3K27ac Active Vs Inactive</h3>
<p>In order to ascertain whether any co-factors are playing a role in
active vs inactive regulatory regions, the set of sequences associated
with binding of all four targets were compared to each other, based on
H3K27ac overlap as the marker of an active regulatory region. Enriched
motifs were those with an <span class="math inline">\(E\)</span>-value
&lt; 0.01 and with a True Positive match in &gt; 10% of sequences.</p>
<p>Due to the high-levels of similarity between identified motifs,
enriched motifs were clustered by:</p>
<ol style="list-style-type: decimal">
<li>Finding the pairwise correlations</li>
<li>Setting all pairwise correlations &lt; 0.8 to zero</li>
<li>Using hierarchical clustering to cluster those with retained
correlations</li>
</ol>
<p>For each motif cluster the values used for visualisation were the
lowest <span class="math inline">\(E\)</span>-value within the cluster
and the high estimate of True Positive matches.</p>
<pre class="r"><code>x &lt;- ame_active_vs_inactive$`AR FOXA1 GATA3 TFAP2B` %&gt;%
  dplyr::filter(
    evalue &lt; 0.01,
    tp_percent &gt; 10
  ) %&gt;%
  distinct(motif_id) %&gt;% 
  pull(&quot;motif_id&quot;)
cormat &lt;- meme_db %&gt;% 
  dplyr::filter(name %in% x) %&gt;% 
  to_list() %&gt;% 
  compare_motifs()
cormat[cormat &lt; 0.80] &lt;- 0
cormat[is.na(cormat)] &lt;- 0
cl_active_vs_inactive &lt;- as.dist(1 - cormat) %&gt;% 
  hclust() %&gt;% 
  cutree(h= 0.9)
ame_active_vs_inactive$`AR FOXA1 GATA3 TFAP2B` %&gt;% 
  dplyr::filter(motif_id %in% names(cl_active_vs_inactive)) %&gt;% 
  mutate(
    cluster = cl_active_vs_inactive[motif_id], 
    in_rime = motif_alt_id %in% rime_detected$gene_name,
    motif_alt_id = ifelse(in_rime, paste0(motif_alt_id, &quot;*&quot;), motif_alt_id)
  ) %&gt;% 
  group_by(cluster) %&gt;% 
  summarise(
    tp_percent = max(tp_percent),
    p = min(evalue),
    id = motif_alt_id %&gt;% 
      unique %&gt;% 
      sort() %&gt;% 
      paste(collapse = &quot;/&quot;)
  ) %&gt;% 
  arrange(p) %&gt;% 
  mutate(
    id = str_trunc(id, width = 40) %&gt;% fct_inorder()
  ) %&gt;% 
  ggplot(
    aes(tp_percent, fct_rev(id), fill = -log10(p))
  ) +
  geom_col() +
  scale_x_continuous(expand = expansion(c(0, 0.05))) +
  scale_fill_viridis_c() +
  labs(
    x = &quot;True Positive Matches (%)&quot;,
    y = &quot;Motif Cluster&quot;
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/motif_analysis.Rmd/plot-motifs-active-vs-inactive-1.png" alt="Motifs found enriched when comparing sequences from peaks overlapping an active H3K27ac regulatory region, with those from peaks not overlapping an H3K27ac-marked region. Motifs were clustered as described in the text, with any TFs detected as bound to AR in the RIME dataset indicated with an asterisk. Interestingly, TFAP2B motifs appeared to be enriched in this set of sequences." width="960" />
<p class="caption">
Motifs found enriched when comparing sequences from peaks overlapping an
active H3K27ac regulatory region, with those from peaks not overlapping
an H3K27ac-marked region. Motifs were clustered as described in the
text, with any TFs detected as bound to AR in the RIME dataset indicated
with an asterisk. Interestingly, TFAP2B motifs appeared to be enriched
in this set of sequences.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-motifs-active-vs-inactive-1">
Past versions of plot-motifs-active-vs-inactive-1.png
</button>
</p>
<div id="fig-plot-motifs-active-vs-inactive-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/motif_analysis.Rmd/plot-motifs-active-vs-inactive-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/0c127407bf36122be17950abac95eb8d8ddfc9b8/docs/figure/motif_analysis.Rmd/plot-motifs-active-vs-inactive-1.png" target="_blank">0c12740</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/8dde309446d2e562a920f7fa7c87e825ff5cb64d/docs/figure/motif_analysis.Rmd/plot-motifs-active-vs-inactive-1.png" target="_blank">8dde309</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="h3k27ac-inctive-vs-active" class="section level3">
<h3>H3K27ac Inctive Vs Active</h3>
<p>The same approach was then take for sequences associated with
<em>inactive</em> regulatory regions, by comparing these sequences to
those from <em>active</em> regulatory regions.</p>
<pre class="r"><code>x &lt;- ame_inactive_vs_active$`AR FOXA1 GATA3 TFAP2B` %&gt;%
  dplyr::filter(evalue &lt; 0.01, tp_percent &gt; 10) %&gt;%
  distinct(motif_id) %&gt;% 
  pull(&quot;motif_id&quot;)
cormat &lt;- meme_db %&gt;% 
  dplyr::filter(name %in% x) %&gt;% 
  to_list() %&gt;% 
  compare_motifs()
cormat[cormat &lt; 0.8] &lt;- 0
cormat[is.na(cormat)] &lt;- 0
cl_inactive_vs_active &lt;- as.dist(1 - cormat) %&gt;% 
  hclust() %&gt;% 
  cutree(h= 0.9)
ame_inactive_vs_active$`AR FOXA1 GATA3 TFAP2B` %&gt;% 
  dplyr::filter(motif_id %in% names(cl_inactive_vs_active)) %&gt;% 
  mutate(
    cluster = cl_inactive_vs_active[motif_id], 
    in_rime = motif_alt_id %in% rime_detected$gene_name,
    motif_alt_id = ifelse(in_rime, paste0(motif_alt_id, &quot;*&quot;), motif_alt_id)
  ) %&gt;% 
  group_by(cluster) %&gt;% 
  summarise(
    tp_percent = max(tp_percent),
    p = min(evalue),
    id = motif_alt_id %&gt;% 
      unique %&gt;% 
      sort() %&gt;% 
      paste(collapse = &quot;/&quot;)
  ) %&gt;% 
  arrange(p) %&gt;% 
  mutate(
    id = str_trunc(id, width = 40) %&gt;% fct_inorder()
  ) %&gt;% 
  ggplot(
    aes(tp_percent, fct_rev(id), fill = -log10(p))
  ) +
  geom_col() +
  scale_x_continuous(expand = expansion(c(0, 0.05))) +
  scale_fill_viridis_c() +
  labs(
    x = &quot;True Positive Matches (%)&quot;,
    y = &quot;Motif Cluster&quot;
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/motif_analysis.Rmd/plot-motifs-inactive-vs-active-1.png" alt="Motifs found enriched when comparing sequences from peaks overlapping an inactive H3K27ac regulatory region, with those from peaks overlapping an H3K27ac-marked region. Motifs were clustered as described in the text, with any TFs detected as bound to AR in the RIME dataset indicated with an asterisk. Both FOXA1 and GATA3 motifs appeared to be enriched in this set of sequences." width="960" />
<p class="caption">
Motifs found enriched when comparing sequences from peaks overlapping an
inactive H3K27ac regulatory region, with those from peaks overlapping an
H3K27ac-marked region. Motifs were clustered as described in the text,
with any TFs detected as bound to AR in the RIME dataset indicated with
an asterisk. Both FOXA1 and GATA3 motifs appeared to be enriched in this
set of sequences.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-motifs-inactive-vs-active-1">
Past versions of plot-motifs-inactive-vs-active-1.png
</button>
</p>
<div id="fig-plot-motifs-inactive-vs-active-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/motif_analysis.Rmd/plot-motifs-inactive-vs-active-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/0c127407bf36122be17950abac95eb8d8ddfc9b8/docs/figure/motif_analysis.Rmd/plot-motifs-inactive-vs-active-1.png" target="_blank">0c12740</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/8dde309446d2e562a920f7fa7c87e825ff5cb64d/docs/figure/motif_analysis.Rmd/plot-motifs-inactive-vs-active-1.png" target="_blank">8dde309</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="further-investigations" class="section level2">
<h2>Further Investigations</h2>
<p>Given the unexpected result that TFAP2<span
class="math inline">\(\beta\)</span> motifs were enriched in the
<em>H3K27ac marked peaks</em> (i.e. active regulatory regions), whilst
FOXA1 and GATA motifs were enriched in the peaks <strong>not
marked</strong> by H3K27ac (i.e. inactive regulatory regions), the
co-occurrence of these motifs was explored across both sets of peaks as
a predictive tool for an active regulatory region. Sequences were
checked for the simple presence of each motif, using matches with a
score &gt; 60% of the maximum possible score as a positive match, to be
as inclusive as possible. For all sequences, only <em>the highest
scoring match to the consensus motif</em> was retained. For
comparability across motifs, motif match scores were normalised to
Z-scores.</p>
<pre class="r"><code>inv.logit &lt;- binomial()$linkinv
min_score &lt;- percent(0.6, accuracy = 1)
all4_motifs &lt;- meme_db_detected %&gt;%
  dplyr::filter(altname %in% targets) %&gt;% 
  to_list() %&gt;% 
  mclapply(
    getBestMatch, 
    subject = seq_by_targets$`AR FOXA1 GATA3 TFAP2B` %&gt;% 
      unlist() %&gt;% 
      as(&quot;DNAStringSet&quot;) %&gt;% 
      as(&quot;Views&quot;) %&gt;% 
      setNames(
        seq_by_targets$`AR FOXA1 GATA3 TFAP2B` %&gt;% 
          unlist() %&gt;% 
          names() %&gt;% 
          str_remove_all(&quot;^(TRUE|FALSE)\\.&quot;)
      ),
    min.score = min_score,
    mc.cores = length(targets)
  ) %&gt;% 
  setNames(
    dplyr::filter(meme_db_detected, altname %in% targets)$name
  ) %&gt;% 
  bind_rows(.id = &quot;name&quot;) %&gt;% 
  left_join(
    dplyr::select(meme_info, name, altname, motif_length),
    by = &quot;name&quot;
  ) %&gt;% 
  dplyr::rename(range = view_name) %&gt;% 
  full_join(
    grl_by_targets$`AR FOXA1 GATA3 TFAP2B` %&gt;% 
      as_tibble(),
    by = &quot;range&quot; 
  ) 
glm_h3K27ac &lt;- all4_motifs %&gt;% 
  group_by(name) %&gt;% 
  mutate(
    score = (score - mean(score, na.rm = TRUE)) / sd(score, na.rm = TRUE)
  ) %&gt;% 
  pivot_wider(
    names_from = &quot;name&quot;, values_from = &quot;score&quot;, id_cols = c(&quot;range&quot;, &quot;H3K27ac&quot;)
  ) %&gt;% 
  dplyr::filter(
    !is.na(MA0811.1), !is.na(MA0812.1), !is.na(MA0813.1)
  ) %&gt;% 
  mutate(
    PC1 = cbind(MA0811.1, MA0812.1, MA0813.1) %&gt;% 
      prcomp() %&gt;% 
      .[[&quot;x&quot;]] %&gt;% 
      .[,&quot;PC1&quot;] %&gt;% 
      multiply_by(-1), # This always seems to come out in the opposite direction to observed
    width = width(GRanges(range))
  ) %&gt;% 
  with(
    glm(H3K27ac ~ width + (PC1 + MA0007.2 + MA0148.4 + MA0037.3)^4 , family = &quot;binomial&quot;)
  ) %&gt;% 
  step() %&gt;% 
  broom::tidy() %&gt;% 
  mutate(
    term = term %&gt;% 
      str_replace_all(&quot;PC1&quot;, &quot;TFAP2B (PC1)&quot;) %&gt;% 
      str_replace_all(&quot;MA0007.2&quot;, &quot;AR&quot;) %&gt;% 
      str_replace_all(&quot;MA0148.4&quot;, &quot;FOXA1&quot;) %&gt;% 
      str_replace_all(&quot;MA0037.3&quot;, &quot;GATA3&quot;),
    fdr = p.adjust(p.value, &quot;fdr&quot;),
    ` ` = case_when(
      fdr &gt; 0.1 ~ &quot;&quot;,
      fdr &gt; 0.05 ~ &quot;.&quot;,
      fdr &gt; 0.01 ~ &quot;*&quot;,
      fdr &gt; 0.001 ~ &quot;**&quot;,
      TRUE ~ &quot;***&quot;
    )
  ) </code></pre>
<pre class="r"><code>all4_motifs %&gt;% 
  group_by(name) %&gt;% 
  mutate(
    score = (score - mean(score, na.rm = TRUE)) / sd(score, na.rm = TRUE)
  ) %&gt;% 
  ggplot(aes(name, score, fill = H3K27ac)) +
  geom_boxplot() +
  facet_grid(.~altname, scales = &quot;free&quot;, space = &quot;free&quot;) +
  scale_fill_manual(values = c(&quot;grey50&quot;, &quot;red&quot;)) +
  labs(
    x = &quot;Motif name&quot;, y = &quot;Normalised Motif Match Score&quot;
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/motif_analysis.Rmd/plot-all4-scores-1.png" alt="Distribution of Z-Scores for matches to each motif, separated by whether the peak was associated with an H3K27ac mark. Motifs with higher scores were clearly associated with H3K27ac marks for all TFAP2B motifs, whilst some tendency towards lower scores was noted for FOXA1 and GATA3, in keeping with the above enrichment results." width="960" />
<p class="caption">
Distribution of Z-Scores for matches to each motif, separated by whether
the peak was associated with an H3K27ac mark. Motifs with higher scores
were clearly associated with H3K27ac marks for all TFAP2B motifs, whilst
some tendency towards lower scores was noted for FOXA1 and GATA3, in
keeping with the above enrichment results.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot-all4-scores-1">
Past versions of plot-all4-scores-1.png
</button>
</p>
<div id="fig-plot-all4-scores-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/motif_analysis.Rmd/plot-all4-scores-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/0c127407bf36122be17950abac95eb8d8ddfc9b8/docs/figure/motif_analysis.Rmd/plot-all4-scores-1.png" target="_blank">0c12740</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>To quantify the impact of motif presence within the sequences where
all 4 ChIP targets were detected, the probability of a peak being
associated with H3K27ac marks was modelled using the strength of each
motif in respect to the PWM. To avoid the high correlations between the
three TFAP2<span class="math inline">\(\beta\)</span> motifs, these were
combined using PC1 returned from PCA across all three motifs as a
representative score. This may make the direction of the probability
more difficult to interpret as a lower values on PC1 may represent a
higher match across all 3 motifs.</p>
<p>A logistic regression model was fitted using H3K27ac status as a the
binary response variable with the scores for AR (MA0007.2), FOXA1
(MA0148.4) and GATA3 (MA0037.3) motifs being combined with PC1 across
all 3 TFAP2<span class="math inline">\(\beta\)</span> motifs as the
predictors. The initial model was fitted incorporating 4-way
interactions as the initial model, along with peak width, with terms
being removed in a step-wise manner using Akaike’s Information Criterion
(AIC).</p>
<pre class="r"><code>cp &lt;- htmltools::tags$caption(
  glue(
    &quot;
    Results from final logistic regression model for the probability of a peak 
    being marked by H3K27ac, based on the score of each detected motif&#39;s match 
    to the PWM, and the overall width of the peak. Whilst the width of the peak 
    clearly had a significant influence on the probability of a peak being 
    considered active, via H3K27ac, the presence of a strong TFAP2B motif also 
    had a clear influence on the likelihood of a peak being considered active.
    Conversely, the stronger matches to the core FOXA1 and GATA3 motifs 
    indicated a decreased association with H3K27ac marks, as predicted by the 
    sequence enrichment analysis.
    &quot;
  )
)
tbl &lt;- glm_h3K27ac %&gt;% 
  mutate(term = str_replace_all(term, &quot;width&quot;, &quot;Peak Width&quot;)) %&gt;% 
  reactable(
    columns = list(
      term = colDef(
        name = &quot;Model Term&quot;,
        cell = function(value) str_replace_all(value, &quot;:&quot;, &quot; : &quot;),
        minWidth = 150
      ),
      estimate = colDef(
        name = &quot;Estimate&quot;, format = colFormat(digits = 3)
      ),
      std.error = colDef(show = FALSE),
      statistic = colDef(name = &quot;Test Statistic&quot;),
      p.value = colDef(
        show = FALSE,
        cell = function(value) {
          ifelse(
            value &lt; 0.001,
            sprintf(&quot;%.2e&quot;, value),
            sprintf(&quot;%.3f&quot;, value)
          )
        }
      ),
      fdr = colDef(
        name = &quot;P&lt;sub&gt;FDR&lt;/sub&gt;&quot;, html = TRUE,
        cell = function(value) {
          ifelse(
            value &lt; 0.001,
            sprintf(&quot;%.2e&quot;, value),
            sprintf(&quot;%.3f&quot;, value)
          )
        }
      )
    ),
    defaultColDef = colDef(
      format = colFormat(digits = 2)
    )
  )
div(class = &quot;table&quot;,
  div(class = &quot;table-header&quot;,
      div(class = &quot;caption&quot;, cp),
      tbl
  )
)</code></pre>
<div class="table">
<div class="table-header">
<div class="caption">
<caption>Results from final logistic regression model for the probability of a peak 
being marked by H3K27ac, based on the score of each detected motif's match 
to the PWM, and the overall width of the peak. Whilst the width of the peak 
clearly had a significant influence on the probability of a peak being 
considered active, via H3K27ac, the presence of a strong TFAP2B motif also 
had a clear influence on the likelihood of a peak being considered active.
Conversely, the stronger matches to the core FOXA1 and GATA3 motifs 
indicated a decreased association with H3K27ac marks, as predicted by the 
sequence enrichment analysis.</caption>
</div>
<div id="htmlwidget-9929d7c7769fe1c64f8c" class="reactable html-widget" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-9929d7c7769fe1c64f8c">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"term":["(Intercept)","Peak Width","TFAP2B (PC1)","FOXA1","GATA3"],"estimate":[-2.05560320217623,0.0021141284570377,0.279677145224525,-0.269696957190027,-0.222170612642584],"std.error":[0.0881680195131966,0.000103232982071587,0.0237604626827377,0.0298273969275014,0.028086800238523],"statistic":[-23.3146124130479,20.4791958404501,11.7706944077194,-9.04192068270501,-7.91014322585105],"p.value":[3.15166919251173e-120,3.30062320916958e-93,5.52676945188586e-32,1.53942773045154e-19,2.57092940115215e-15],"fdr":[1.57583459625586e-119,8.25155802292394e-93,9.21128241980977e-32,1.92428466306442e-19,2.57092940115215e-15]," ":["***","***","***","***","***"]},"columns":[{"accessor":"term","name":"Model Term","type":"character","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"cell":["(Intercept)","Peak Width","TFAP2B (PC1)","FOXA1","GATA3"],"minWidth":150},{"accessor":"estimate","name":"Estimate","type":"numeric","format":{"cell":{"digits":3},"aggregated":{"digits":3}}},{"accessor":"std.error","name":"std.error","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"show":false},{"accessor":"statistic","name":"Test Statistic","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"accessor":"p.value","name":"p.value","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"show":false,"cell":["3.15e-120","3.30e-93","5.53e-32","1.54e-19","2.57e-15"]},{"accessor":"fdr","name":"P<sub>FDR<\/sub>","type":"numeric","format":{"cell":{"digits":2},"aggregated":{"digits":2}},"cell":["1.58e-119","8.25e-93","9.21e-32","1.92e-19","2.57e-15"],"html":true},{"accessor":" ","name":" ","type":"character","format":{"cell":{"digits":2},"aggregated":{"digits":2}}}],"defaultPageSize":10,"paginationType":"numbers","showPageInfo":true,"minRows":1,"dataKey":"e4245c498c84d5e89c78fcfd4a6b62b2"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</div>
</div>
<div id="distance-between-motifs" class="section level1 tabset">
<h1 class="tabset">Distance Between Motifs</h1>
<p>The distance between the best match to each motif was then assessed
for the set of peaks for which <em>all four targets were detected</em>,
separated by H3K27ac status.</p>
<pre class="r"><code>pwm_lengths &lt;- structure(meme_info$motif_length, names = meme_info$name)
all4_motifs %&gt;% 
  arrange(view) %&gt;% 
  ## The best strategy is to find the distance between centres, then add half
  ## the width of each motif
  mutate(match_centre = (start_in_view + end_in_view) / 2) %&gt;% 
  pivot_wider(
    names_from = &quot;name&quot;, values_from = &quot;match_centre&quot;, id_cols = c(&quot;range&quot;, &quot;H3K27ac&quot;)
  ) %&gt;% 
  mutate(
    ## AR
    MA0007.2_MA0037.3 = abs(MA0007.2  - MA0037.3 ) - 0.5*(sum(pwm_lengths[c(&quot;MA0007.2&quot;, &quot;MA0037.3&quot;)])),
    MA0007.2_MA0148.4 = abs(MA0007.2  - MA0148.4 ) - 0.5*(sum(pwm_lengths[c(&quot;MA0007.2&quot;, &quot;MA0148.4&quot;)])),
    MA0007.2_MA0811.1 = abs(MA0007.2  - MA0811.1 ) - 0.5*(sum(pwm_lengths[c(&quot;MA0007.2&quot;, &quot;MA0811.1&quot;)])),
    MA0007.2_MA0812.1 = abs(MA0007.2  - MA0812.1 ) - 0.5*(sum(pwm_lengths[c(&quot;MA0007.2&quot;, &quot;MA0812.1&quot;)])),
    MA0007.2_MA0813.1 = abs(MA0007.2  - MA0813.1 ) - 0.5*(sum(pwm_lengths[c(&quot;MA0007.2&quot;, &quot;MA0813.1&quot;)])),
    ## FOXA1
    MA0148.4_MA0037.3 = abs(MA0148.4  - MA0037.3 ) - 0.5*(sum(pwm_lengths[c(&quot;MA0148.4&quot;, &quot;MA0037.3&quot;)])),
    MA0148.4_MA0811.1 = abs(MA0148.4  - MA0811.1 ) - 0.5*(sum(pwm_lengths[c(&quot;MA0148.4&quot;, &quot;MA0811.1&quot;)])),
    MA0148.4_MA0812.1 = abs(MA0148.4  - MA0812.1 ) - 0.5*(sum(pwm_lengths[c(&quot;MA0148.4&quot;, &quot;MA0812.1&quot;)])),
    MA0148.4_MA0813.1 = abs(MA0148.4  - MA0813.1 ) - 0.5*(sum(pwm_lengths[c(&quot;MA0148.4&quot;, &quot;MA0813.1&quot;)])),
    ## GATA3
    MA0037.3_MA0811.1 = abs(MA0037.3  - MA0811.1 ) - 0.5*(sum(pwm_lengths[c(&quot;MA0037.3&quot;, &quot;MA0811.1&quot;)])),
    MA0037.3_MA0812.1 = abs(MA0037.3  - MA0812.1 ) - 0.5*(sum(pwm_lengths[c(&quot;MA0037.3&quot;, &quot;MA0812.1&quot;)])),
    MA0037.3_MA0813.1 = abs(MA0037.3  - MA0813.1 ) - 0.5*(sum(pwm_lengths[c(&quot;MA0037.3&quot;, &quot;MA0813.1&quot;)])),
  ) %&gt;% 
  dplyr::select(range, H3K27ac, contains(&quot;_&quot;)) %&gt;% 
  pivot_longer(starts_with(&quot;MA&quot;), names_to = &quot;motifs&quot;, values_to = &quot;d&quot;) %&gt;% 
  separate(motifs, into = c(&quot;pwm_1&quot;, &quot;pwm_2&quot;), sep = &quot;_&quot;) %&gt;% 
  left_join(
    dplyr::select(meme_info, pwm_1 = name, altname_1 = altname), by = &quot;pwm_1&quot;
  ) %&gt;% 
  left_join(
    dplyr::select(meme_info, pwm_2 = name, altname_2 = altname), by = &quot;pwm_2&quot;
  ) %&gt;% 
  arrange(range, d) %&gt;% 
  distinct(range, altname_1, altname_2, .keep_all = TRUE) %&gt;% 
  unite(&quot;comparison&quot;, starts_with(&quot;altname&quot;), sep = &quot; - &quot;) %&gt;%
  mutate(
    d = ifelse(d &lt; 0, 0, d),
    H3K27ac = ifelse(H3K27ac, &quot;Active&quot;, &quot;Inactive&quot;)
  ) %&gt;% 
  ggplot(aes(d, stat(density), colour = H3K27ac)) +
  geom_density() +
  geom_vline(
    aes(xintercept = med, colour = H3K27ac),
    data = . %&gt;% 
      group_by(comparison, H3K27ac) %&gt;% 
      summarise(med = median(d), .groups = &quot;drop&quot;),
    show.legend = FALSE
  ) +
  facet_grid(comparison ~ ., scales = &quot;free_y&quot;) +
  scale_colour_manual(values = c(&quot;red&quot;, &quot;black&quot;)) +
  coord_cartesian(xlim = c(0, 500)) +
  theme(legend.position = &quot;top&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/motif_analysis.Rmd/fig-compare-distance-1.png" alt="Distances between motifs for the best match for each motif. For comparisons to TFAP2B motifs, the nearest of the three was selected. Whilst the median distances between AR, FOXA1 and GATA3 motifs tended to be between 160 and 200bp, the median distances between TFAP2B motifs and the the other motifs tended to be around 80 to 100bp, suggesting that TFAP2B may bind between two of the other three targets in many complexes. A trend towards motifs being further aart within the active regulatory regions was moted, but a simple explanation for this remains elusive." width="768" />
<p class="caption">
Distances between motifs for the best match for each motif. For
comparisons to TFAP2B motifs, the nearest of the three was selected.
Whilst the median distances between AR, FOXA1 and GATA3 motifs tended to
be between 160 and 200bp, the median distances between TFAP2B motifs and
the the other motifs tended to be around 80 to 100bp, suggesting that
TFAP2B may bind between two of the other three targets in many
complexes. A trend towards motifs being further aart within the active
regulatory regions was moted, but a simple explanation for this remains
elusive.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig-compare-distance-1">
Past versions of fig-compare-distance-1.png
</button>
</p>
<div id="fig-fig-compare-distance-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DRMCRL/apocrine_signature_mdamb453/blob/1f959c32bb2c748b3bb8b8748bbc6b305c418348/docs/figure/motif_analysis.Rmd/fig-compare-distance-1.png" target="_blank">1f959c3</a>
</td>
<td>
Steve Pederson
</td>
<td>
2022-06-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="alignment-of-key-motifs" class="section level1">
<h1>Alignment Of Key Motifs</h1>
<p>Showing all motifs in Richard’s manually curated list</p>
<pre class="r"><code>ids &lt;- c(&quot;MA0640.2&quot;, &quot;MA0136.2&quot;, &quot;MA0760.1&quot;, &quot;MA0686.1&quot;)
meme_db %&gt;% 
  dplyr::filter(name %in% ids) %&gt;% 
  mutate(name = factor(name, levels = ids)) %&gt;% 
  arrange(name) %&gt;% 
  mutate(name = as.character(name)) %&gt;% 
  to_list() %&gt;% 
  lapply(
    function(x) {
      x@name &lt;- paste0(str_pad(x@altname, 16, side = &quot;both&quot;), &quot;\n(&quot;, x@name, &quot;)&quot;)
      x
    }
  ) %&gt;% 
  view_motifs(names.pos = &quot;right&quot;)</code></pre>
<p><img src="figure/motif_analysis.Rmd/plot-motifs-elf3-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Showing only the motifs associated with detected genes, noting that
<em>ELF5</em> expression was not detected in the RNA-seq dataset</p>
<pre class="r"><code>meme_db_detected %&gt;% 
  dplyr::filter(name %in% ids) %&gt;% 
  mutate(name = factor(name, levels = ids)) %&gt;% 
  arrange(name) %&gt;% 
  mutate(name = as.character(name)) %&gt;% 
  to_list() %&gt;% 
  lapply(
    function(x) {
      x@name &lt;- paste0(str_pad(x@altname, 16, side = &quot;both&quot;), &quot;\n(&quot;, x@name, &quot;)&quot;)
      x
    }
  ) %&gt;% 
  view_motifs(names.pos = &quot;right&quot;)</code></pre>
<p><img src="figure/motif_analysis.Rmd/plot-motifs-elf-detected-1.png" width="960" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] multcomp_1.4-19                   TH.data_1.1-1                    
 [3] MASS_7.3-56                       survival_3.2-13                  
 [5] mvtnorm_1.1-3                     htmltools_0.5.2                  
 [7] reactable_0.3.0                   UpSetR_1.4.0                     
 [9] readxl_1.4.0                      gt_0.6.0                         
[11] cowplot_1.1.1                     glue_1.6.2                       
[13] pander_0.6.5                      scales_1.2.0                     
[15] BSgenome.Hsapiens.UCSC.hg19_1.4.3 BSgenome_1.64.0                  
[17] Biostrings_2.64.0                 XVector_0.36.0                   
[19] magrittr_2.0.3                    rtracklayer_1.56.0               
[21] extraChIPs_1.0.1                  SummarizedExperiment_1.26.1      
[23] Biobase_2.56.0                    MatrixGenerics_1.8.0             
[25] matrixStats_0.62.0                BiocParallel_1.30.2              
[27] plyranges_1.16.0                  GenomicRanges_1.48.0             
[29] GenomeInfoDb_1.32.2               IRanges_2.30.0                   
[31] S4Vectors_0.34.0                  BiocGenerics_0.42.0              
[33] forcats_0.5.1                     stringr_1.4.0                    
[35] dplyr_1.0.9                       purrr_0.3.4                      
[37] readr_2.1.2                       tidyr_1.2.0                      
[39] tibble_3.1.7                      ggplot2_3.3.6                    
[41] tidyverse_1.3.1                   universalmotif_1.14.1            
[43] memes_1.4.1                       workflowr_1.7.0                  

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3             R.methodsS3_1.8.1         
  [3] csaw_1.30.1                bit64_4.0.5               
  [5] knitr_1.39                 R.utils_2.11.0            
  [7] DelayedArray_0.22.0        data.table_1.14.2         
  [9] rpart_4.1.16               KEGGREST_1.36.0           
 [11] RCurl_1.98-1.6             AnnotationFilter_1.20.0   
 [13] doParallel_1.0.17          generics_0.1.2            
 [15] GenomicFeatures_1.48.3     callr_3.7.0               
 [17] EnrichedHeatmap_1.26.0     usethis_2.1.6             
 [19] RSQLite_2.2.14             bit_4.0.4                 
 [21] tzdb_0.3.0                 xml2_1.3.3                
 [23] lubridate_1.8.0            httpuv_1.6.5              
 [25] assertthat_0.2.1           xfun_0.31                 
 [27] hms_1.1.1                  jquerylib_0.1.4           
 [29] evaluate_0.15              promises_1.2.0.1          
 [31] fansi_1.0.3                restfulr_0.0.13           
 [33] progress_1.2.2             dbplyr_2.1.1              
 [35] igraph_1.3.1               DBI_1.1.2                 
 [37] htmlwidgets_1.5.4          ellipsis_0.3.2            
 [39] crosstalk_1.2.0            backports_1.4.1           
 [41] biomaRt_2.52.0             vctrs_0.4.1               
 [43] here_1.0.1                 ensembldb_2.20.1          
 [45] cachem_1.0.6               withr_2.5.0               
 [47] ggforce_0.3.3              Gviz_1.40.1               
 [49] vroom_1.5.7                checkmate_2.1.0           
 [51] GenomicAlignments_1.32.0   prettyunits_1.1.1         
 [53] cluster_2.1.3              lazyeval_0.2.2            
 [55] crayon_1.5.1               labeling_0.4.2            
 [57] edgeR_3.38.1               pkgconfig_2.0.3           
 [59] tweenr_1.0.2               pkgload_1.2.4             
 [61] ProtGenerics_1.28.0        nnet_7.3-17               
 [63] rlang_1.0.2                lifecycle_1.0.1           
 [65] sandwich_3.0-1             filelock_1.0.2            
 [67] BiocFileCache_2.4.0        modelr_0.1.8              
 [69] dichromat_2.0-0.1          cellranger_1.1.0          
 [71] rprojroot_2.0.3            polyclip_1.10-0           
 [73] reactR_0.4.4               Matrix_1.4-1              
 [75] ggseqlogo_0.1              zoo_1.8-10                
 [77] reprex_2.0.1               base64enc_0.1-3           
 [79] whisker_0.4                GlobalOptions_0.1.2       
 [81] processx_3.5.3             viridisLite_0.4.0         
 [83] png_0.1-7                  rjson_0.2.21              
 [85] GenomicInteractions_1.30.0 bitops_1.0-7              
 [87] R.oo_1.24.0                cmdfun_1.0.2              
 [89] getPass_0.2-2              blob_1.2.3                
 [91] shape_1.4.6                jpeg_0.1-9                
 [93] memoise_2.0.1              plyr_1.8.7                
 [95] zlibbioc_1.42.0            compiler_4.2.0            
 [97] scatterpie_0.1.7           BiocIO_1.6.0              
 [99] RColorBrewer_1.1-3         clue_0.3-61               
[101] Rsamtools_2.12.0           cli_3.3.0                 
[103] ps_1.7.0                   htmlTable_2.4.0           
[105] Formula_1.2-4              ggside_0.2.0.9990         
[107] tidyselect_1.1.2           stringi_1.7.6             
[109] highr_0.9                  yaml_2.3.5                
[111] locfit_1.5-9.5             latticeExtra_0.6-29       
[113] ggrepel_0.9.1              grid_4.2.0                
[115] sass_0.4.1                 VariantAnnotation_1.42.1  
[117] tools_4.2.0                circlize_0.4.15           
[119] rstudioapi_0.13            foreach_1.5.2             
[121] foreign_0.8-82             git2r_0.30.1              
[123] metapod_1.4.0              gridExtra_2.3             
[125] farver_2.1.0               digest_0.6.29             
[127] Rcpp_1.0.8.3               broom_0.8.0               
[129] later_1.3.0                httr_1.4.3                
[131] AnnotationDbi_1.58.0       biovizBase_1.44.0         
[133] ComplexHeatmap_2.12.0      colorspace_2.0-3          
[135] rvest_1.0.2                brio_1.1.3                
[137] XML_3.99-0.9               fs_1.5.2                  
[139] splines_4.2.0              jsonlite_1.8.0            
[141] ggfun_0.0.6                testthat_3.1.4            
[143] R6_2.5.1                   Hmisc_4.7-0               
[145] pillar_1.7.0               fastmap_1.1.0             
[147] codetools_0.2-18           utf8_1.2.2                
[149] lattice_0.20-45            bslib_0.3.1               
[151] curl_4.3.2                 limma_3.52.1              
[153] rmarkdown_2.14             InteractionSet_1.24.0     
[155] desc_1.4.1                 munsell_0.5.0             
[157] GetoptLong_1.0.5           GenomeInfoDbData_1.2.8    
[159] iterators_1.0.14           haven_2.5.0               
[161] gtable_0.3.0              </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
