---
title: "Motif Analysis"
author: "Steve Pederson"
date: "`r format(Sys.Date(), '%d %B, %Y')`"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r setup, echo=FALSE}
knitr::opts_chunk$set(
  message = FALSE, warning = FALSE,
  fig.width = 10, fig.height = 8
)
```


## Introduction

This step takes the peaks defined previously and 

1. Assesses motif enrichment for all ChIP targets within the peaks
2. Detect additional binding motifs 

As AR is only found bound in the presence of DHT, all peaks are taken using DHT-treatment as the reference condition.

```{r packages}
library(memes)
library(universalmotif)
library(tidyverse)
library(plyranges)
library(extraChIPs)
library(rtracklayer)
library(magrittr)
library(BSgenome.Hsapiens.UCSC.hg19)
library(scales)
library(pander)
library(glue)
library(cowplot)
library(parallel)
library(gt)
library(parallel)
library(readxl)
library(UpSetR)
library(reactable)
library(htmltools)
library(multcomp)
```

```{r options}
options(meme_bin = "/opt/meme/bin/")
check_meme_install()
theme_set(theme_bw())
panderOptions("big.mark", ",")
n_cores <- 12
```

```{r extra-funs}
getBestMatch <- function(x, subject, min.score = "80%", rc = TRUE, ...) {
  
  pwm <- NULL
  if (is.matrix(x)) pwm <- x
  if (is(x, "universalmotif")) pwm <- x@motif
  stopifnot(!is.null(pwm))
  stopifnot(is(subject, "XStringViews"))
  
  matches <- matchPWM(
    pwm, subject, min.score = min.score, with.score = TRUE, ...
  )
  df <- as.data.frame(matches)
  df$score <- mcols(matches)$score
  df$strand <- "+"
  
  if (rc) {
    rc_matches <-  matchPWM(
      reverseComplement(pwm), subject ,
      min.score = min.score, with.score = TRUE, 
      ...
    )
    rc_df <- as.data.frame(rc_matches)
    rc_df$score <- mcols(rc_matches)$score
    rc_df$strand <- "-"
    df <- bind_rows(df, rc_df)
  }
  
  grp_cols <- c("start", "end", "width", "seq", "score")
  grp_df <- group_by(df, !!!syms(grp_cols))
  summ_df <- summarise(
    grp_df, strand = paste(strand, collapse = " "), .groups=  "drop"
  )
  summ_df$strand <- str_replace_all(summ_df$strand, ". .", "*")
  summ_df$strand <- factor(summ_df$strand, levels = c("+", "-", "*"))
  summ_df$view <- findInterval(summ_df$start, start(subject))
  summ_df <- arrange(summ_df, view, desc(score))
  
  best_df <- distinct(summ_df, view, .keep_all = TRUE)
  best_df <- mutate(
    best_df,
    start_in_view = start - start(subject)[view] + 1,
    end_in_view = start_in_view + width
  )
  ## This will only be added if the views have names
  best_df$view_name <- names(subject[best_df$view])
  
  best_df
  
}
bar_style <- function(width = 1, fill = "#e6e6e6", height = "75%", align = c("left", "right"), color = NULL) {
  align <- match.arg(align)
  if (align == "left") {
    position <- paste0(width * 100, "%")
    image <- sprintf("linear-gradient(90deg, %1$s %2$s, transparent %2$s)", fill, position)
  } else {
    position <- paste0(100 - width * 100, "%")
    image <- sprintf("linear-gradient(90deg, transparent %1$s, %2$s %1$s)", position, fill)
  }
  list(
    backgroundImage = image,
    backgroundSize = paste("100%", height),
    backgroundRepeat = "no-repeat",
    backgroundPosition = "center",
    color = color
  )
}
```


```{r define-genome}
hg19 <- BSgenome.Hsapiens.UCSC.hg19
sq19 <- seqinfo(hg19)
```

```{r load-rna}
all_gr <- here::here("data", "annotations", "all_gr.rds") %>%
  read_rds() %>% 
  lapply(
    function(x) {
      seqinfo(x) <- sq19
      x
    }
  )
counts <- here::here("data", "rnaseq", "counts.out.gz") %>%
  read_tsv(skip = 1) %>%
  dplyr::select(Geneid, ends_with("bam")) %>%
  dplyr::rename(gene_id = Geneid)
detected <- all_gr$gene %>%
  as_tibble() %>%
  distinct(gene_id, gene_name) %>%
  left_join(counts) %>%
  pivot_longer(
    cols = ends_with("bam"), names_to = "sample", values_to = "counts"
  ) %>%
  mutate(detected = counts > 0) %>%
  group_by(gene_id, gene_name) %>%
  summarise(detected = mean(detected) > 0.25, .groups = "drop") %>%
  dplyr::filter(detected) %>%
  dplyr::select(starts_with("gene"))
```


Pre-existing RNA-seq data was again loaded and the set of `r comma(nrow(detected))` detected genes was defined as those with $\geq 1$ aligned read in at least 25% of samples.
This dataset represented the DHT response in MDA-MB-453 cells, however, given the existence of both Vehicle and DHT conditions, represents a good representative measure of the polyadenylated components of the transcriptome.


```{r load-features}
features <- here::here("data", "h3k27ac") %>%
  list.files(full.names = TRUE, pattern = "bed$") %>%
  sapply(import.bed, seqinfo = sq19) %>%
  lapply(granges) %>%
  setNames(basename(names(.))) %>%
  setNames(str_remove_all(names(.), "s.bed")) %>%
  GRangesList() %>%
  unlist() %>%
  names_to_column("feature") %>%
  sort()
```

The set of H3K27ac-defined features was again used as a marker of regulatory activity, with `r pander(comma(as.integer(table(features$feature))))` regions defined as enhancers or promoters respectively.


```{r load-rime}
uniprot_to_ensg <- here::here("data", "external", "uniprot_to_ensg.txt.gz") %>% 
  read_tsv(col_names = c("uniprot_id", "protein", "gene_id")) %>% 
  mutate(
    gene_id = lapply(gene_id, str_split, pattern = "; ") %>% 
      lapply(unlist)
  ) %>% 
  unnest(gene_id) %>% 
  mutate(gene_id = str_replace_all(gene_id, "(ENSG[0-9]+)\\..*", "\\1")) %>% 
  left_join(
    as_tibble(all_gr$gene) %>% 
      dplyr::select(gene_id, gene_name)
  ) %>% 
  dplyr::filter(!is.na(gene_id)) %>% 
  dplyr::select(-protein)
rime_detected <- here::here("data", "external", "MDA-MB-453_RIME_processed.xlsx") %>% 
  read_excel() %>% 
  dplyr::select(-starts_with("AVG"), -ends_with("Coverage2"), -contains("delta")) %>% 
  pivot_longer(cols = contains("_")) %>% 
  separate(name, c("sample", "measure"), sep = "_", extra = "merge") %>% 
  mutate(
    IP = str_remove(sample, "[0-9]$"),
    replicate = str_extract(sample, "[0-9]$"),
    measure = paste(measure, IP, sep = "_")
  ) %>% 
  pivot_wider(
    names_from = "measure", values_from = "value", 
    id_cols = c("Accession", "Description", "Protein", "replicate")
  ) %>% 
  mutate(
    logfc_cov = log2(Coverage_AR + 1) - log2(Coverage_IgG + 1),
    logfc_peptides = log2(Unique_peptides_AR + 1) - log2(Unique_peptides_IgG + 1)
  ) %>% 
  dplyr::filter(logfc_cov > 0) %>% 
  group_by(Accession, Description, Protein) %>% 
  summarise(
    n_detected = dplyr::n(),
    mn_logfc_coverage = mean(logfc_cov),
    mn_coverage_AR = mean(Coverage_AR),
    mn_coverage_IgG = mean(Coverage_IgG),
    mn_peptides_AR = mean(Unique_peptides_AR),
    mn_peptides_IgG = mean(Unique_peptides_IgG),
    .groups = "drop"
  ) %>% 
  dplyr::filter(n_detected > 1) %>% 
  arrange(desc(mn_logfc_coverage)) %>% 
  left_join(uniprot_to_ensg, by = c("Accession" = "uniprot_id")) %>% 
  dplyr::select(
    Accession, Description, Protein, starts_with("gene"), everything()
  ) %>% 
  dplyr::filter(!is.na(gene_name))
```

AR-RIME data for MDA-MB-453 cells was also used, with protein to transcript ID mappings performed using the mappings obtained using the following `bash` command
```
curl https://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/idmapping/by_organism/HUMAN_9606_idmapping_selected.tab.gz |\
  zcat |\
  cut -f1-2,19 |\
  gzip -c > data/external/uniprot_to_ensg.txt.gz
```

For RIME data, the criteria for detection within a sample was simply defined as the coverage being greater when using AR as the IP-target, than when using IgG as the IP-target.
Under this approach, `r nrow(rime_detected)` proteins were considered as detected in $>2$ of 3 samples and were retained for inclusion with motif analysis.
When further comparing to the RNA-Seq data only `r collapseGenes(setdiff(rime_detected$gene_name, detected$gene_name))` was found in the RIME data without supporting RNA evidence.
Any proteins with no gene_name associated (i.e. only an Ensembl Gene ID) were also excluded.


```{r load-motif-db}
meme_db <- here::here("data", "external", "JASPAR2022_CORE_Hsapiens.meme") %>%
  read_meme() %>%
  to_df() %>%
  as_tibble() %>%
  mutate(
    across(everything(), vctrs::vec_proxy),
    altname = str_remove(altname, "MA[0-9]+\\.[0-9]*\\."),
    organism = "Homo sapiens"
  )
meme_altname <- setNames(meme_db$altname, meme_db$name)
meme_db_detected <- meme_db %>%
  mutate(
    gene_name = str_split(altname, pattern = "::", n = 2)
  ) %>%
  dplyr::filter(
    vapply(
      gene_name,
      function(x) all(x %in% detected$gene_name),
      logical(1)
    )
  )
meme_info <- meme_db_detected %>% 
  mutate(motif_length = vapply(motif, function(x){ncol(x@motif)}, integer(1))) %>% 
  dplyr::select(name, altname, icscore, motif_length) 
```


The motif database was initially defined as all `r nrow(meme_db)` CORE motifs found in *Homo sapiens*, as contained in [JASPAR2022](https://jaspar2022.genereg.net/).
In order to minimise irrelevant results, only the `r nrow(meme_db_detected)` motifs associated with a detected gene, using presence within the RNA-Seq dataset as evidence of detection.

```{r load-peaks}
dht_consensus <- read_rds(
  here::here("output/dht_consensus.rds")
)
seqinfo(dht_consensus) <- sq19
dht_peaks <- here::here("data", "peaks") %>%
  list.files(recursive = TRUE, pattern = "oracle", full.names = TRUE) %>%
  sapply(read_rds, simplify = FALSE) %>%
  lapply(function(x) x[["DHT"]]) %>%
  lapply(setNames, nm = c()) %>%
  setNames(str_extract_all(names(.), "AR|FOXA1|GATA3|TFAP2B")) %>% 
  lapply(
    function(x) {
      seqinfo(x) <- sq19
      subsetByOverlaps(x, dht_consensus)
    }
  )
targets <- names(dht_peaks)
```


# Analysis of Motif Enrichment (AME)

```{r grl-by-targets}
grl_by_targets <- dht_consensus %>% 
  as_tibble() %>% 
  pivot_longer(
    all_of(targets), names_to = "target", values_to = "detected"
  ) %>% 
  dplyr::filter(detected) %>% 
  group_by(range, H3K27ac) %>% 
  summarise(
    targets = paste(sort(target), collapse = " "),
    .groups = "drop"
  ) %>% 
  colToRanges(var = "range", seqinfo = sq19) %>% 
  sort() %>% 
  splitAsList(.$targets)
dht_consensus$targets <- bestOverlap(dht_consensus, grl_by_targets)
```


Union ranges were defined by grouping by the combination of detected `r glue_collapse(targets, sep = ", ", last = " and ")` binding, then sequences obtained corresponding to each union range.

```{r tbl-union-ranges}
grl_by_targets %>% 
  lapply(
    function(x) {
      tibble(
        `N Sites` = length(x), 
        `% H3K27ac` = mean(x$H3K27ac),
        `Median Width` = floor(median(width(x)))
      )
    }
  ) %>% 
  bind_rows(.id = "Targets") %>% 
  mutate(
    `N Targets` = str_count(Targets, " ") + 1,
    `% H3K27ac` = percent(`% H3K27ac`, 0.1)
  ) %>% 
  dplyr::select(contains("Targets"), everything()) %>% 
  arrange(`N Targets`, Targets) %>% 
  pander(
    justify = "lrrrr",
    caption = "Summary of union ranges and which combinations of the targets were detected."
  )
```


```{r run-ame}
seq_by_targets <- grl_by_targets %>% 
  lapply(function(x) split(x, x$H3K27ac)) %>% 
  lapply(GRangesList) %>% 
  lapply(get_sequence, genome = hg19)

## All bound sites
rds_ame_by_targets <- here::here("output", "ame_by_targets.rds")
if (!file.exists(rds_ame_by_targets)) {
  ame_by_targets <- seq_by_targets %>% 
    lapply(unlist) %>% 
    mclapply(
      runAme, 
      database = to_list(meme_db_detected), 
      mc.cores = min(n_cores, length(.))
    )
  nulls <- ame_by_targets %>% 
    vapply(is, logical(1), "data.frame") %>% 
    not() %>% 
    which() %>% 
    names()
  if (length(nulls) > 0) {
    ame_by_targets[nulls] <- seq_by_targets[nulls] %>% 
      lapply(unlist) %>% 
      lapply(
        runAme, 
        database = to_list(meme_db_detected)
      )
  }
  write_rds(ame_by_targets, rds_ame_by_targets, compress = "gz")
} else {
  ame_by_targets <- read_rds(rds_ame_by_targets)
}

## Only sites marked as active by H3K27ac
rds_ame_by_targets_active <- here::here("output", "ame_by_targets_active.rds")
if (!file.exists(rds_ame_by_targets_active)) {
  ame_by_targets_active <- seq_by_targets %>% 
    lapply(function(x) x[["TRUE"]]) %>% 
    mclapply(
      runAme, 
      database = to_list(meme_db_detected), 
      mc.cores = min(n_cores, length(.))
    )
  nulls <- ame_by_targets_active %>% 
    vapply(is, logical(1), "data.frame") %>% 
    not() %>% 
    which() %>% 
    names()
  if (length(nulls) > 0) {
    ame_by_targets_active[nulls] <- seq_by_targets[nulls] %>% 
      lapply(function(x) x[["TRUE"]]) %>% 
      lapply(
        runAme, 
        database = to_list(meme_db_detected)
      )
  }
  write_rds(ame_by_targets_active, rds_ame_by_targets_active, compress = "gz")
} else {
  ame_by_targets_active <- read_rds(rds_ame_by_targets_active)
}

## Compare the active and inactive sites for each set of targets
rds_ame_active_vs_inactive <- here::here("output", "ame_active_vs_inactive.rds")
if (!file.exists(rds_ame_active_vs_inactive)) {
  ame_active_vs_inactive <- seq_by_targets %>% 
    mclapply(
      function(x) {
        runAme(x[["TRUE"]], x[["FALSE"]], database = to_list(meme_db_detected))
      },
      mc.cores = min(n_cores, length(.))
    )
  nulls <- ame_active_vs_inactive %>% 
    vapply(is, logical(1), "data.frame") %>% 
    not() %>% 
    which() %>% 
    names()
  if (length(nulls) > 0) {
    ame_active_vs_inactive[nulls] <- seq_by_targets[nulls] %>% 
      lapply(
        function(x) {
          runAme(x[["TRUE"]], x[["FALSE"]], database = to_list(meme_db_detected))
        }
      )
  }
  write_rds(ame_active_vs_inactive, rds_ame_active_vs_inactive, compress = "gz")
} else {
  ame_active_vs_inactive <- read_rds(rds_ame_active_vs_inactive)
}

## Compare the inactive and active sites for each set of targets
rds_ame_inactive_vs_active <- here::here("output", "ame_inactive_vs_active.rds")
if (!file.exists(rds_ame_inactive_vs_active)) {
  ame_inactive_vs_active <- seq_by_targets %>% 
    mclapply(
      function(x) {
        runAme(x[["FALSE"]], x[["TRUE"]], database = to_list(meme_db_detected))
      },
      mc.cores = min(n_cores, length(.))
    )
  nulls <- ame_inactive_vs_active %>% 
    vapply(is, logical(1), "data.frame") %>% 
    not() %>% 
    which() %>% 
    names()
  if (length(nulls) > 0) {
    ame_inactive_vs_active[nulls] <- seq_by_targets[nulls] %>% 
      lapply(
        function(x) {
          runAme(x[["FALSE"]], x[["TRUE"]], database = to_list(meme_db_detected))
        }
      )
  }
  write_rds(ame_inactive_vs_active, rds_ame_inactive_vs_active, compress = "gz")
} else {
  ame_inactive_vs_active <- read_rds(rds_ame_inactive_vs_active)
}
```


[AME from the meme-suite](https://meme-suite.org/meme/doc/ame.html?man_type=web) was then run on each sequence to detect the presence of motifs associated with each target within each set of sequences.
Under this approach the optimal score is estimated for considering a PWM match to be a true positive.
Matches above this value are classed as true positives, whilst those below are considered as false positives.
Significance is determined using using Fisher's Exact Test against the set of control sequences, which are either shuffled input sequences, or a discrete set provided.
Adjusted p-values are determined within each motif based on resampling, with the e-values representing a Bonferroni adjustment of the within-motif adjusted p-values.

Motif enrichment inevitably produces hits to highly-related motifs
For unsupervised enrichment, the following strategy was used to aid interpretation

1. Find enriched motifs
2. Calculate similarity of enriched motifs as a correlation matrix
    + Reduce correlations < $\rho$ to zero, commonly using $\rho=$ 0.8.
    + Converted to a distance matrix for clustering
3. Cluster the motifs using `hclust` and a height of 0.9
4. Report the results for all motifs within a cluster as a single result

## Enrichment Of ChIP Target Motifs 

```{r plot-motifs, fig.cap = "All motifs matching the set of ChIP targets. AR, FOXA1 and GATA3 are not aligned with each other, whilst all TFAP2B motifs are shown using the best alignment to each other."}
plot_grid(
  plot_grid(
    plotlist = lapply(
      meme_db_detected %>% 
        dplyr::filter(altname %in% targets) %>% 
        arrange(altname) %>% 
        dplyr::filter(altname != "TFAP2B") %>% 
        to_list() %>% 
        lapply(
          function(x){
            x@name <- paste0(
              x@name, " (", x@altname, ")"
            )
            x
          }
        ),
      function(x) {
        view_motifs(x) +
          ggtitle(x@name) +
          theme(plot.title = element_text(hjust = 0.5, size = 11))
      }
    ),
    ncol = 1
  ),
  plot_grid(
    meme_db_detected %>% 
      dplyr::filter(altname == "TFAP2B") %>% 
      to_list() %>% 
      lapply(
          function(x){
            x@name <- paste0(
              x@name, " (", x@altname, ")"
            )
            x
          }
      ) %>% 
      view_motifs() +
      theme(plot.title = element_text(hjust = 0.5, size = 9))
  ),
  ncol = 2
)
```


### All Sequences

AME was used to assess enrichment of motifs within the sequences derived from each combination of ChIP targets.
Enrichment was considered relative to *shuffled sequences* as the control set, with AME only returning matches considered to be statistically significant.

```{r ame-heatmap, fig.cap = "Estimated percentage of true positive matches to all binding motifs associated with each ChIP target within all sequences. Only motifs considered to be enriched are shown, such that missing values are able to be confidently assumed as not enriched for the motif. All provide supporting evidence for direct binding of each target to it's motif."}
ame_by_targets %>% 
  bind_rows(.id = "targets") %>% 
  dplyr::filter(motif_alt_id %in% globalenv()$targets) %>% 
  mutate(
    motif_id = as.factor(motif_id),
    targets = as.factor(targets),
    TFAP2B = ifelse(
      str_detect(targets, "TFAP2B"), "TFAP2B", "No TFAP2B"
    )
  ) %>% 
  plot_ame_heatmap(
    id = motif_id,
    value = tp_percent,
    group = fct_rev(targets)
  ) +
  geom_label(
    aes(label = percent(tp_percent/100, 0.1)),
    size = 3.5,
    show.legend = FALSE
  ) +
  facet_grid(TFAP2B ~ motif_alt_id, scales = "free", space = "free") +
  scale_x_discrete(expand = expansion(c(0, 0))) +
  scale_y_discrete(expand = expansion(c(0, 0))) +
  labs(
    x = "Detected Motif",
    y = "Detected ChIP Targets",
    fill = expression(widehat("TP%"))
  ) +
  theme(
    axis.text.x = element_text(angle = 0, hjust = 0.5, size = 10),
    axis.text.y = element_text(angle = 0, hjust = 1, size = 10),
    panel.grid = element_blank()
  )
```


### H3K27ac-Associated Sequences

Restricting the test for motif enrichment to all combinations of ChIP targets which overlap an active H3K27ac peak, similar patterns were observed.
Of possible note is reduced enrichment for TFAP2$\beta$ motifs amongst the sites where TFAP2$\beta$ was not observed.

The set of H3K27-associated peaks where AR, GATA3 and TFAP2$\beta$ were found in combination were notably enriched for one specific TFAP2$\beta$ motif (MA0812) to the exclusion of others suggesting that FOXA1 independent formation of the regulatory complex may have a distinct mechanism.
In conjunction with the increased occurrence of this motif, this set of peaks also contained the lowest enrichment for AR motifs amongst the AR-associated peaks.

Another clear association between the presence of FOXA1 and possible increased enrichment for the MA0811.1 TFAP2$\beta$ motif may also be of interest.

```{r ame-heatmap-active, fig.cap = "Estimated percentage of true positive matches to all binding motifs associated with each ChIP target, restricting sequences to **those marked by overlapping H3K27ac signal**. Only motif considered to be enriched are shown, such that missing values are able to be confidently assumed as not enriched for the motif. All provide supporting evidence for direct binding of each target to it's motif. Interestingly FOXA1 in combination with TFAP2B appears to favour the motif MA0811.1, whilst AR and GATA3 in combination with TFAP2B appears to favour MA0812.1"}
ame_by_targets_active %>% 
  bind_rows(.id = "targets") %>% 
  dplyr::filter(motif_alt_id %in% globalenv()$targets) %>% 
  mutate(
    motif_id = as.factor(motif_id),
    targets = as.factor(targets),
    TFAP2B = ifelse(
      str_detect(targets, "TFAP2B"), "TFAP2B", "No TFAP2B"
    )
  ) %>% 
  plot_ame_heatmap(
    id = motif_id,
    value = tp_percent,
    group = fct_rev(targets)
  ) +
  geom_label(
    aes(label = percent(tp_percent/100, 0.1)),
    size = 3.5,
    show.legend = FALSE
  ) +
  facet_grid(TFAP2B ~ motif_alt_id, scales = "free", space = "free") +
  scale_x_discrete(expand = expansion(c(0, 0))) +
  scale_y_discrete(expand = expansion(c(0, 0))) +
  labs(
    x = "Detected Motif",
    y = "Detected ChIP Targets",
    fill = expression(widehat("TP%"))
  ) +
  theme(
    axis.text.x = element_text(angle = 0, hjust = 0.5, size = 10),
    axis.text.y = element_text(angle = 0, hjust = 1, size = 10),
    panel.grid = element_blank()
  )
```

## Enrichment of Other Motifs

Motifs associated with any detected transcription factor besides the ChIP targets were also assessed for enrichment.
Motifs which matched GATA[0-9], Forkhead (FOX[A-Z][0-9]) or TFAP2[A-Z] were omitted due to the high-level of motif redundancy and these being likely to represent the ChIP target, instead of an additional protein.
Given that an extremely large number of motifs were found, only those found in >50% of sites in at least one group of ChIP targets were retained

### All Sequences

The set of motifs associated with detectable genes (using RNA) was tested for enrichment against all sequences from all combinations of ChIP target binding, using shuffled sequences as controls.
Given that `r length(unique(bind_rows(ame_by_targets)$motif_id))` motifs  were considered to be enriched in at least one set of peaks, the plot below only shows those motifs which were found in >50% of sequences in one or more sets of peaks.

The [NFIX binding motif](https://jaspar.genereg.net/matrix/MA0671.1/) was strongly associated with most sets of peaks which were not bound by TFAP2$\beta$, particularly with FOXA1 in the absence of TFAP2$\beta$.
However, a very strong association with all four bound targets was also observed.
Given the short (6nt) nature of this motif, this may be artefactual.

The TRPS1 motif was also found to be strongly enriched where GATA3 was bound, however, this motif bears a very strong resemblance to the GATA3 motif.


```{r motif-gata3-trps1, fig.height=4, fig.width=8, fig.cap = "Comparison between GATA3 and TRPS1 motifs"}
meme_db %>% 
  dplyr::filter(str_detect(altname, "GATA3|TRPS1")) %>% 
  to_list() %>%
  lapply(function(x) {x@name <- glue("{x@name}\n({x@altname})"); x}) %>% 
  view_motifs(names.pos = "right")
```


```{r plot-ame-other-all, fig.cap = "Enriched motifs when analysing all sequences associated with one or more ChIP targets. Motifs are only shown if found in >50% of sequences from at least one set of peaks."}
x <- ame_by_targets %>% 
  bind_rows(.id = "targets") %>% 
  dplyr::filter(
    !motif_alt_id %in% targets,
    !str_detect(motif_alt_id, "(GATA[0-9]|FOX[A-Z][0-9]|TFAP2[A-Z])"),
    tp_percent > 50
  ) %>% 
  distinct(motif_id) %>% 
  pull("motif_id")
cormat <- meme_db %>% 
  dplyr::filter(name %in% x) %>% 
  to_list() %>% 
  compare_motifs()
cormat[cormat < 0.9] <- 0
cormat[is.na(cormat)] <- 0
cl <- as.dist(1 - cormat) %>% 
  hclust() %>% 
  cutree(h= 0.9)
ame_by_targets %>% 
  bind_rows(.id = "targets") %>% 
  dplyr::filter(motif_id %in% names(cl)) %>% 
  mutate(
    cluster = cl[motif_id],
    motif_id = as.factor(motif_id),
    targets = as.factor(targets),
    TFAP2B = ifelse(
      str_detect(targets, "TFAP2B"), "TFAP2B", "No TFAP2B"
    )
  ) %>% 
  group_by(cluster) %>% 
  mutate(
    best_tp = max(tp_percent)
  ) %>% 
  ungroup() %>% 
  group_by(motif_id) %>% 
  dplyr::filter(any(tp_percent == best_tp)) %>% 
  ungroup() %>% 
  plot_ame_heatmap(
    id = motif_id,
    value = tp_percent,
    group = fct_rev(targets)
  ) +
  geom_label(
    aes(label = percent(tp_percent/100, 0.1)),
    size = 3.5,
    show.legend = FALSE
  ) +
  facet_grid(TFAP2B ~ motif_alt_id, scales = "free", space = "free") +
  scale_x_discrete(expand = expansion(c(0, 0))) +
  scale_y_discrete(expand = expansion(c(0, 0))) +
  labs(
    x = "Detected Motif",
    y = "Detected ChIP Targets",
    fill = expression(widehat("TP%"))
  ) +
  theme(
    axis.text.x = element_text(angle = 0, hjust = 0.5, size = 10),
    axis.text.y = element_text(angle = 0, hjust = 1, size = 10),
    panel.grid = element_blank()
  )
```

#### H3K27ac Active Sequences

When restricting the peaks to being any ChIP-target binding peak associated with H3K27ac significantly more motifs were considered to be enriched.

Notably:

- [NFIX](https://jaspar.genereg.net/matrix/MA0671.1/) enrichment was mostly lost, especially in the set of peaks associated with all four targets. Significance was retained for FOXA1 and GATA3 in combination, but only in the absence of TFAP2$\beta$
- [HIC2](https://jaspar.genereg.net/matrix/MA0738.1/) was enriched in all combinations where AR and FOXA1 were bound in combination. However, HIC2 was not found in combination with AR in the RIME dataset.

No other striking patterns were observed.

```{r plot-ame-active-all, fig.cap = "Enriched motifs when analysing all sequences associated with one or more ChIP targets **and which overlap an H3K27ac peak**. Motifs are only shown if found in >50% of sequences from at least one set of peaks."}
x <- ame_by_targets_active %>% 
  bind_rows(.id = "targets") %>% 
  dplyr::filter(
    !motif_alt_id %in% targets,
    !str_detect(motif_alt_id, "(GATA[0-9]|FOX[A-Z][0-9]|TFAP2[A-Z])"),
    tp_percent > 50
  ) %>% 
  distinct(motif_id) %>% 
  pull("motif_id")
cormat <- meme_db %>% 
  dplyr::filter(name %in% x) %>% 
  to_list() %>% 
  compare_motifs()
cormat[cormat < 0.9] <- 0
cormat[is.na(cormat)] <- 0
cl <- as.dist(1 - cormat) %>% 
  hclust() %>% 
  cutree(h= 0.9)
ame_by_targets_active %>% 
  bind_rows(.id = "targets") %>% 
  dplyr::filter(motif_id %in% names(cl)) %>% 
  mutate(
    cluster = cl[motif_id],
    motif_id = as.factor(motif_id),
    targets = as.factor(targets),
    TFAP2B = ifelse(
      str_detect(targets, "TFAP2B"), "TFAP2B", "No TFAP2B"
    )
  ) %>% 
  group_by(cluster) %>% 
  mutate(
    best_tp = max(tp_percent)
  ) %>% 
  ungroup() %>% 
  group_by(motif_id) %>% 
  dplyr::filter(any(tp_percent == best_tp)) %>% 
  ungroup() %>% 
  plot_ame_heatmap(
    id = motif_id,
    value = tp_percent,
    group = fct_rev(targets)
  ) +
  geom_label(
    aes(label = percent(tp_percent/100, 0.1)),
    size = 3,
    show.legend = FALSE
  ) +
  facet_grid(TFAP2B ~ motif_alt_id, scales = "free", space = "free") +
  scale_x_discrete(expand = expansion(c(0, 0))) +
  scale_y_discrete(expand = expansion(c(0, 0))) +
  labs(
    x = "Detected Motif",
    y = "Detected ChIP Targets",
    fill = expression(widehat("TP%"))
  ) +
  theme(
    axis.text.x = element_text(angle = 0, hjust = 0.5, size = 8),
    axis.text.y = element_text(angle = 0, hjust = 1, size = 8),
    panel.grid = element_blank()
  )
```


### H3K27ac Active Vs Inactive

In order to ascertain whether any co-factors are playing a role in active vs inactive regulatory regions, the set of sequences associated with binding of all four targets were compared to each other, based on H3K27ac overlap as the marker of an active regulatory region.
Enriched motifs were those with an $E$-value < 0.01 and with a True Positive match in > 10% of sequences.

Due to the high-levels of similarity between identified motifs, enriched motifs were clustered by:

1. Finding the pairwise correlations
2. Setting all pairwise correlations < 0.8 to zero
3. Using hierarchical clustering to cluster those with retained correlations

For each motif cluster the values used for visualisation were the lowest $E$-value within the cluster and the high estimate of True Positive matches.

```{r plot-motifs-active-vs-inactive, fig.height=12, fig.cap = "Motifs found enriched when comparing sequences from peaks overlapping an active H3K27ac regulatory region, with those from peaks not overlapping an H3K27ac-marked region. Motifs were clustered as described in the text, with any TFs detected as bound to AR in the RIME dataset indicated with an asterisk. Interestingly, TFAP2B motifs appeared to be enriched in this set of sequences."}
x <- ame_active_vs_inactive$`AR FOXA1 GATA3 TFAP2B` %>%
  dplyr::filter(
    evalue < 0.01,
    tp_percent > 10
  ) %>%
  distinct(motif_id) %>% 
  pull("motif_id")
cormat <- meme_db %>% 
  dplyr::filter(name %in% x) %>% 
  to_list() %>% 
  compare_motifs()
cormat[cormat < 0.80] <- 0
cormat[is.na(cormat)] <- 0
cl_active_vs_inactive <- as.dist(1 - cormat) %>% 
  hclust() %>% 
  cutree(h= 0.9)
ame_active_vs_inactive$`AR FOXA1 GATA3 TFAP2B` %>% 
  dplyr::filter(motif_id %in% names(cl_active_vs_inactive)) %>% 
  mutate(
    cluster = cl_active_vs_inactive[motif_id], 
    in_rime = motif_alt_id %in% rime_detected$gene_name,
    motif_alt_id = ifelse(in_rime, paste0(motif_alt_id, "*"), motif_alt_id)
  ) %>% 
  group_by(cluster) %>% 
  summarise(
    tp_percent = max(tp_percent),
    p = min(evalue),
    id = motif_alt_id %>% 
      unique %>% 
      sort() %>% 
      paste(collapse = "/")
  ) %>% 
  arrange(p) %>% 
  mutate(
    id = str_trunc(id, width = 40) %>% fct_inorder()
  ) %>% 
  ggplot(
    aes(tp_percent, fct_rev(id), fill = -log10(p))
  ) +
  geom_col() +
  scale_x_continuous(expand = expansion(c(0, 0.05))) +
  scale_fill_viridis_c() +
  labs(
    x = "True Positive Matches (%)",
    y = "Motif Cluster"
  )
```

### H3K27ac Inctive Vs Active

The same approach was then take for sequences associated with *inactive* regulatory regions, by comparing these sequences to those from *active* regulatory regions.

```{r plot-motifs-inactive-vs-active, fig.height = 6, fig.cap = "Motifs found enriched when comparing sequences from peaks overlapping an inactive H3K27ac regulatory region, with those from peaks overlapping an H3K27ac-marked region. Motifs were clustered as described in the text, with any TFs detected as bound to AR in the RIME dataset indicated with an asterisk. Both FOXA1 and GATA3 motifs appeared to be enriched in this set of sequences."}
x <- ame_inactive_vs_active$`AR FOXA1 GATA3 TFAP2B` %>%
  dplyr::filter(evalue < 0.01, tp_percent > 10) %>%
  distinct(motif_id) %>% 
  pull("motif_id")
cormat <- meme_db %>% 
  dplyr::filter(name %in% x) %>% 
  to_list() %>% 
  compare_motifs()
cormat[cormat < 0.8] <- 0
cormat[is.na(cormat)] <- 0
cl_inactive_vs_active <- as.dist(1 - cormat) %>% 
  hclust() %>% 
  cutree(h= 0.9)
ame_inactive_vs_active$`AR FOXA1 GATA3 TFAP2B` %>% 
  dplyr::filter(motif_id %in% names(cl_inactive_vs_active)) %>% 
  mutate(
    cluster = cl_inactive_vs_active[motif_id], 
    in_rime = motif_alt_id %in% rime_detected$gene_name,
    motif_alt_id = ifelse(in_rime, paste0(motif_alt_id, "*"), motif_alt_id)
  ) %>% 
  group_by(cluster) %>% 
  summarise(
    tp_percent = max(tp_percent),
    p = min(evalue),
    id = motif_alt_id %>% 
      unique %>% 
      sort() %>% 
      paste(collapse = "/")
  ) %>% 
  arrange(p) %>% 
  mutate(
    id = str_trunc(id, width = 40) %>% fct_inorder()
  ) %>% 
  ggplot(
    aes(tp_percent, fct_rev(id), fill = -log10(p))
  ) +
  geom_col() +
  scale_x_continuous(expand = expansion(c(0, 0.05))) +
  scale_fill_viridis_c() +
  labs(
    x = "True Positive Matches (%)",
    y = "Motif Cluster"
  )
```

## Further Investigations

Given the unexpected result that TFAP2$\beta$ motifs were enriched in the *H3K27ac marked peaks* (i.e. active regulatory regions), whilst FOXA1 and GATA motifs were enriched in the peaks **not marked** by H3K27ac (i.e. inactive regulatory regions), the co-occurrence of these motifs was explored across both sets of peaks as a predictive tool for an active regulatory region.
Sequences were checked for the simple presence of each motif, using matches with a score > 60% of the maximum possible score as a positive match, to be as inclusive as possible.
For all sequences, only *the highest scoring match to the consensus motif* was retained.
For comparability across motifs, motif match scores were normalised to Z-scores.

```{r glm-all4-h3k27ac, results='hide'}
inv.logit <- binomial()$linkinv
min_score <- percent(0.6, accuracy = 1)
all4_motifs <- meme_db_detected %>%
  dplyr::filter(altname %in% targets) %>% 
  to_list() %>% 
  mclapply(
    getBestMatch, 
    subject = seq_by_targets$`AR FOXA1 GATA3 TFAP2B` %>% 
      unlist() %>% 
      as("DNAStringSet") %>% 
      as("Views") %>% 
      setNames(
        seq_by_targets$`AR FOXA1 GATA3 TFAP2B` %>% 
          unlist() %>% 
          names() %>% 
          str_remove_all("^(TRUE|FALSE)\\.")
      ),
    min.score = min_score,
    mc.cores = length(targets)
  ) %>% 
  setNames(
    dplyr::filter(meme_db_detected, altname %in% targets)$name
  ) %>% 
  bind_rows(.id = "name") %>% 
  left_join(
    dplyr::select(meme_info, name, altname, motif_length),
    by = "name"
  ) %>% 
  dplyr::rename(range = view_name) %>% 
  full_join(
    grl_by_targets$`AR FOXA1 GATA3 TFAP2B` %>% 
      as_tibble(),
    by = "range" 
  ) 
glm_h3K27ac <- all4_motifs %>% 
  group_by(name) %>% 
  mutate(
    score = (score - mean(score, na.rm = TRUE)) / sd(score, na.rm = TRUE)
  ) %>% 
  pivot_wider(
    names_from = "name", values_from = "score", id_cols = c("range", "H3K27ac")
  ) %>% 
  dplyr::filter(
    !is.na(MA0811.1), !is.na(MA0812.1), !is.na(MA0813.1)
  ) %>% 
  mutate(
    PC1 = cbind(MA0811.1, MA0812.1, MA0813.1) %>% 
      prcomp() %>% 
      .[["x"]] %>% 
      .[,"PC1"] %>% 
      multiply_by(-1), # This always seems to come out in the opposite direction to observed
    width = width(GRanges(range))
  ) %>% 
  with(
    glm(H3K27ac ~ width + (PC1 + MA0007.2 + MA0148.4 + MA0037.3)^4 , family = "binomial")
  ) %>% 
  step() %>% 
  broom::tidy() %>% 
  mutate(
    term = term %>% 
      str_replace_all("PC1", "TFAP2B (PC1)") %>% 
      str_replace_all("MA0007.2", "AR") %>% 
      str_replace_all("MA0148.4", "FOXA1") %>% 
      str_replace_all("MA0037.3", "GATA3"),
    fdr = p.adjust(p.value, "fdr"),
    ` ` = case_when(
      fdr > 0.1 ~ "",
      fdr > 0.05 ~ ".",
      fdr > 0.01 ~ "*",
      fdr > 0.001 ~ "**",
      TRUE ~ "***"
    )
  ) 
```

```{r plot-all4-scores, fig.height = 6, fig.cap = "Distribution of Z-Scores for matches to each motif, separated by whether the peak was associated with an H3K27ac mark. Motifs with higher scores were clearly associated with H3K27ac marks for all TFAP2B motifs, whilst some tendency towards lower scores was noted for FOXA1 and GATA3, in keeping with the above enrichment results."}
all4_motifs %>% 
  group_by(name) %>% 
  mutate(
    score = (score - mean(score, na.rm = TRUE)) / sd(score, na.rm = TRUE)
  ) %>% 
  ggplot(aes(name, score, fill = H3K27ac)) +
  geom_boxplot() +
  facet_grid(.~altname, scales = "free", space = "free") +
  scale_fill_manual(values = c("grey50", "red")) +
  labs(
    x = "Motif name", y = "Normalised Motif Match Score"
  )
```

To quantify the impact of motif presence within the sequences where all 4 ChIP targets were detected, the probability of a peak being associated with H3K27ac marks was modelled using the strength of each motif in respect to the PWM.
To avoid the high correlations between the three TFAP2$\beta$ motifs, these were combined using PC1 returned from PCA across all three motifs as a representative score.
This may make the direction of the probability more difficult to interpret as a lower values on PC1 may represent a higher match across all 3 motifs.

A logistic regression model was fitted using H3K27ac status as a the binary response variable with the scores for AR (MA0007.2), FOXA1 (MA0148.4) and GATA3 (MA0037.3) motifs being combined with PC1 across all 3 TFAP2$\beta$ motifs as the predictors.
The initial model was fitted incorporating 4-way interactions as the initial model, along with peak width, with terms being removed in a step-wise manner using Akaike's Information Criterion (AIC).


```{r tbl-glm-h3k27ac}
cp <- htmltools::tags$caption(
  glue(
    "
    Results from final logistic regression model for the probability of a peak 
    being marked by H3K27ac, based on the score of each detected motif's match 
    to the PWM, and the overall width of the peak. Whilst the width of the peak 
    clearly had a significant influence on the probability of a peak being 
    considered active, via H3K27ac, the presence of a strong TFAP2B motif also 
    had a clear influence on the likelihood of a peak being considered active.
    Conversely, the stronger matches to the core FOXA1 and GATA3 motifs 
    indicated a decreased association with H3K27ac marks, as predicted by the 
    sequence enrichment analysis.
    "
  )
)
tbl <- glm_h3K27ac %>% 
  mutate(term = str_replace_all(term, "width", "Peak Width")) %>% 
  reactable(
    columns = list(
      term = colDef(
        name = "Model Term",
        cell = function(value) str_replace_all(value, ":", " : "),
        minWidth = 150
      ),
      estimate = colDef(
        name = "Estimate", format = colFormat(digits = 3)
      ),
      std.error = colDef(show = FALSE),
      statistic = colDef(name = "Test Statistic"),
      p.value = colDef(
        show = FALSE,
        cell = function(value) {
          ifelse(
            value < 0.001,
            sprintf("%.2e", value),
            sprintf("%.3f", value)
          )
        }
      ),
      fdr = colDef(
        name = "P<sub>FDR</sub>", html = TRUE,
        cell = function(value) {
          ifelse(
            value < 0.001,
            sprintf("%.2e", value),
            sprintf("%.3f", value)
          )
        }
      )
    ),
    defaultColDef = colDef(
      format = colFormat(digits = 2)
    )
  )
div(class = "table",
  div(class = "table-header",
      div(class = "caption", cp),
      tbl
  )
)
```



# Distance Between Motifs {.tabset}

The distance between the best match to each motif was then assessed for the set of peaks for which *all four targets were detected*, separated by H3K27ac status.


```{r fig-compare-distance, fig.width=8, fig.height=10, fig.cap= "Distances between motifs for the best match for each motif. For comparisons to TFAP2B motifs, the nearest of the three was selected. Whilst the median distances between AR, FOXA1 and GATA3 motifs tended to be between 160 and 200bp, the median distances between TFAP2B motifs and the the other motifs tended to be around 80 to 100bp, suggesting that TFAP2B may bind between two of the other three targets in many complexes. A trend towards motifs being further aart within the active regulatory regions was moted, but a simple explanation for this remains elusive."}
pwm_lengths <- structure(meme_info$motif_length, names = meme_info$name)
all4_motifs %>% 
  arrange(view) %>% 
  ## The best strategy is to find the distance between centres, then add half
  ## the width of each motif
  mutate(match_centre = (start_in_view + end_in_view) / 2) %>% 
  pivot_wider(
    names_from = "name", values_from = "match_centre", id_cols = c("range", "H3K27ac")
  ) %>% 
  mutate(
    ## AR
    MA0007.2_MA0037.3 = abs(MA0007.2  - MA0037.3 ) - 0.5*(sum(pwm_lengths[c("MA0007.2", "MA0037.3")])),
    MA0007.2_MA0148.4 = abs(MA0007.2  - MA0148.4 ) - 0.5*(sum(pwm_lengths[c("MA0007.2", "MA0148.4")])),
    MA0007.2_MA0811.1 = abs(MA0007.2  - MA0811.1 ) - 0.5*(sum(pwm_lengths[c("MA0007.2", "MA0811.1")])),
    MA0007.2_MA0812.1 = abs(MA0007.2  - MA0812.1 ) - 0.5*(sum(pwm_lengths[c("MA0007.2", "MA0812.1")])),
    MA0007.2_MA0813.1 = abs(MA0007.2  - MA0813.1 ) - 0.5*(sum(pwm_lengths[c("MA0007.2", "MA0813.1")])),
    ## FOXA1
    MA0148.4_MA0037.3 = abs(MA0148.4  - MA0037.3 ) - 0.5*(sum(pwm_lengths[c("MA0148.4", "MA0037.3")])),
    MA0148.4_MA0811.1 = abs(MA0148.4  - MA0811.1 ) - 0.5*(sum(pwm_lengths[c("MA0148.4", "MA0811.1")])),
    MA0148.4_MA0812.1 = abs(MA0148.4  - MA0812.1 ) - 0.5*(sum(pwm_lengths[c("MA0148.4", "MA0812.1")])),
    MA0148.4_MA0813.1 = abs(MA0148.4  - MA0813.1 ) - 0.5*(sum(pwm_lengths[c("MA0148.4", "MA0813.1")])),
    ## GATA3
    MA0037.3_MA0811.1 = abs(MA0037.3  - MA0811.1 ) - 0.5*(sum(pwm_lengths[c("MA0037.3", "MA0811.1")])),
    MA0037.3_MA0812.1 = abs(MA0037.3  - MA0812.1 ) - 0.5*(sum(pwm_lengths[c("MA0037.3", "MA0812.1")])),
    MA0037.3_MA0813.1 = abs(MA0037.3  - MA0813.1 ) - 0.5*(sum(pwm_lengths[c("MA0037.3", "MA0813.1")])),
  ) %>% 
  dplyr::select(range, H3K27ac, contains("_")) %>% 
  pivot_longer(starts_with("MA"), names_to = "motifs", values_to = "d") %>% 
  separate(motifs, into = c("pwm_1", "pwm_2"), sep = "_") %>% 
  left_join(
    dplyr::select(meme_info, pwm_1 = name, altname_1 = altname), by = "pwm_1"
  ) %>% 
  left_join(
    dplyr::select(meme_info, pwm_2 = name, altname_2 = altname), by = "pwm_2"
  ) %>% 
  arrange(range, d) %>% 
  distinct(range, altname_1, altname_2, .keep_all = TRUE) %>% 
  unite("comparison", starts_with("altname"), sep = " - ") %>%
  mutate(
    d = ifelse(d < 0, 0, d),
    H3K27ac = ifelse(H3K27ac, "Active", "Inactive")
  ) %>% 
  ggplot(aes(d, stat(density), colour = H3K27ac)) +
  geom_density() +
  geom_vline(
    aes(xintercept = med, colour = H3K27ac),
    data = . %>% 
      group_by(comparison, H3K27ac) %>% 
      summarise(med = median(d), .groups = "drop"),
    show.legend = FALSE
  ) +
  facet_grid(comparison ~ ., scales = "free_y") +
  scale_colour_manual(values = c("red", "black")) +
  coord_cartesian(xlim = c(0, 500)) +
  theme(legend.position = "top")
```


```{r plot-motif-strad, echo=FALSE, eval=FALSE}
all4_motifs %>% 
  arrange(view) %>%
  pivot_wider(names_from = "name", values_from = "strand", id_cols = c("range", "H3K27ac")) %>% 
  group_by(H3K27ac, MA0007.2, MA0037.3, MA0148.4, MA0811.1) %>% 
  tally() %>% 
  ungroup() %>%
  group_by(H3K27ac) %>% 
  mutate(p = n / sum(n)) %>% 
  ungroup() %>%
  unite(strand, starts_with("MA"), sep = "/") %>% 
  dplyr::filter(!str_detect(strand, "\\*")) %>% 
  dplyr::mutate(
    n_plus = str_count(strand, "\\+"),
    n_neg = str_count(strand, "\\-"),
    n_same = case_when(
      n_plus == n_neg ~ 2L,
      n_plus > n_neg ~ n_plus,
      n_plus < n_neg ~ n_neg
    ) %>% as.factor(),
    strand = case_when(
      strand %in% c("+/+/+/+", "-/-/-/-") ~ "Same Strand",
      strand %in% c("-/+/+/+", "+/-/-/-") ~ "AR Diff",
      strand %in% c("-/+/-/+", "+/-/+/-") ~ "AR/GATA3 Paired  + FOXA1/TFAP2B Paired Inverse",
      strand %in% c("-/-/+/+", "+/+/-/-") ~ "AR/FOXA1 Paired  + GATA3/TFAP2B Paired Inverse",
      strand %in% c("-/+/+/-", "+/-/-/+") ~ "AR/TFAP2B Paired  + FOXA1/GATA3 Paired Inverse",
      strand %in% c("+/-/+/+", "-/+/-/-") ~ "FOXA1 Diff",
      strand %in% c("+/+/-/+", "-/-/+/-") ~ "GATA3 Diff",
      strand %in% c("+/+/+/-", "-/-/-/+") ~ "TFAP2B Diff",
    )
  ) %>% 
  group_by(strand, H3K27ac) %>% 
  summarise(p = sum(p), n_same = unique(n_same)) %>% 
  ggplot(aes(p, fct_rev(strand), fill = H3K27ac)) + 
  geom_col(position = "dodge") + 
  facet_grid(fct_rev(n_same) ~ ., scales = "free_y", space = "free_y")
```

