---
title: "Motif Analysis"
author: "Steve Pederson"
date: "2022-05-25"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r setup, echo=FALSE}
knitr::opts_chunk$set(
  message = FALSE, warning = FALSE,
  fig.width = 10, fig.height = 8
)
```


## Introduction

This step takes the peaks defined previously and 

1. Assesses motif enrichment for all ChIP targets within the peaks
2. Detect additional binding motifs 

As AR is only found bound in the presence of DHT, all peaks are taken using DHT-treatment as the reference condition.

```{r packages}
library(memes)
library(universalmotif)
library(tidyverse)
library(plyranges)
library(extraChIPs)
library(rtracklayer)
library(magrittr)
library(BSgenome.Hsapiens.UCSC.hg19)
library(scales)
library(pander)
library(glue)
library(cowplot)
library(parallel)
library(gt)
```

```{r options}
options(meme_bin = "/opt/meme/bin/")
check_meme_install()
theme_set(theme_bw())
panderOptions("big.mark", ",")
```

```{r extra-funs}
stringSetToViews <- function(x) {
  stopifnot(is(x, "XStringSet"))
  w <- unique(width(x))
  stopifnot(length(w) == 1)
  n <- length(x)
  starts <- (seq_along(x) - 1) * w + 1
  seq <- unlist(x)
  Views(subject = seq, start = starts, width = w)
}
getBestMatch <- function(x, subject, min.score = "80%", rc = TRUE, ...) {
  pwm <- NULL
  if (is.matrix(x)) pwm <- x
  if (is(x, "universalmotif")) pwm <- x@motif
  stopifnot(!is.null(pwm))
  
  stopifnot(is(subject, "XStringViews"))
  w <- unique(width(subject))
  stopifnot(length(w) == 1)
  
  matches <- matchPWM(
    pwm, subject, min.score = min.score, with.score = TRUE, ...
  )
  df <- as.data.frame(matches)
  df$score <- mcols(matches)$score
  df$id <- ceiling(df$start/ w)
  df$pos <- df$start - (df$id - 1) * w 
  df$strand <- "+"
  
  if (rc) {
    rc_matches <-  matchPWM(
      reverseComplement(pwm), subject ,
      min.score = min.score, with.score = TRUE, 
      ...
    )
    rc_df <- as.data.frame(rc_matches)
    rc_df$score <- mcols(rc_matches)$score
    rc_df$id <- ceiling(rc_df$start/ w)
    rc_df$pos <- rc_df$start - (rc_df$id - 1) * w + 1
    rc_df$strand <- "-"
    df <- bind_rows(df, rc_df)
  }
  
  df <- arrange(df, id, desc(score))
  df <- distinct(df, id, .keep_all = TRUE)
  df <- dplyr::select(df, id, pos, seq, strand)
  df$strand <- factor(df$strand, levels = c("+", "-", "*"))
  as_tibble(df)
  
}
```


```{r define-genome}
hg19 <- BSgenome.Hsapiens.UCSC.hg19
sq19 <- seqinfo(hg19)
```

```{r load-rna}
all_gr <- here::here("data", "annotations", "all_gr.rds") %>%
  read_rds() %>% 
  lapply(
    function(x) {
      seqinfo(x) <- sq19
      x
    }
  )
counts <- here::here("data", "rnaseq", "counts.out.gz") %>%
  read_tsv(skip = 1) %>%
  dplyr::select(Geneid, ends_with("bam")) %>%
  dplyr::rename(gene_id = Geneid)
detected <- all_gr$gene %>%
  as_tibble() %>%
  distinct(gene_id, gene_name) %>%
  left_join(counts) %>%
  pivot_longer(
    cols = ends_with("bam"), names_to = "sample", values_to = "counts"
  ) %>%
  mutate(detected = counts > 0) %>%
  group_by(gene_id, gene_name) %>%
  summarise(detected = mean(detected) > 0.25, .groups = "drop") %>%
  dplyr::filter(detected) %>%
  dplyr::select(starts_with("gene"))
```


Pre-existing RNA-seq data was again loaded and the set of `r comma(nrow(detected))` detected genes was defined as those with $\geq 1$ aligned read in at least 25% of samples.
This dataset represented the DHT response in MDA-MB-453 cells, however, given the existence of both Vehicle and DHT conditions, represents a good representative measure of the polyadenylated components of the transcriptome.


```{r load-features}
features <- here::here("data", "h3k27ac") %>%
  list.files(full.names = TRUE, pattern = "bed$") %>%
  sapply(import.bed, seqinfo = sq19) %>%
  lapply(granges) %>%
  setNames(basename(names(.))) %>%
  setNames(str_remove_all(names(.), "s.bed")) %>%
  GRangesList() %>%
  unlist() %>%
  names_to_column("feature") %>%
  sort()
```

The set of H3K27ac-defined features was again used as a marker of regulatory activity, with `r pander(comma(as.integer(table(features$feature))))` regions defined as enhancers or promoters respectively.

```{r load-motif-db}
meme_db <- here::here("data", "external", "JASPAR2022_CORE_Hsapiens.meme") %>%
  read_meme() %>%
  to_df() %>%
  as_tibble() %>%
  mutate(
    across(everything(), vctrs::vec_proxy),
    altname = str_remove(altname, "MA[0-9]+\\.[0-9]*\\."),
    organism = "Homo sapiens"
  )
meme_db_detected <- meme_db %>%
  mutate(
    gene_name = str_split(altname, pattern = "::", n = 2)
  ) %>%
  dplyr::filter(
    vapply(
      gene_name,
      function(x) all(x %in% detected$gene_name),
      logical(1)
    )
  )
meme_info <- meme_db_detected %>% 
  mutate(motif_length = vapply(motif, function(x){ncol(x@motif)}, integer(1))) %>% 
  dplyr::select(name, altname, icscore, motif_length) 
```

The motif database was initially defined as all `r nrow(meme_db)` CORE motifs found in *Homo sapiens*, as contained in [JASPAR2022](https://jaspar2022.genereg.net/).
In order to minimise irrelevant results, only the `r nrow(meme_db_detected)` motifs associated with a detected gene were then retained.

```{r load-peaks}
dht_peaks <- here::here("data", "peaks") %>%
  list.files(recursive = TRUE, pattern = "oracle", full.names = TRUE) %>%
  sapply(read_rds, simplify = FALSE) %>%
  lapply(function(x) x[["DHT"]]) %>%
  lapply(setNames, nm = c()) %>%
  setNames(str_extract_all(names(.), "AR|FOXA1|GATA3|TFAP2B")) %>% 
  lapply(
    function(x) {
      seqinfo(x) <- sq19
      x
    }
  )
targets <- names(dht_peaks)
dht_consensus <- dht_peaks %>%
  lapply(granges) %>%
  GRangesList() %>%
  unlist() %>%
  reduce() %>%
  mutate(
    AR = overlapsAny(., dht_peaks$AR),
    FOXA1 = overlapsAny(., dht_peaks$FOXA1),
    GATA3 = overlapsAny(., dht_peaks$GATA3),
    TFAP2B = overlapsAny(., dht_peaks$TFAP2B),
    promoter = bestOverlap(., features, var = "feature", missing = "None") == "promoter",
    enhancer = bestOverlap(., features, var = "feature", missing = "None") == "enhancer",
    H3K27ac = promoter | enhancer
  )
```

```{r plot-peak-shape, fig.height=6, fig.cap = "Distributions of A) peak width and B) summit location with peaks."}
a <- dht_peaks %>% 
  lapply(mutate, w = width) %>% 
  lapply(as_tibble) %>% 
  bind_rows(.id = "target") %>% 
  ggplot(aes(w, stat(density), colour = target)) +
  geom_density() +
  scale_x_log10(breaks = c(100, 300, 600, 1e3, 2e3, 3e3), labels = comma) +
  labs(
    x = "Peak Width (bp)", y = "Density", colour = "Target"
  )
b <- dht_peaks %>% 
  lapply(mutate, w = width) %>% 
  lapply(as_tibble) %>% 
  bind_rows(.id = "target") %>% 
  mutate(skew = 0.5 - peak / w) %>% 
  ggplot(aes(skew, stat(density), colour = target)) +
  geom_density() +
  scale_x_continuous(labels = percent) +
  labs(
    x = "% Width Between Peak Summit and Peak Centre",
    y = "Density",
    colour = "ChIP Target"
  )
plot_grid(
  a + theme(legend.position = "none"), 
  b + theme(legend.position = "none"),
  get_legend(b),
  rel_widths = c(1, 1, 0.2), nrow = 1,
  labels = c("A", "B")
)
```


# Enrichment of Expected Motifs

## Analysis of Motif Enrichment (AME)

```{r grl-by-targets}
grl_by_targets <- dht_consensus %>% 
  as_tibble() %>% 
  pivot_longer(
    all_of(targets), names_to = "target", values_to = "detected"
  ) %>% 
  dplyr::filter(detected) %>% 
  group_by(range, H3K27ac) %>% 
  summarise(
    targets = paste(sort(target), collapse = " "),
    .groups = "drop"
  ) %>% 
  colToRanges(var = "range", seqinfo = sq19) %>% 
  sort() %>% 
  splitAsList(.$targets)
```


Union ranges were grouped by the combination of detected `r glue_collapse(targets, sep = ", ", last = " and ")` binding, then sequences obtained corresponding to each union range.

```{r tbl-union-ranges}
grl_by_targets %>% 
  lapply(
    function(x) {
      tibble(
        `N Sites` = length(x), 
        `% H3K27ac` = mean(x$H3K27ac),
        `Median Width` = floor(median(width(x)))
      )
    }
  ) %>% 
  bind_rows(.id = "Targets") %>% 
  mutate(
    `N Targets` = str_count(Targets, " ") + 1,
    `% H3K27ac` = percent(`% H3K27ac`, 0.1)
  ) %>% 
  dplyr::select(contains("Targets"), everything()) %>% 
  arrange(`N Targets`, Targets) %>% 
  pander(
    justify = "lrrrr",
    caption = "Summary of union ranges and which combinations of the targets were detected."
  )
```


```{r ame-by-targets}
meme_db_targets <- meme_db_detected %>% 
  dplyr::filter(altname %in% targets) %>% 
  to_list()
seq_by_targets <- grl_by_targets %>% 
  get_sequence(hg19)
ame_by_targets <- seq_by_targets %>% 
  runAme(database = meme_db_targets)
```

[AME from the meme-suite](https://meme-suite.org/meme/doc/ame.html?man_type=web) was then run on each sequence to detect the presence of motifs associated with each target within each set of sequences.

```{r ame-heatmap, fig.cap = "Estimated percentage of true positive matches to all binding motifs associated with each ChIP target. Only values with an enrichment adjusted p-value < 0.05 are shown, such that missing values are able to be confidently assumed as not enriched for the motif. With the possible exception of GATA3 binding in the presence of AR and FOXA1 only, all provide supporting evidence for direct binding of each target to it's motif"}
ame_by_targets %>% 
  bind_rows(.id = "targets") %>% 
  mutate(
    motif_id = as.factor(motif_id),
    targets = as.factor(targets),
    TFAP2B = ifelse(
      str_detect(targets, "TFAP2B"), "TFAP2B", "No TFAP2B"
    )
  ) %>% 
  dplyr::filter(adj.pvalue < 0.05) %>% 
  plot_ame_heatmap(
    id = motif_id,
    value = tp_percent,
    group = targets
  ) +
  geom_label(
    aes(label = percent(tp_percent/100, 0.1)),
    size = 3,
    show.legend = FALSE
  ) +
  facet_grid(TFAP2B ~ motif_alt_id, scales = "free", space = "free") +
  scale_x_discrete(expand = expansion(c(0, 0))) +
  scale_y_discrete(expand = expansion(c(0, 0))) +
  labs(
    x = "Detected Motif",
    y = "Detected Targets",
    fill = expression(widehat("TP%"))
  ) +
  theme(
    axis.text.x = element_text(angle = 0, hjust = 0.5, size = 10),
    axis.text.y = element_text(angle = 0, hjust = 1, size = 10),
    panel.grid = element_blank()
  )
```


# Motif Centrality Relative to Each Target {.tabset}

```{r centrality-params}
seq_width <- 200
min_score <- percent(0.75, accuracy = 1)
```

In order to assess any association in binding patterns, the binding peak (i.e.) summit of each ChIP target was taken as a viewpoint and the centrality of the best matching sequence motif for that target was assessed.
Sequences were extended `r glue("{seq_width}bp")` from each summit and the best sequence match to each binding motif were retained.
Sequence matches with a score > `r min_score` of the highest possible score were initially included before reducing matches to those with the highest score.

The motifs obtained form the larger database were the `r length(meme_db_targets)` motifs defined as candidate binding sites for each of the `r length(targets)` ChIP targets.

```{r tbl-motifs}
meme_info %>% 
  dplyr::filter(altname %in% targets) %>% 
  arrange(altname, desc(icscore)) %>% 
  dplyr::rename(
    `Motif ID` = name, 
    Target = altname, 
    IC = icscore, 
    `Length (bp)` = motif_length
  ) %>% 
  pander(
    justify = "llrr",
    caption = paste(
      "Motifs use for checking defined target binding sites. Motifs are ranked",
      "within each ChIP target by information content (IC)."
    )
  )
```

```{r plot-motifs, fig.cap = "All motifs matching the set of ChIP targets. Motifc in A-C are not aligned with each other, whilst all TFAP2B motifs are shown using the best alignment."}
plot_grid(
  plot_grid(
    plotlist = lapply(
      meme_db_targets %>% 
        to_df() %>% 
        arrange(altname) %>% 
        dplyr::filter(altname != "TFAP2B") %>% 
        to_list() %>% 
        lapply(
          function(x){
            x@name <- paste0(
              x@name, ": ", x@altname, " (IC = ",
              round(x@icscore, 1), ")"
            )
            x
          }
        ),
      function(x) {
        view_motifs(x) +
          ggtitle(x@name) +
          theme(plot.title = element_text(hjust = 0.5, size = 11))
      }
    ),
    labels = c("A", "B", "C"),
    ncol = 1
  ),
  plot_grid(
    meme_db_targets %>% 
      to_df() %>% 
      dplyr::filter(altname == "TFAP2B") %>% 
      to_list() %>% 
      lapply(
          function(x){
            x@name <- paste0(
              x@name, ": ", x@altname, " (IC = ",
              round(x@icscore, 1), ")"
            )
            x
          }
      ) %>% 
      view_motifs() +
      theme(plot.title = element_text(hjust = 0.5, size = 9))
  ),
  ncol = 2,
  labels = c("", "D")
)
```


## AR

```{r ar-best-motif}
ar_summits <- dht_peaks$AR %>% 
  resize(width = 1, fix = 'start') %>% 
  shift(shift = .$peak) %>% 
  resize(width = seq_width, fix = 'center') %>% 
  granges() %>% 
  mutate(
    AR = overlapsAny(., subset(dht_consensus, AR)),
    FOXA1 = overlapsAny(., subset(dht_consensus, FOXA1)),
    GATA3 = overlapsAny(., subset(dht_consensus, GATA3)),
    TFAP2B = overlapsAny(., subset(dht_consensus, TFAP2B)),
    H3K27ac = overlapsAny(., features)
  ) 
seq_ar_summits <- ar_summits %>% 
  get_sequence(hg19)
views_ar_summits <- stringSetToViews(seq_ar_summits)
matches_ar_summits <- meme_db_targets %>%
  lapply(
    getBestMatch, 
    subject = views_ar_summits,
    min.score = min_score
  ) %>% 
  setNames(
    vapply(meme_db_targets, function(x) x@name, character(1))
  ) %>% 
  bind_rows(.id = "name") %>% 
  left_join(
    dplyr::select(meme_info, name, altname, motif_length),
    by = "name"
  ) %>% 
  left_join(
    ar_summits %>% 
      as_tibble() %>% 
      mutate(id =seq_along(range)),
    by = "id"
  ) %>% 
  arrange(id) %>% 
  mutate(
    target_detected = case_when(
      altname == "AR" ~ AR,
      altname == "FOXA1" ~ FOXA1,
      altname == "GATA3" ~ GATA3,
      altname == "TFAP2B" ~ TFAP2B
    )
  ) %>% 
  dplyr::select(
    range, 
    any_of(targets), H3K27ac, 
    name, altname, target_detected, 
    seq, motif_length, pos, strand
  )
```

```{r plot-centrality-ar, fig.heigth = 6, fig.cap = "Position of the best match for each motif using sequences centred at the summits of AR binding sites. Sites where the ChIP target was not detected are shown as dashed lines, and panels are separated by H3K27ac status. No visual difference in AR motif centrality was detected due to H3K27ac signal. A slight tendency for the best FOXA1  to also appear centrally was noted."}
matches_ar_summits  %>% 
  mutate(
    label = glue("{name}\n({altname})"),
    H3K27ac = ifelse(H3K27ac, "H3K27ac", "No H3K27ac")
  ) %>% 
  ggplot(aes(pos, stat(density), colour = altname)) +
  geom_density(aes(linetype = target_detected)) +
  geom_text(
    aes(x = seq_width/2, y = 0.001, label = lab),
    data = . %>% 
      group_by(label, H3K27ac) %>% 
      summarise(
        n = dplyr::n(), .groups = "drop"
      ) %>% 
      mutate(
        lab = case_when(
          H3K27ac == "H3K27ac" ~ glue(
            "n = {comma(n)}\n({percent(n/sum(ar_summits$H3K27ac), 0.1)})"
          ),
          H3K27ac == "No H3K27ac" ~ glue(
            "n = {comma(n)}\n({percent(n/sum(!ar_summits$H3K27ac), 0.1)})"
          )
        )
      ),
    colour = "black",
    inherit.aes = FALSE
  ) +
  facet_grid(H3K27ac ~ label) +
  scale_linetype_manual(values = c(3, 1)) +
  labs(
    x = "Best Match Position In Sequence",
    y = "Density",
    colour = "Motif Target",
    linetype = "ChIP/Motif\nTarget Detected"
  )
```

Across all motifs which were considered to represent possible matches to the binding site of AR, matches to one or more of the motifs were found in `r comma(length(unique(dplyr::filter(matches_ar_summits, altname == "AR")$range)))` (`r percent(length(unique(dplyr::filter(matches_ar_summits, altname == "AR")$range))/ length(ar_summits), 0.1)`) of the `r comma(length(ar_summits))` queried sequences.

## FOXA1

```{r foxa1-best-motif}
foxa1_summits <- dht_peaks$FOXA1 %>% 
  resize(width = 1, fix = 'start') %>% 
  shift(shift = .$peak) %>% 
  resize(width = seq_width, fix = 'center') %>% 
  granges() %>% 
  mutate(
    AR = overlapsAny(., subset(dht_consensus, AR)),
    FOXA1 = overlapsAny(., subset(dht_consensus, FOXA1)),
    GATA3 = overlapsAny(., subset(dht_consensus, GATA3)),
    TFAP2B = overlapsAny(., subset(dht_consensus, TFAP2B)),
    H3K27ac = overlapsAny(., features)
  ) 
seq_foxa1_summits <- foxa1_summits %>% 
  get_sequence(hg19)
views_foxa1_summits <- stringSetToViews(seq_foxa1_summits)
matches_foxa1_summits <- meme_db_targets %>%
  lapply(
    getBestMatch, 
    subject = views_foxa1_summits,
    min.score = min_score
  ) %>% 
  setNames(
    vapply(meme_db_targets, function(x) x@name, character(1))
  ) %>% 
  bind_rows(.id = "name") %>% 
  left_join(
    dplyr::select(meme_info, name, altname, motif_length),
    by = "name"
  ) %>% 
  left_join(
    foxa1_summits %>% 
      as_tibble() %>% 
      mutate(id =seq_along(range)),
    by = "id"
  ) %>% 
  arrange(id) %>% 
  mutate(
    target_detected = case_when(
      altname == "AR" ~ AR,
      altname == "FOXA1" ~ FOXA1,
      altname == "GATA3" ~ GATA3,
      altname == "TFAP2B" ~ TFAP2B
    )
  ) %>% 
  dplyr::select(
    range, 
    any_of(targets), H3K27ac, 
    name, altname, target_detected, 
    seq, motif_length, pos, strand
  )
```

```{r plot-centrality-foxa1, fig.height=6, fig.cap = "Position of the best match for each motif using sequences centred at the summits of FOXA1 binding sites. Sites where the ChIP target was not detected are shown as dashed lines, and panels are separated by H3K27ac status.  No visual difference in FOXA1 motif centrality was detected due to H3K27ac signal. A slight tendency for the best FOXA1  to also appear centrally was noted."}
matches_foxa1_summits  %>% 
  mutate(
    label = glue("{name}\n({altname})"),
    H3K27ac = ifelse(H3K27ac, "H3K27ac", "No H3K27ac")
  ) %>% 
  ggplot(aes(pos, stat(density), colour = altname)) +
  geom_density(aes(linetype = target_detected)) +
  geom_text(
    aes(x = seq_width/2, y = 0.001, label = lab),
    data = . %>% 
      group_by(label, H3K27ac) %>% 
      summarise(
        n = dplyr::n(), .groups = "drop"
      ) %>% 
      mutate(
        lab = case_when(
          H3K27ac == "H3K27ac" ~ glue(
            "n = {comma(n)}\n({percent(n/sum(foxa1_summits$H3K27ac), 0.1)})"
          ),
          H3K27ac == "No H3K27ac" ~ glue(
            "n = {comma(n)}\n({percent(n/sum(!foxa1_summits$H3K27ac), 0.1)})"
          )
        )
      ),
    colour = "black",
    inherit.aes = FALSE
  ) +
  facet_grid(H3K27ac ~ label) +
  scale_linetype_manual(values = c(3, 1)) +
  labs(
    x = "Best Match Position In Sequence",
    y = "Density",
    colour = "Motif Target",
    linetype = "ChIP/Motif\nTarget Detected"
  )
```

Across all motifs which were considered to represent possible matches to the binding site of FOXA1, matches to one or more of the motifs were found in `r comma(length(unique(dplyr::filter(matches_foxa1_summits, altname == "FOXA1")$range)))` (`r percent(length(unique(dplyr::filter(matches_foxa1_summits, altname == "FOXA1")$range))/ length(foxa1_summits), 0.1)`) of the `r comma(length(foxa1_summits))` queried sequences.

## GATA3

```{r gata3-best-motif}
gata3_summits <- dht_peaks$GATA3 %>% 
  resize(width = 1, fix = 'start') %>% 
  shift(shift = .$peak) %>% 
  resize(width = seq_width, fix = 'center') %>% 
  granges() %>% 
  mutate(
    AR = overlapsAny(., subset(dht_consensus, AR)),
    FOXA1 = overlapsAny(., subset(dht_consensus, FOXA1)),
    GATA3 = overlapsAny(., subset(dht_consensus, GATA3)),
    TFAP2B = overlapsAny(., subset(dht_consensus, TFAP2B)),
    H3K27ac = overlapsAny(., features)
  ) 
seq_gata3_summits <- gata3_summits %>% 
  get_sequence(hg19)
views_gata3_summits <- stringSetToViews(seq_gata3_summits)
matches_gata3_summits <- meme_db_targets %>%
  lapply(
    getBestMatch, 
    subject = views_gata3_summits,
    min.score = min_score
  ) %>% 
  setNames(
    vapply(meme_db_targets, function(x) x@name, character(1))
  ) %>% 
  bind_rows(.id = "name") %>% 
  left_join(
    dplyr::select(meme_info, name, altname, motif_length),
    by = "name"
  ) %>% 
  left_join(
    gata3_summits %>% 
      as_tibble() %>% 
      mutate(id =seq_along(range)),
    by = "id"
  ) %>% 
  arrange(id) %>% 
  mutate(
    target_detected = case_when(
      altname == "AR" ~ AR,
      altname == "FOXA1" ~ FOXA1,
      altname == "GATA3" ~ GATA3,
      altname == "TFAP2B" ~ TFAP2B
    )
  ) %>% 
  dplyr::select(
    range, 
    any_of(targets), H3K27ac, 
    name, altname, target_detected, 
    seq, motif_length, pos, strand
  )
```

```{r plot-centrality-gata3, fig.height = 6, fig.cap = "Position of the best match for each motif using sequences centred at the summits of GATA3 binding sites. Sites where the ChIP target was not detected are shown as dashed lines, and panels are separated by H3K27ac status. No visual difference in GATA3 motif centrality was detected due to H3K27ac signal. A slight tendency for the best FOXA1  to also appear centrally was noted."}
matches_gata3_summits  %>% 
  mutate(
    label = glue("{name}\n({altname})"),
    H3K27ac = ifelse(H3K27ac, "H3K27ac", "No H3K27ac")
  ) %>% 
  ggplot(aes(pos, stat(density), colour = altname)) +
  geom_density(aes(linetype = target_detected)) +
  geom_text(
    aes(x = seq_width/2, y = 0.001, label = lab),
    data = . %>% 
      group_by(label, H3K27ac) %>% 
      summarise(
        n = dplyr::n(), .groups = "drop"
      ) %>% 
      mutate(
        lab = case_when(
          H3K27ac == "H3K27ac" ~ glue(
            "n = {comma(n)}\n({percent(n/sum(gata3_summits$H3K27ac), 0.1)})"
          ),
          H3K27ac == "No H3K27ac" ~ glue(
            "n = {comma(n)}\n({percent(n/sum(!gata3_summits$H3K27ac), 0.1)})"
          )
        )
      ),
    colour = "black",
    inherit.aes = FALSE
  ) +
  facet_grid(H3K27ac ~ label) +
  scale_linetype_manual(values = c(3, 1)) +
  labs(
    x = "Best Match Position In Sequence",
    y = "Density",
    colour = "Motif Target",
    linetype = "ChIP/Motif\nTarget Detected"
  )
```

Across all motifs which were considered to represent possible matches to the binding site of GATA3, matches to one or more of the motifs were found in `r comma(length(unique(dplyr::filter(matches_gata3_summits, altname == "GATA3")$range)))` (`r percent(length(unique(dplyr::filter(matches_gata3_summits, altname == "GATA3")$range))/ length(gata3_summits), 0.1)`) of the `r comma(length(gata3_summits))` queried sequences.


## TFAP2B

```{r tfap2b-best-motif}
tfap2b_summits <- dht_peaks$TFAP2B %>% 
  resize(width = 1, fix = 'start') %>% 
  shift(shift = .$peak) %>% 
  resize(width = seq_width, fix = 'center') %>% 
  granges() %>% 
  mutate(
    AR = overlapsAny(., subset(dht_consensus, AR)),
    FOXA1 = overlapsAny(., subset(dht_consensus, FOXA1)),
    GATA3 = overlapsAny(., subset(dht_consensus, GATA3)),
    TFAP2B = overlapsAny(., subset(dht_consensus, TFAP2B)),
    H3K27ac = overlapsAny(., features)
  ) 
seq_tfap2b_summits <- tfap2b_summits %>% 
  get_sequence(hg19)
views_tfap2b_summits <- stringSetToViews(seq_tfap2b_summits)
matches_tfap2b_summits <- meme_db_targets %>%
  lapply(
    getBestMatch, 
    subject = views_tfap2b_summits,
    min.score = min_score
  ) %>% 
  setNames(
    vapply(meme_db_targets, function(x) x@name, character(1))
  ) %>% 
  bind_rows(.id = "name") %>% 
  left_join(
    dplyr::select(meme_info, name, altname, motif_length),
    by = "name"
  ) %>% 
  left_join(
    tfap2b_summits %>% 
      as_tibble() %>% 
      mutate(id =seq_along(range)),
    by = "id"
  ) %>% 
  arrange(id) %>% 
  mutate(
    target_detected = case_when(
      altname == "AR" ~ AR,
      altname == "FOXA1" ~ FOXA1,
      altname == "GATA3" ~ GATA3,
      altname == "TFAP2B" ~ TFAP2B
    )
  ) %>% 
  dplyr::select(
    range, 
    any_of(targets), H3K27ac, 
    name, altname, target_detected, 
    seq, motif_length, pos, strand
  )
```

```{r plot-centrality-tfap2b, fig.height = 6, fig.cap = "Position of the best match for each motif using sequences centred at the summits of TFAP2B binding sites. Sites where the ChIP target was not detected are shown as dashed lines, and panels are separated by H3K27ac status. No visual difference in TFAP2B motif centrality was detected. All other motifs appeared to be uniformly distributed amongst the sequences."}
matches_tfap2b_summits  %>% 
  mutate(
    label = glue("{name}\n({altname})"),
    H3K27ac = ifelse(H3K27ac, "H3K27ac", "No H3K27ac")
  ) %>% 
  ggplot(aes(pos, stat(density), colour = altname)) +
  geom_density(aes(linetype = target_detected)) +
  geom_text(
    aes(x = seq_width/2, y = 0.001, label = lab),
    data = . %>% 
      group_by(label, H3K27ac) %>% 
      summarise(
        n = dplyr::n(), .groups = "drop"
      ) %>% 
      mutate(
        lab = case_when(
          H3K27ac == "H3K27ac" ~ glue(
            "n = {comma(n)}\n({percent(n/sum(tfap2b_summits$H3K27ac), 0.1)})"
          ),
          H3K27ac == "No H3K27ac" ~ glue(
            "n = {comma(n)}\n({percent(n/sum(!tfap2b_summits$H3K27ac), 0.1)})"
          )
        )
      ),
    colour = "black",
    inherit.aes = FALSE
  ) +
  facet_grid(H3K27ac ~ label) +
  scale_linetype_manual(values = c(3, 1)) +
  labs(
    x = "Best Match Position In Sequence",
    y = "Density",
    colour = "Motif Target",
    linetype = "ChIP/Motif\nTarget Detected"
  )
```

Across all motifs which were considered to represent possible matches to the binding site of TFAP2$\beta$, matches to one or more of the motifs were found in `r comma(length(unique(dplyr::filter(matches_tfap2b_summits, altname == "TFAP2B")$range)))` (`r percent(length(unique(dplyr::filter(matches_tfap2b_summits, altname == "TFAP2B")$range))/ length(tfap2b_summits), 0.1)`) of the `r comma(length(tfap2b_summits))` queried sequences.

# Distance Between Motifs {.tabset}

Whilst the above section revealed some degree of association between AR and FOXA1 sites, as well as clear separation between all targets and TFAP2B motifs, the distribution of distances between motif matches can also be addressed directly.

## AR

```{r plot-ar-dist, fig.height=6, fig.cap = "Distance between best matching motif and the best matching AR motif, showing data to the 50th percentile of distances. No noticable difference was noticed between individual motifs, H3K27ac status, or whether the ChIP target was detected within the AR peak."}
matches_ar_summits %>%
  group_by(range) %>% 
  dplyr::filter(any(altname == "AR")) %>% 
  mutate(
    ar_pos = pos[altname == "AR"],
    ar_strand = strand[altname == "AR"]
  ) %>% 
  ungroup() %>% 
  mutate(
    ar_motif_dist = abs(pos - ar_pos),
    label = glue("{name}\n({altname})"),
    H3K27ac = ifelse(H3K27ac, "H3K27ac", "No H3K27ac")
  ) %>% 
  dplyr::filter(altname != "AR") %>% 
  group_by(name, target_detected, H3K27ac) %>% 
  mutate(
    q = rank(ar_motif_dist, ties = "first") / length(ar_motif_dist)
  ) %>% 
  dplyr::filter(q <= 0.5) %>%
  ggplot(
    aes(ar_motif_dist, q, colour = altname, linetype = target_detected)
  ) +
  geom_line() +
  facet_grid(H3K27ac ~ label) +
  scale_y_continuous(labels= percent) +
  scale_linetype_manual(values = c(3, 1)) +
  labs(
    x = "Distance from AR motif",
    y = "Percentile",
    colour = "Motif Target",
    linetype = "ChIP/Motif\nTarget Detected"
  )
```

## FOXA1

```{r plot-foxa1-dist, fig.height=6, fig.cap = "Distance between best matching motif and the best matching FOXA1 motif, showing data to the 50th percentile of distances. No noticable difference was noticed between individual motifs, H3K27ac status, or whether the ChIP target was detected within the FOXA1 peak."}
matches_foxa1_summits %>%
  group_by(range) %>% 
  dplyr::filter(any(altname == "FOXA1")) %>% 
  mutate(
    foxa1_pos = pos[altname == "FOXA1"],
    foxa1_strand = strand[altname == "FOXA1"]
  ) %>% 
  ungroup() %>% 
  mutate(
    foxa1_motif_dist = abs(pos - foxa1_pos),
    label = glue("{name}\n({altname})"),
    H3K27ac = ifelse(H3K27ac, "H3K27ac", "No H3K27ac")
  ) %>% 
  dplyr::filter(altname != "FOXA1") %>% 
  group_by(name, target_detected, H3K27ac) %>% 
  mutate(
    q = rank(foxa1_motif_dist, ties = "first") / length(foxa1_motif_dist)
  ) %>% 
  dplyr::filter(q <= 0.5) %>%
  ggplot(
    aes(foxa1_motif_dist, q, colour = altname, linetype = target_detected)
  ) +
  geom_line() +
  facet_grid(H3K27ac ~ label) +
  scale_y_continuous(labels= percent) +
  scale_linetype_manual(values = c(3, 1)) +
  labs(
    x = "Distance from FOXA1 motif",
    y = "Percentile",
    colour = "Motif Target",
    linetype = "ChIP/Motif\nTarget Detected"
  )
```


## GATA3

```{r plot-gata3-dist, fig.height=6, fig.cap = "Distance between best matching motif and the best matching GATA3 motif, showing data to the 50th percentile of distances. FOXA1 motifs tended to be further from the GATA3 motif for peaks where FOXA1 was detected."}
matches_gata3_summits %>%
  group_by(range) %>% 
  dplyr::filter(any(altname == "GATA3")) %>% 
  mutate(
    gata3_pos = pos[altname == "GATA3"],
    gata3_strand = strand[altname == "GATA3"]
  ) %>% 
  ungroup() %>% 
  mutate(
    gata3_motif_dist = abs(pos - gata3_pos),
    label = glue("{name}\n({altname})"),
    H3K27ac = ifelse(H3K27ac, "H3K27ac", "No H3K27ac")
  ) %>% 
  dplyr::filter(altname != "GATA3") %>% 
  group_by(name, target_detected, H3K27ac) %>% 
  mutate(
    q = rank(gata3_motif_dist, ties = "first") / length(gata3_motif_dist)
  ) %>% 
  dplyr::filter(q <= 0.5) %>%
  ggplot(
    aes(gata3_motif_dist, q, colour = altname, linetype = target_detected)
  ) +
  geom_line() +
  facet_grid(H3K27ac ~ label) +
  scale_y_continuous(labels= percent) +
  scale_linetype_manual(values = c(3, 1)) +
  labs(
    x = "Distance from GATA3 motif",
    y = "Percentile",
    colour = "Motif Target",
    linetype = "ChIP/Motif\nTarget Detected"
  )
```


## TFAP2B

```{r plot-tfap2b-dist, fig.height=6, fig.width = 8, fig.cap = "Distance between best matching motif and the best matching TFAP2B motif, showing data to the 50th percentile of distances. No noticable difference was noticed between individual motifs, H3K27ac status, or whether the ChIP target was detected within the TFAP2B peak."}
matches_tfap2b_summits %>%
  group_by(range) %>% 
  dplyr::filter(any(altname == "TFAP2B")) %>% 
  mutate(
    tfap2b_pos = pos[altname == "TFAP2B"][1],
  ) %>% 
  ungroup() %>% 
  mutate(
    tfap2b_motif_dist = abs(pos - tfap2b_pos),
    label = glue("{name}\n({altname})"),
    H3K27ac = ifelse(H3K27ac, "H3K27ac", "No H3K27ac")
  ) %>% 
  dplyr::filter(altname != "TFAP2B") %>% 
  group_by(name, target_detected, H3K27ac) %>% 
  mutate(
    q = rank(tfap2b_motif_dist, ties = "first") / length(tfap2b_motif_dist)
  ) %>% 
  dplyr::filter(q <= 0.5) %>%
  ggplot(
    aes(tfap2b_motif_dist, q, colour = altname, linetype = target_detected)
  ) +
  geom_line() +
  facet_grid(H3K27ac ~ label) +
  scale_y_continuous(labels= percent) +
  scale_linetype_manual(values = c(3, 1)) +
  labs(
    x = "Distance from TFAP2B motif",
    y = "Percentile",
    colour = "Motif Target",
    linetype = "ChIP/Motif\nTarget Detected"
  )
```


# Motif Discovery {.tabset}

The above steps essentially perform checks on the data and verify that the expected binding motifs are generally driving binding of each target.
A further step would be to determine any enriched binding motifs within the union peaks which are able to be confidently assigned to putative co-factors.
For this section, key comparisons were chosen as this enables detection of cofactors which may be driving key binding patterns.
Novel sequence motifs were first defined using STREME, which uses a suffix tree based approach to determining enrichment of sequence motifs.

## All 4 Targets Bound: H3K27ac Vs No H3K27ac

A comparison between binding sites where all 4 targets were found was performed based on this showing regulatory activity (overlap with H3K27ac-derived features) and no regulatory activity (i.e. no H3K27ac overlap).

```{r streme-all4-by-h3k27ac}
seq_all4_by_h3k27ac <- dht_consensus %>%
  filter(!!!syms(targets)) %>%
  # resize(width = 800, fix = 'center') %>%
  split(.$H3K27ac) %>%
  setNames(c("no_h3k27ac", "h3k27ac")) %>%
  get_sequence(hg19) %>%
  as.list()
streme_all4_by_h3k27ac_rds <- here::here(
  "output", "streme", "all4_by_h3k27ac.rds"
)
if (!dir.exists(dirname(streme_all4_by_h3k27ac_rds))) dir.create(
  dirname(streme_all4_by_h3k27ac_rds)
)
if (!file.exists(streme_all4_by_h3k27ac_rds)) {
  streme_all4_by_h3k27ac <- runStreme(
    input = seq_all4_by_h3k27ac$h3k27ac,
    control = seq_all4_by_h3k27ac$no_h3k27ac,
    thresh = 0.01
  ) %>% 
    as_tibble() %>% 
    mutate(
      eval = case_when(
        is.na(eval) ~ pval * nrow(.),
        !is.na(eval) ~ eval
      )
    )
  write_rds(
    x = streme_all4_by_h3k27ac,
    file = streme_all4_by_h3k27ac_rds
  )
} else {
  streme_all4_by_h3k27ac <- read_rds(streme_all4_by_h3k27ac_rds)
}
tomtom_all4_by_h3k27ac <- streme_all4_by_h3k27ac %>%
  to_list() %>%
  runTomTom(
    database = to_list(meme_db_detected),
    thresh = 0.1
  )
streme_to_keep_all4_by_h3k27ac <- tomtom_all4_by_h3k27ac %>% 
  lapply(function(x) x$best_match_name) %>% 
  vapply(length, integer(1)) %>% 
  is_greater_than(0)
```

Using the default setting of STREME, `r length(streme_all4_by_h3k27ac)` candidate sequence motifs were detected.
TomTom was subsequently used to match candidate motifs with those in the database of detected sequences.
To ensure only high-quality candidates were retained, matches with a p-value > 0.1 for similarity were discarded.
This strategy gave `r sum(streme_to_keep_all4_by_h3k27ac)` candidate motifs able to be matched to the database and retained as high-quality motifs.

```{r plot-motif-streme-all4-by-h3k27ac, fig.height=8, fig.cap = "All motifs detected by STREME using a threshold of $p \\leq  0.01$ and with high-quality matches to the database of motifs. The name of the best matching motif is given as the second line of the title, although this may not be the true TF which binds to the motif. Motifs are ordered by the number of times they appear in the sequences, noting the motifs with high information content tend to lead to fewer hits."}
pl <- tomtom_all4_by_h3k27ac[streme_to_keep_all4_by_h3k27ac] %>%
  bind_rows() %>% 
  arrange(desc(nsites)) %>% 
  to_list() %>% 
  lapply(
    function(x) {
      nm <- paste0(
        x@altname, 
        " (IC: ", round(x@icscore, 1), "; n = ", x@nsites, ")\n",
        x@extrainfo[["best_match_altname"]]
      )
      view_motifs(x) +
        ggtitle(nm) +
        theme(
          plot.title = element_text(hjust = 0.5, size = 8),
          axis.text = element_text(size = 8),
          axis.title = element_text(size = 8)
        )
    }
  )
plot_grid(plotlist = pl)
```



```{r tbl-tomtom-all4-by-h3k27ac}
tomtom_all4_by_h3k27ac[streme_to_keep_all4_by_h3k27ac] %>% 
  lapply(
    function(x) {
      tt <- x$tomtom[[1]]
      mt <- lapply(
        tt$match_motif,
        function(um) {
          nm <- paste0(um@name, " (", um@altname, ")")
          um@name <- nm
          um
        }
      )
      x$tomtom[[1]]$match_motif <- mt
      x
    }
  ) %>% 
  lapply(
    function(x) {
      mutate(x, tomtom_logos = view_tomtom_hits(x))
    }
  ) %>% 
  bind_rows() %>% 
  # dplyr::slice(1:2) %>%
  dplyr::select(
    motif, id = altname, IC = icscore, nsites,
    best_match_name, best_match_altname, best_match_motif,
    tomtom_logos, tomtom, pval , eval
  ) %>% 
  mutate(
    tomtom_logos = tomtom_logos %>% 
      lapply(
        function(x) {
          n <- length(levels(x$data$motif.id))
          ggplot_image(x, height = px(60 + n*60), aspect_ratio = 180 / (100 + n*30))
        }
      ),
    `All Matches` = vapply(
      tomtom,
      function(x) {
        paste(x$match_altname, collapse = ", ")
      },
      character(1)
    )
  ) %>% 
  dplyr::select(
    `STREME ID` = id, `Best Match` = best_match_altname, `All Matches`,
    IC, `Found In` = nsites, `P-value` = pval, `E-value` = eval,
    `TOMTOM Matches` = tomtom_logos
  ) %>% 
  gt(
    caption = paste(
      "All STREME-detected motifs with putative matches in the motif database.",
      "The motif database was restricted to motifs associated with detected genes.",
      "The information content (IC) is given along with the number of sequences",
      "the motif was found in. The E-value is equivalent to an adjusted p-value",
      "for the likelihood of the detected motif being non-artefactual."
    ),
    id = "tomtom"
  ) %>% 
  fmt_markdown(contains("TOMTOM")) %>% 
  fmt_number(contains("val"), decimals = 4) %>% 
  fmt_number("Found In", decimals = 0, use_seps = TRUE) %>% 
  fmt_number("IC", decimals = 2) %>% 
  tab_spanner(
    label = "Database Matches",
    columns = all_of(c("Best Match", "All Matches"))
  ) %>% 
  tab_spanner(
    label = "Statistics",
    columns = c("IC", "Found In", "P-value", "E-value")
  ) %>% 
  text_transform(
    locations = cells_body(columns = "All Matches"),
    fn = function(x) str_replace_all(x, ", ", "<br>")
  ) %>% 
  tab_options(
    table.width = pct(100),
    table.font.size = px(12)
  ) %>% 
  opt_css(
    css = "
    #tomtom .gt_row{
      vertical-align: top
    }
    "
  )
```

